--- net/minecraft/world/level/levelgen/feature/configurations/FeatureConfiguration.java
+++ net/minecraft/world/level/levelgen/feature/configurations/FeatureConfiguration.java
@@ -1,10 +1,5 @@
 package net.minecraft.world.level.levelgen.feature.configurations;
 
-import com.mojang.datafixers.Dynamic;
-import com.mojang.datafixers.types.DynamicOps;
-
 public interface FeatureConfiguration {
-   NoneFeatureConfiguration NONE = new NoneFeatureConfiguration();
-
-   <T> Dynamic<T> serialize(DynamicOps<T> var1);
+   NoneFeatureConfiguration NONE = NoneFeatureConfiguration.INSTANCE;
 }
