--- net/minecraft/commands/arguments/NbtPathArgument.java
+++ net/minecraft/commands/arguments/NbtPathArgument.java
@@ -30,11 +30,11 @@
 import net.minecraft.network.chat.TranslatableComponent;
 import org.apache.commons.lang3.mutable.MutableBoolean;
 
 public class NbtPathArgument implements ArgumentType<NbtPathArgument.NbtPath> {
    private static final Collection<String> EXAMPLES = Arrays.asList("foo", "foo.bar", "foo[0]", "[0]", "[]", "{foo=bar}");
-   public static final SimpleCommandExceptionType ERROR_INVALID_NODE = new SimpleCommandExceptionType(new TranslatableComponent("arguments.nbtpath.node.invalid", new Object[0]));
+   public static final SimpleCommandExceptionType ERROR_INVALID_NODE = new SimpleCommandExceptionType(new TranslatableComponent("arguments.nbtpath.node.invalid"));
    public static final DynamicCommandExceptionType ERROR_NOTHING_FOUND = new DynamicCommandExceptionType((var0) -> {
       return new TranslatableComponent("arguments.nbtpath.nothing_found", new Object[]{var0});
    });
 
    public NbtPathArgument() {
