--- net/minecraft/commands/arguments/OperationArgument.java
+++ net/minecraft/commands/arguments/OperationArgument.java
@@ -16,12 +16,12 @@
 import net.minecraft.util.Mth;
 import net.minecraft.world.scores.Score;
 
 public class OperationArgument implements ArgumentType<OperationArgument.Operation> {
    private static final Collection<String> EXAMPLES = Arrays.asList("=", ">", "<");
-   private static final SimpleCommandExceptionType ERROR_INVALID_OPERATION = new SimpleCommandExceptionType(new TranslatableComponent("arguments.operation.invalid", new Object[0]));
-   private static final SimpleCommandExceptionType ERROR_DIVIDE_BY_ZERO = new SimpleCommandExceptionType(new TranslatableComponent("arguments.operation.div0", new Object[0]));
+   private static final SimpleCommandExceptionType ERROR_INVALID_OPERATION = new SimpleCommandExceptionType(new TranslatableComponent("arguments.operation.invalid"));
+   private static final SimpleCommandExceptionType ERROR_DIVIDE_BY_ZERO = new SimpleCommandExceptionType(new TranslatableComponent("arguments.operation.div0"));
 
    public OperationArgument() {
    }
 
    public static OperationArgument operation() {
