--- net/minecraft/commands/arguments/TimeArgument.java
+++ net/minecraft/commands/arguments/TimeArgument.java
@@ -16,11 +16,11 @@
 import net.minecraft.commands.SharedSuggestionProvider;
 import net.minecraft.network.chat.TranslatableComponent;
 
 public class TimeArgument implements ArgumentType<Integer> {
    private static final Collection<String> EXAMPLES = Arrays.asList("0d", "0s", "0t", "0");
-   private static final SimpleCommandExceptionType ERROR_INVALID_UNIT = new SimpleCommandExceptionType(new TranslatableComponent("argument.time.invalid_unit", new Object[0]));
+   private static final SimpleCommandExceptionType ERROR_INVALID_UNIT = new SimpleCommandExceptionType(new TranslatableComponent("argument.time.invalid_unit"));
    private static final DynamicCommandExceptionType ERROR_INVALID_TICK_COUNT = new DynamicCommandExceptionType((var0) -> {
       return new TranslatableComponent("argument.time.invalid_tick_count", new Object[]{var0});
    });
    private static final Object2IntMap<String> UNITS = new Object2IntOpenHashMap();
 
