--- net/minecraft/world/level/ClipContext.java
+++ net/minecraft/world/level/ClipContext.java
@@ -1,10 +1,11 @@
 package net.minecraft.world.level;
 
 import java.util.function.Predicate;
 import net.minecraft.core.BlockPos;
 import net.minecraft.world.entity.Entity;
+import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.material.FluidState;
 import net.minecraft.world.phys.Vec3;
 import net.minecraft.world.phys.shapes.CollisionContext;
 import net.minecraft.world.phys.shapes.Shapes;
@@ -64,12 +65,13 @@
    public interface ShapeGetter {
       VoxelShape get(BlockState var1, BlockGetter var2, BlockPos var3, CollisionContext var4);
    }
 
    public static enum Block implements ClipContext.ShapeGetter {
-      COLLIDER(BlockState::getCollisionShape),
-      OUTLINE(BlockState::getShape);
+      COLLIDER(BlockBehaviour.BlockStateBase::getCollisionShape),
+      OUTLINE(BlockBehaviour.BlockStateBase::getShape),
+      VISUAL(BlockBehaviour.BlockStateBase::getVisualShape);
 
       private final ClipContext.ShapeGetter shapeGetter;
 
       private Block(ClipContext.ShapeGetter var3) {
          this.shapeGetter = var3;
