--- net/minecraft/client/renderer/debug/PathfindingRenderer.java
+++ net/minecraft/client/renderer/debug/PathfindingRenderer.java
@@ -114,12 +114,12 @@
 
       if (var3) {
          for(var11 = 0; var11 < var0.getSize(); ++var11) {
             var12 = var0.get(var11);
             if (distanceToCamera(var12.asBlockPos(), var4, var6, var8) <= 80.0F) {
-               DebugRenderer.renderFloatingText(String.format("%s", var12.type), (double)var12.x + 0.5D, (double)var12.y + 0.75D, (double)var12.z + 0.5D, -1);
-               DebugRenderer.renderFloatingText(String.format(Locale.ROOT, "%.2f", var12.costMalus), (double)var12.x + 0.5D, (double)var12.y + 0.25D, (double)var12.z + 0.5D, -1);
+               DebugRenderer.renderFloatingText(String.format("%s", var12.type), (double)var12.x + 0.5D, (double)var12.y + 0.75D, (double)var12.z + 0.5D, -1, 0.02F, true, 0.0F, true);
+               DebugRenderer.renderFloatingText(String.format(Locale.ROOT, "%.2f", var12.costMalus), (double)var12.x + 0.5D, (double)var12.y + 0.25D, (double)var12.z + 0.5D, -1, 0.02F, true, 0.0F, true);
             }
          }
       }
 
    }
