--- net/minecraft/client/gui/screens/recipebook/AbstractFurnaceRecipeBookComponent.java
+++ net/minecraft/client/gui/screens/recipebook/AbstractFurnaceRecipeBookComponent.java
@@ -1,16 +1,18 @@
 package net.minecraft.client.gui.screens.recipebook;
 
 import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Set;
 import javax.annotation.Nullable;
 import net.minecraft.client.gui.GuiComponent;
 import net.minecraft.client.gui.screens.Screen;
-import net.minecraft.client.resources.language.I18n;
 import net.minecraft.core.NonNullList;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TranslatableComponent;
 import net.minecraft.world.inventory.Slot;
 import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.crafting.Ingredient;
 import net.minecraft.world.item.crafting.Recipe;
@@ -54,15 +56,15 @@
 
    protected void initFilterButtonTextures() {
       this.filterButton.initTextureValues(152, 182, 28, 18, RECIPE_BOOK_LOCATION);
    }
 
-   protected String getFilterButtonTooltip() {
-      return I18n.get(this.filterButton.isStateTriggered() ? this.getRecipeFilterName() : "gui.recipebook.toggleRecipes.all");
+   protected Component getFilterButtonTooltip() {
+      return (Component)(this.filterButton.isStateTriggered() ? this.getRecipeFilterName() : new TranslatableComponent("gui.recipebook.toggleRecipes.all"));
    }
 
-   protected abstract String getRecipeFilterName();
+   protected abstract Component getRecipeFilterName();
 
    public void slotClicked(@Nullable Slot var1) {
       super.slotClicked(var1);
       if (var1 != null && var1.index < this.menu.getSize()) {
          this.fuelSlot = null;
@@ -98,23 +100,23 @@
 
    }
 
    protected abstract Set<Item> getFuelItems();
 
-   public void renderGhostRecipe(int var1, int var2, boolean var3, float var4) {
-      super.renderGhostRecipe(var1, var2, var3, var4);
+   public void renderGhostRecipe(PoseStack var1, int var2, int var3, boolean var4, float var5) {
+      super.renderGhostRecipe(var1, var2, var3, var4, var5);
       if (this.fuelSlot != null) {
          if (!Screen.hasControlDown()) {
-            this.time += var4;
+            this.time += var5;
          }
 
-         int var5 = this.fuelSlot.x + var1;
-         int var6 = this.fuelSlot.y + var2;
-         GuiComponent.fill(var5, var6, var5 + 16, var6 + 16, 822018048);
-         this.minecraft.getItemRenderer().renderAndDecorateItem(this.minecraft.player, this.getFuel().getDefaultInstance(), var5, var6);
+         int var6 = this.fuelSlot.x + var2;
+         int var7 = this.fuelSlot.y + var3;
+         GuiComponent.fill(var1, var6, var7, var6 + 16, var7 + 16, 822018048);
+         this.minecraft.getItemRenderer().renderAndDecorateItem(this.minecraft.player, this.getFuel().getDefaultInstance(), var6, var7);
          RenderSystem.depthFunc(516);
-         GuiComponent.fill(var5, var6, var5 + 16, var6 + 16, 822083583);
+         GuiComponent.fill(var1, var6, var7, var6 + 16, var7 + 16, 822083583);
          RenderSystem.depthFunc(515);
       }
    }
 
    private Item getFuel() {
