--- net/minecraft/client/gui/screens/recipebook/SmeltingRecipeBookComponent.java
+++ net/minecraft/client/gui/screens/recipebook/SmeltingRecipeBookComponent.java
@@ -1,34 +1,20 @@
 package net.minecraft.client.gui.screens.recipebook;
 
 import java.util.Set;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TranslatableComponent;
 import net.minecraft.world.item.Item;
 import net.minecraft.world.level.block.entity.AbstractFurnaceBlockEntity;
 
 public class SmeltingRecipeBookComponent extends AbstractFurnaceRecipeBookComponent {
    public SmeltingRecipeBookComponent() {
    }
 
-   protected boolean getFilteringCraftable() {
-      return this.book.isFurnaceFilteringCraftable();
+   protected Component getRecipeFilterName() {
+      return new TranslatableComponent("gui.recipebook.toggleRecipes.smeltable");
    }
 
-   protected void setFilteringCraftable(boolean var1) {
-      this.book.setFurnaceFilteringCraftable(var1);
-   }
-
-   protected boolean isGuiOpen() {
-      return this.book.isFurnaceGuiOpen();
-   }
-
-   protected void setGuiOpen(boolean var1) {
-      this.book.setFurnaceGuiOpen(var1);
-   }
-
-   protected String getRecipeFilterName() {
-      return "gui.recipebook.toggleRecipes.smeltable";
-   }
-
    protected Set<Item> getFuelItems() {
       return AbstractFurnaceBlockEntity.getFuel().keySet();
    }
 }
