--- net/minecraft/world/level/pathfinder/NodeEvaluator.java
+++ net/minecraft/world/level/pathfinder/NodeEvaluator.java
@@ -1,9 +1,10 @@
 package net.minecraft.world.level.pathfinder;
 
 import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
 import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+import net.minecraft.core.BlockPos;
 import net.minecraft.util.Mth;
 import net.minecraft.world.entity.Mob;
 import net.minecraft.world.level.BlockGetter;
 import net.minecraft.world.level.PathNavigationRegion;
 
@@ -33,10 +34,14 @@
    public void done() {
       this.level = null;
       this.mob = null;
    }
 
+   protected Node getNode(BlockPos var1) {
+      return this.getNode(var1.getX(), var1.getY(), var1.getZ());
+   }
+
    protected Node getNode(int var1, int var2, int var3) {
       return (Node)this.nodes.computeIfAbsent(Node.createHash(var1, var2, var3), (var3x) -> {
          return new Node(var1, var2, var3);
       });
    }
