--- net/minecraft/world/entity/HumanoidArm.java
+++ net/minecraft/world/entity/HumanoidArm.java
@@ -2,12 +2,12 @@
 
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.TranslatableComponent;
 
 public enum HumanoidArm {
-   LEFT(new TranslatableComponent("options.mainHand.left", new Object[0])),
-   RIGHT(new TranslatableComponent("options.mainHand.right", new Object[0]));
+   LEFT(new TranslatableComponent("options.mainHand.left")),
+   RIGHT(new TranslatableComponent("options.mainHand.right"));
 
    private final Component name;
 
    private HumanoidArm(Component var3) {
       this.name = var3;
