--- net/minecraft/util/datafix/fixes/BlockEntityJukeboxFix.java
+++ net/minecraft/util/datafix/fixes/BlockEntityJukeboxFix.java
@@ -1,14 +1,14 @@
 package net.minecraft.util.datafix.fixes;
 
 import com.mojang.datafixers.DSL;
-import com.mojang.datafixers.Dynamic;
 import com.mojang.datafixers.OpticFinder;
 import com.mojang.datafixers.Typed;
 import com.mojang.datafixers.schemas.Schema;
 import com.mojang.datafixers.types.Type;
-import java.util.Optional;
+import com.mojang.datafixers.util.Pair;
+import com.mojang.serialization.Dynamic;
 
 public class BlockEntityJukeboxFix extends NamedEntityFix {
    public BlockEntityJukeboxFix(Schema var1, boolean var2) {
       super(var1, var2, "BlockEntityJukeboxFix", References.BLOCK_ENTITY, "minecraft:jukebox");
    }
@@ -24,13 +24,13 @@
          String var7 = ItemStackTheFlatteningFix.updateItem(ItemIdFix.getItem(var6), 0);
          if (var7 != null) {
             Dynamic var8 = var5.emptyMap();
             var8 = var8.set("id", var8.createString(var7));
             var8 = var8.set("Count", var8.createByte((byte)1));
-            return var1.set(var4, (Typed)((Optional)var3.readTyped(var8).getSecond()).orElseThrow(() -> {
+            return var1.set(var4, (Typed)((Pair)var3.readTyped(var8).result().orElseThrow(() -> {
                return new IllegalStateException("Could not create record item stack.");
-            })).set(DSL.remainderFinder(), (Object)var5);
+            })).getFirst()).set(DSL.remainderFinder(), (Object)var5);
          }
       }
 
       return var1;
    }
