--- net/minecraft/util/datafix/fixes/EntityRenameFix.java
+++ net/minecraft/util/datafix/fixes/EntityRenameFix.java
@@ -2,14 +2,14 @@
 
 import com.mojang.datafixers.DataFix;
 import com.mojang.datafixers.TypeRewriteRule;
 import com.mojang.datafixers.Typed;
 import com.mojang.datafixers.schemas.Schema;
-import com.mojang.datafixers.types.DynamicOps;
 import com.mojang.datafixers.types.Type;
 import com.mojang.datafixers.types.templates.TaggedChoice;
 import com.mojang.datafixers.util.Pair;
+import com.mojang.serialization.DynamicOps;
 
 public abstract class EntityRenameFix extends DataFix {
    protected final String name;
 
    public EntityRenameFix(String var1, Schema var2, boolean var3) {
