--- net/minecraft/util/datafix/fixes/OptionsLowerCaseLanguageFix.java
+++ net/minecraft/util/datafix/fixes/OptionsLowerCaseLanguageFix.java
@@ -13,11 +13,11 @@
    }
 
    public TypeRewriteRule makeRule() {
       return this.fixTypeEverywhereTyped("OptionsLowerCaseLanguageFix", this.getInputSchema().getType(References.OPTIONS), (var0) -> {
          return var0.update(DSL.remainderFinder(), (var0x) -> {
-            Optional var1 = var0x.get("lang").asString();
+            Optional var1 = var0x.get("lang").asString().result();
             return var1.isPresent() ? var0x.set("lang", var0x.createString(((String)var1.get()).toLowerCase(Locale.ROOT))) : var0x;
          });
       });
    }
 }
