--- net/minecraft/util/datafix/fixes/OptionsKeyTranslationFix.java
+++ net/minecraft/util/datafix/fixes/OptionsKeyTranslationFix.java
@@ -1,13 +1,13 @@
 package net.minecraft.util.datafix.fixes;
 
 import com.mojang.datafixers.DSL;
 import com.mojang.datafixers.DataFix;
-import com.mojang.datafixers.Dynamic;
 import com.mojang.datafixers.TypeRewriteRule;
 import com.mojang.datafixers.schemas.Schema;
 import com.mojang.datafixers.util.Pair;
+import com.mojang.serialization.Dynamic;
 import java.util.Map;
 import java.util.stream.Collectors;
 
 public class OptionsKeyTranslationFix extends DataFix {
    public OptionsKeyTranslationFix(Schema var1, boolean var2) {
@@ -26,10 +26,10 @@
                      }
                   }
 
                   return Pair.of(var1x.getKey(), var1x.getValue());
                }).collect(Collectors.toMap(Pair::getFirst, Pair::getSecond)));
-            }).orElse(var0x);
+            }).result().orElse(var0x);
          });
       });
    }
 }
