--- net/minecraft/core/dispenser/AbstractProjectileDispenseBehavior.java
+++ net/minecraft/core/dispenser/AbstractProjectileDispenseBehavior.java
@@ -1,27 +1,27 @@
 package net.minecraft.core.dispenser;
 
 import net.minecraft.core.BlockSource;
 import net.minecraft.core.Direction;
 import net.minecraft.core.Position;
-import net.minecraft.world.entity.Entity;
+import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.entity.projectile.Projectile;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.DispenserBlock;
 
 public abstract class AbstractProjectileDispenseBehavior extends DefaultDispenseItemBehavior {
    public AbstractProjectileDispenseBehavior() {
    }
 
    public ItemStack execute(BlockSource var1, ItemStack var2) {
-      Level var3 = var1.getLevel();
+      ServerLevel var3 = var1.getLevel();
       Position var4 = DispenserBlock.getDispensePosition(var1);
       Direction var5 = (Direction)var1.getBlockState().getValue(DispenserBlock.FACING);
       Projectile var6 = this.getProjectile(var3, var4, var2);
       var6.shoot((double)var5.getStepX(), (double)((float)var5.getStepY() + 0.1F), (double)var5.getStepZ(), this.getPower(), this.getUncertainty());
-      var3.addFreshEntity((Entity)var6);
+      var3.addFreshEntity(var6);
       var2.shrink(1);
       return var2;
    }
 
    protected void playSound(BlockSource var1) {
