--- net/minecraft/core/dispenser/BoatDispenseItemBehavior.java
+++ net/minecraft/core/dispenser/BoatDispenseItemBehavior.java
@@ -1,14 +1,14 @@
 package net.minecraft.core.dispenser;
 
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.BlockSource;
 import net.minecraft.core.Direction;
+import net.minecraft.server.level.ServerLevel;
 import net.minecraft.tags.FluidTags;
 import net.minecraft.world.entity.vehicle.Boat;
 import net.minecraft.world.item.ItemStack;
-import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.DispenserBlock;
 
 public class BoatDispenseItemBehavior extends DefaultDispenseItemBehavior {
    private final DefaultDispenseItemBehavior defaultDispenseItemBehavior = new DefaultDispenseItemBehavior();
    private final Boat.Type type;
@@ -17,11 +17,11 @@
       this.type = var1;
    }
 
    public ItemStack execute(BlockSource var1, ItemStack var2) {
       Direction var3 = (Direction)var1.getBlockState().getValue(DispenserBlock.FACING);
-      Level var4 = var1.getLevel();
+      ServerLevel var4 = var1.getLevel();
       double var5 = var1.x() + (double)((float)var3.getStepX() * 1.125F);
       double var7 = var1.y() + (double)((float)var3.getStepY() * 1.125F);
       double var9 = var1.z() + (double)((float)var3.getStepZ() * 1.125F);
       BlockPos var11 = var1.getPos().relative(var3);
       double var12;
