--- net/minecraft/server/commands/data/EntityDataAccessor.java
+++ net/minecraft/server/commands/data/EntityDataAccessor.java
@@ -18,11 +18,11 @@
 import net.minecraft.network.chat.TranslatableComponent;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.player.Player;
 
 public class EntityDataAccessor implements DataAccessor {
-   private static final SimpleCommandExceptionType ERROR_NO_PLAYERS = new SimpleCommandExceptionType(new TranslatableComponent("commands.data.entity.invalid", new Object[0]));
+   private static final SimpleCommandExceptionType ERROR_NO_PLAYERS = new SimpleCommandExceptionType(new TranslatableComponent("commands.data.entity.invalid"));
    public static final Function<String, DataCommands.DataProvider> PROVIDER = (var0) -> {
       return new DataCommands.DataProvider() {
          public DataAccessor access(CommandContext<CommandSourceStack> var1) throws CommandSyntaxException {
             return new EntityDataAccessor(EntityArgument.getEntity(var1, var0));
          }
