--- com/mojang/authlib/Environment.java
+++ com/mojang/authlib/Environment.java
@@ -1,0 +1,39 @@
+package com.mojang.authlib;
+
+import java.util.StringJoiner;
+
+public interface Environment {
+   String getAuthHost();
+
+   String getAccountsHost();
+
+   String getSessionHost();
+
+   String getName();
+
+   String asString();
+
+   static Environment create(final String var0, final String var1, final String var2, final String var3) {
+      return new Environment() {
+         public String getAuthHost() {
+            return var0;
+         }
+
+         public String getAccountsHost() {
+            return var1;
+         }
+
+         public String getSessionHost() {
+            return var2;
+         }
+
+         public String getName() {
+            return var3;
+         }
+
+         public String asString() {
+            return (new StringJoiner(", ", "", "")).add("authHost='" + this.getAuthHost() + "'").add("accountsHost='" + this.getAccountsHost() + "'").add("sessionHost='" + this.getSessionHost() + "'").add("name='" + this.getName() + "'").toString();
+         }
+      };
+   }
+}
