--- net/minecraft/client/KeyMapping.java
+++ net/minecraft/client/KeyMapping.java
@@ -3,15 +3,16 @@
 import com.google.common.collect.Maps;
 import com.google.common.collect.Sets;
 import com.mojang.blaze3d.platform.InputConstants;
 import java.util.Iterator;
 import java.util.Map;
-import java.util.Objects;
 import java.util.Set;
 import java.util.function.Supplier;
 import net.minecraft.Util;
 import net.minecraft.client.resources.language.I18n;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TranslatableComponent;
 
 public class KeyMapping implements Comparable<KeyMapping> {
    private static final Map<String, KeyMapping> ALL = Maps.newHashMap();
    private static final Map<InputConstants.Key, KeyMapping> MAP = Maps.newHashMap();
    private static final Set<String> CATEGORIES = Sets.newHashSet();
@@ -130,14 +131,14 @@
 
    public int compareTo(KeyMapping var1) {
       return this.category.equals(var1.category) ? I18n.get(this.name).compareTo(I18n.get(var1.name)) : ((Integer)CATEGORY_SORT_ORDER.get(this.category)).compareTo((Integer)CATEGORY_SORT_ORDER.get(var1.category));
    }
 
-   public static Supplier<String> createNameSupplier(String var0) {
+   public static Supplier<Component> createNameSupplier(String var0) {
       KeyMapping var1 = (KeyMapping)ALL.get(var0);
       return var1 == null ? () -> {
-         return var0;
+         return new TranslatableComponent(var0);
       } : var1::getTranslatedKeyMessage;
    }
 
    public boolean same(KeyMapping var1) {
       return this.key.equals(var1.key);
@@ -157,27 +158,12 @@
 
    public boolean matchesMouse(int var1) {
       return this.key.getType() == InputConstants.Type.MOUSE && this.key.getValue() == var1;
    }
 
-   public String getTranslatedKeyMessage() {
-      String var1 = this.key.getName();
-      int var2 = this.key.getValue();
-      String var3 = null;
-      switch(this.key.getType()) {
-      case KEYSYM:
-         var3 = InputConstants.translateKeyCode(var2);
-         break;
-      case SCANCODE:
-         var3 = InputConstants.translateScanCode(var2);
-         break;
-      case MOUSE:
-         String var4 = I18n.get(var1);
-         var3 = Objects.equals(var4, var1) ? I18n.get(InputConstants.Type.MOUSE.getDefaultPrefix(), var2 + 1) : var4;
-      }
-
-      return var3 == null ? I18n.get(var1) : var3;
+   public Component getTranslatedKeyMessage() {
+      return this.key.getDisplayName();
    }
 
    public boolean isDefault() {
       return this.key.equals(this.defaultKey);
    }
