--- net/minecraft/client/ProgressOption.java
+++ net/minecraft/client/ProgressOption.java
@@ -3,21 +3,22 @@
 import java.util.function.BiConsumer;
 import java.util.function.BiFunction;
 import java.util.function.Function;
 import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.components.SliderButton;
+import net.minecraft.network.chat.Component;
 import net.minecraft.util.Mth;
 
 public class ProgressOption extends Option {
    protected final float steps;
    protected final double minValue;
    protected double maxValue;
    private final Function<Options, Double> getter;
    private final BiConsumer<Options, Double> setter;
-   private final BiFunction<Options, ProgressOption, String> toString;
+   private final BiFunction<Options, ProgressOption, Component> toString;
 
-   public ProgressOption(String var1, double var2, double var4, float var6, Function<Options, Double> var7, BiConsumer<Options, Double> var8, BiFunction<Options, ProgressOption, String> var9) {
+   public ProgressOption(String var1, double var2, double var4, float var6, Function<Options, Double> var7, BiConsumer<Options, Double> var8, BiFunction<Options, ProgressOption, Component> var9) {
       super(var1);
       this.minValue = var2;
       this.maxValue = var4;
       this.steps = var6;
       this.getter = var7;
@@ -63,9 +64,9 @@
 
    public double get(Options var1) {
       return (Double)this.getter.apply(var1);
    }
 
-   public String getMessage(Options var1) {
-      return (String)this.toString.apply(var1, this);
+   public Component getMessage(Options var1) {
+      return (Component)this.toString.apply(var1, this);
    }
 }
