--- net/minecraft/world/level/block/entity/BlastFurnaceBlockEntity.java
+++ net/minecraft/world/level/block/entity/BlastFurnaceBlockEntity.java
@@ -12,11 +12,11 @@
    public BlastFurnaceBlockEntity() {
       super(BlockEntityType.BLAST_FURNACE, RecipeType.BLASTING);
    }
 
    protected Component getDefaultName() {
-      return new TranslatableComponent("container.blast_furnace", new Object[0]);
+      return new TranslatableComponent("container.blast_furnace");
    }
 
    protected int getBurnDuration(ItemStack var1) {
       return super.getBurnDuration(var1) / 2;
    }
