--- net/minecraft/world/level/block/entity/DropperBlockEntity.java
+++ net/minecraft/world/level/block/entity/DropperBlockEntity.java
@@ -7,8 +7,8 @@
    public DropperBlockEntity() {
       super(BlockEntityType.DROPPER);
    }
 
    protected Component getDefaultName() {
-      return new TranslatableComponent("container.dropper", new Object[0]);
+      return new TranslatableComponent("container.dropper");
    }
 }
