--- net/minecraft/nbt/StringTag.java
+++ net/minecraft/nbt/StringTag.java
@@ -3,10 +3,11 @@
 import java.io.DataInput;
 import java.io.DataOutput;
 import java.io.IOException;
 import java.util.Objects;
 import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
 import net.minecraft.network.chat.TextComponent;
 
 public class StringTag implements Tag {
    public static final TagType<StringTag> TYPE = new TagType<StringTag>() {
       public StringTag load(DataInput var1, int var2, NbtAccounter var3) throws IOException {
@@ -77,11 +78,11 @@
    }
 
    public Component getPrettyDisplay(String var1, int var2) {
       String var3 = quoteAndEscape(this.data);
       String var4 = var3.substring(0, 1);
-      Component var5 = (new TextComponent(var3.substring(1, var3.length() - 1))).withStyle(SYNTAX_HIGHLIGHTING_STRING);
+      MutableComponent var5 = (new TextComponent(var3.substring(1, var3.length() - 1))).withStyle(SYNTAX_HIGHLIGHTING_STRING);
       return (new TextComponent(var4)).append(var5).append(var4);
    }
 
    public static String quoteAndEscape(String var0) {
       StringBuilder var1 = new StringBuilder(" ");
