--- net/minecraft/client/renderer/entity/BoatRenderer.java
+++ net/minecraft/client/renderer/entity/BoatRenderer.java
@@ -43,12 +43,15 @@
       var4.scale(-1.0F, -1.0F, 1.0F);
       var4.mulPose(Vector3f.YP.rotationDegrees(90.0F));
       this.model.setupAnim(var1, var3, 0.0F, -0.1F, 0.0F, 0.0F);
       VertexConsumer var10 = var5.getBuffer(this.model.renderType(this.getTextureLocation(var1)));
       this.model.renderToBuffer(var4, var10, var6, OverlayTexture.NO_OVERLAY, 1.0F, 1.0F, 1.0F, 1.0F);
-      VertexConsumer var11 = var5.getBuffer(RenderType.waterMask());
-      this.model.waterPatch().render(var4, var11, var6, OverlayTexture.NO_OVERLAY);
+      if (!var1.isUnderWater()) {
+         VertexConsumer var11 = var5.getBuffer(RenderType.waterMask());
+         this.model.waterPatch().render(var4, var11, var6, OverlayTexture.NO_OVERLAY);
+      }
+
       var4.popPose();
       super.render(var1, var2, var3, var4, var5, var6);
    }
 
    public ResourceLocation getTextureLocation(Boat var1) {
