--- net/minecraft/client/renderer/entity/CatRenderer.java
+++ net/minecraft/client/renderer/entity/CatRenderer.java
@@ -32,11 +32,11 @@
       super.setupRotations(var1, var2, var3, var4, var5);
       float var6 = var1.getLieDownAmount(var5);
       if (var6 > 0.0F) {
          var2.translate((double)(0.4F * var6), (double)(0.15F * var6), (double)(0.1F * var6));
          var2.mulPose(Vector3f.ZP.rotationDegrees(Mth.rotLerp(var6, 0.0F, 90.0F)));
-         BlockPos var7 = new BlockPos(var1);
+         BlockPos var7 = var1.blockPosition();
          List var8 = var1.level.getEntitiesOfClass(Player.class, (new AABB(var7)).inflate(2.0D, 2.0D, 2.0D));
          Iterator var9 = var8.iterator();
 
          while(var9.hasNext()) {
             Player var10 = (Player)var9.next();
