--- net/minecraft/client/renderer/entity/DragonFireballRenderer.java
+++ net/minecraft/client/renderer/entity/DragonFireballRenderer.java
@@ -6,10 +6,11 @@
 import com.mojang.math.Matrix4f;
 import com.mojang.math.Vector3f;
 import net.minecraft.client.renderer.MultiBufferSource;
 import net.minecraft.client.renderer.RenderType;
 import net.minecraft.client.renderer.texture.OverlayTexture;
+import net.minecraft.core.BlockPos;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.entity.projectile.DragonFireball;
 
 public class DragonFireballRenderer extends EntityRenderer<DragonFireball> {
    private static final ResourceLocation TEXTURE_LOCATION = new ResourceLocation("textures/entity/enderdragon/dragon_fireball.png");
@@ -17,11 +18,11 @@
 
    public DragonFireballRenderer(EntityRenderDispatcher var1) {
       super(var1);
    }
 
-   protected int getBlockLightLevel(DragonFireball var1, float var2) {
+   protected int getBlockLightLevel(DragonFireball var1, BlockPos var2) {
       return 15;
    }
 
    public void render(DragonFireball var1, float var2, float var3, PoseStack var4, MultiBufferSource var5, int var6) {
       var4.pushPose();
