--- net/minecraft/client/renderer/entity/DrownedRenderer.java
+++ net/minecraft/client/renderer/entity/DrownedRenderer.java
@@ -20,11 +20,11 @@
    public ResourceLocation getTextureLocation(Zombie var1) {
       return DROWNED_LOCATION;
    }
 
    protected void setupRotations(Drowned var1, PoseStack var2, float var3, float var4, float var5) {
-      super.setupRotations((Zombie)var1, var2, var3, var4, var5);
+      super.setupRotations(var1, var2, var3, var4, var5);
       float var6 = var1.getSwimAmount(var5);
       if (var6 > 0.0F) {
          var2.mulPose(Vector3f.XP.rotationDegrees(Mth.lerp(var6, var1.xRot, -10.0F - var1.xRot)));
       }
 
