--- net/minecraft/client/gui/components/OptionsList.java
+++ net/minecraft/client/gui/components/OptionsList.java
@@ -1,8 +1,9 @@
 package net.minecraft.client.gui.components;
 
 import com.google.common.collect.ImmutableList;
+import com.mojang.blaze3d.vertex.PoseStack;
 import java.util.List;
 import javax.annotation.Nullable;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.Option;
 import net.minecraft.client.Options;
@@ -51,14 +52,14 @@
       public static OptionsList.Entry small(Options var0, int var1, Option var2, @Nullable Option var3) {
          AbstractWidget var4 = var2.createButton(var0, var1 / 2 - 155, 0, 150);
          return var3 == null ? new OptionsList.Entry(ImmutableList.of(var4)) : new OptionsList.Entry(ImmutableList.of(var4, var3.createButton(var0, var1 / 2 - 155 + 160, 0, 150)));
       }
 
-      public void render(int var1, int var2, int var3, int var4, int var5, int var6, int var7, boolean var8, float var9) {
-         this.children.forEach((var4x) -> {
-            var4x.y = var2;
-            var4x.render(var6, var7, var9);
+      public void render(PoseStack var1, int var2, int var3, int var4, int var5, int var6, int var7, int var8, boolean var9, float var10) {
+         this.children.forEach((var5x) -> {
+            var5x.y = var3;
+            var5x.render(var1, var7, var8, var10);
          });
       }
 
       public List<? extends GuiEventListener> children() {
          return this.children;
