--- net/minecraft/world/phys/shapes/VoxelShape.java
+++ net/minecraft/world/phys/shapes/VoxelShape.java
@@ -85,19 +85,10 @@
          var1.add(new AABB(var1x, var3, var5, var7, var9, var11));
       });
       return var1;
    }
 
-   public double min(Direction.Axis var1, double var2, double var4) {
-      Direction.Axis var6 = AxisCycle.FORWARD.cycle(var1);
-      Direction.Axis var7 = AxisCycle.BACKWARD.cycle(var1);
-      int var8 = this.findIndex(var6, var2);
-      int var9 = this.findIndex(var7, var4);
-      int var10 = this.shape.firstFull(var1, var8, var9);
-      return var10 >= this.shape.getSize(var1) ? Double.POSITIVE_INFINITY : this.get(var1, var10);
-   }
-
    public double max(Direction.Axis var1, double var2, double var4) {
       Direction.Axis var6 = AxisCycle.FORWARD.cycle(var1);
       Direction.Axis var7 = AxisCycle.BACKWARD.cycle(var1);
       int var8 = this.findIndex(var6, var2);
       int var9 = this.findIndex(var7, var4);
