--- net/minecraft/world/level/levelgen/flat/FlatLayerInfo.java
+++ net/minecraft/world/level/levelgen/flat/FlatLayerInfo.java
@@ -1,12 +1,22 @@
 package net.minecraft.world.level.levelgen.flat;
 
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import java.util.function.BiFunction;
 import net.minecraft.core.Registry;
+import net.minecraft.util.Codecs;
 import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.state.BlockState;
 
 public class FlatLayerInfo {
+   public static final Codec<FlatLayerInfo> CODEC = RecordCodecBuilder.create((var0) -> {
+      return var0.group(Codecs.intRange(0, 256).fieldOf("height").forGetter(FlatLayerInfo::getHeight), Registry.BLOCK.fieldOf("block").withDefault((Object)Blocks.AIR).forGetter((var0x) -> {
+         return var0x.getBlockState().getBlock();
+      })).apply(var0, (BiFunction)(FlatLayerInfo::new));
+   });
    private final BlockState blockState;
    private final int height;
    private int start;
 
    public FlatLayerInfo(int var1, Block var2) {
