--- net/minecraft/world/level/newbiome/layer/AddMushroomIslandLayer.java
+++ net/minecraft/world/level/newbiome/layer/AddMushroomIslandLayer.java
@@ -1,16 +1,16 @@
 package net.minecraft.world.level.newbiome.layer;
 
-import net.minecraft.core.Registry;
+import net.minecraft.data.BuiltinRegistries;
 import net.minecraft.world.level.biome.Biomes;
 import net.minecraft.world.level.newbiome.context.Context;
 import net.minecraft.world.level.newbiome.layer.traits.BishopTransformer;
 
 public enum AddMushroomIslandLayer implements BishopTransformer {
    INSTANCE;
 
-   private static final int MUSHROOM_FIELDS = Registry.BIOME.getId(Biomes.MUSHROOM_FIELDS);
+   private static final int MUSHROOM_FIELDS = BuiltinRegistries.BIOME.getId(Biomes.MUSHROOM_FIELDS);
 
    private AddMushroomIslandLayer() {
    }
 
    public int apply(Context var1, int var2, int var3, int var4, int var5, int var6) {
