--- net/minecraft/commands/arguments/blocks/BlockStateArgument.java
+++ net/minecraft/commands/arguments/blocks/BlockStateArgument.java
@@ -8,10 +8,11 @@
 import com.mojang.brigadier.suggestion.SuggestionsBuilder;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.concurrent.CompletableFuture;
 import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.tags.BlockTags;
 
 public class BlockStateArgument implements ArgumentType<BlockInput> {
    private static final Collection<String> EXAMPLES = Arrays.asList("stone", "minecraft:stone", "stone[foo=bar]", "foo{bar=baz}");
 
    public BlockStateArgument() {
@@ -38,11 +39,11 @@
       try {
          var4.parse(true);
       } catch (CommandSyntaxException var6) {
       }
 
-      return var4.fillSuggestions(var2);
+      return var4.fillSuggestions(var2, BlockTags.getAllTags());
    }
 
    public Collection<String> getExamples() {
       return EXAMPLES;
    }
