--- net/minecraft/world/level/block/grower/JungleTreeGrower.java
+++ net/minecraft/world/level/block/grower/JungleTreeGrower.java
@@ -4,22 +4,21 @@
 import javax.annotation.Nullable;
 import net.minecraft.world.level.biome.BiomeDefaultFeatures;
 import net.minecraft.world.level.levelgen.feature.ConfiguredFeature;
 import net.minecraft.world.level.levelgen.feature.Feature;
 import net.minecraft.world.level.levelgen.feature.TreeFeature;
-import net.minecraft.world.level.levelgen.feature.configurations.MegaTreeConfiguration;
-import net.minecraft.world.level.levelgen.feature.configurations.SmallTreeConfiguration;
+import net.minecraft.world.level.levelgen.feature.configurations.TreeConfiguration;
 
 public class JungleTreeGrower extends AbstractMegaTreeGrower {
    public JungleTreeGrower() {
    }
 
    @Nullable
-   protected ConfiguredFeature<SmallTreeConfiguration, ?> getConfiguredFeature(Random var1, boolean var2) {
-      return (new TreeFeature(SmallTreeConfiguration::deserialize)).configured(BiomeDefaultFeatures.JUNGLE_TREE_NOVINE_CONFIG);
+   protected ConfiguredFeature<TreeConfiguration, ?> getConfiguredFeature(Random var1, boolean var2) {
+      return (new TreeFeature(TreeConfiguration.CODEC)).configured(BiomeDefaultFeatures.JUNGLE_TREE_NOVINE_CONFIG);
    }
 
    @Nullable
-   protected ConfiguredFeature<MegaTreeConfiguration, ?> getConfiguredMegaFeature(Random var1) {
-      return Feature.MEGA_JUNGLE_TREE.configured(BiomeDefaultFeatures.MEGA_JUNGLE_TREE_CONFIG);
+   protected ConfiguredFeature<TreeConfiguration, ?> getConfiguredMegaFeature(Random var1) {
+      return Feature.TREE.configured(BiomeDefaultFeatures.MEGA_JUNGLE_TREE_CONFIG);
    }
 }
