--- net/minecraft/core/BlockSource.java
+++ net/minecraft/core/BlockSource.java
@@ -1,11 +1,12 @@
 package net.minecraft.core;
 
+import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.level.block.entity.BlockEntity;
 import net.minecraft.world.level.block.state.BlockState;
 
-public interface BlockSource extends LocatableSource {
+public interface BlockSource extends Position {
    double x();
 
    double y();
 
    double z();
@@ -13,6 +14,8 @@
    BlockPos getPos();
 
    BlockState getBlockState();
 
    <T extends BlockEntity> T getEntity();
+
+   ServerLevel getLevel();
 }
