--- net/minecraft/server/packs/repository/RepositorySource.java
+++ net/minecraft/server/packs/repository/RepositorySource.java
@@ -1,7 +1,7 @@
 package net.minecraft.server.packs.repository;
 
-import java.util.Map;
+import java.util.function.Consumer;
 
 public interface RepositorySource {
-   <T extends UnopenedPack> void loadPacks(Map<String, T> var1, UnopenedPack.UnopenedPackConstructor<T> var2);
+   void loadPacks(Consumer<Pack> var1, Pack.PackConstructor var2);
 }
