--- net/minecraft/client/gui/spectator/PlayerMenuItem.java
+++ net/minecraft/client/gui/spectator/PlayerMenuItem.java
@@ -2,10 +2,11 @@
 
 import com.mojang.authlib.GameProfile;
 import com.mojang.authlib.minecraft.MinecraftProfileTexture;
 import com.mojang.authlib.minecraft.MinecraftProfileTexture.Type;
 import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
 import java.util.Map;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiComponent;
 import net.minecraft.client.resources.DefaultPlayerSkin;
 import net.minecraft.network.chat.Component;
@@ -37,15 +38,15 @@
 
    public Component getName() {
       return new TextComponent(this.profile.getName());
    }
 
-   public void renderIcon(float var1, int var2) {
+   public void renderIcon(PoseStack var1, float var2, int var3) {
       Minecraft.getInstance().getTextureManager().bind(this.location);
-      RenderSystem.color4f(1.0F, 1.0F, 1.0F, (float)var2 / 255.0F);
-      GuiComponent.blit(2, 2, 12, 12, 8.0F, 8.0F, 8, 8, 64, 64);
-      GuiComponent.blit(2, 2, 12, 12, 40.0F, 8.0F, 8, 8, 64, 64);
+      RenderSystem.color4f(1.0F, 1.0F, 1.0F, (float)var3 / 255.0F);
+      GuiComponent.blit(var1, 2, 2, 12, 12, 8.0F, 8.0F, 8, 8, 64, 64);
+      GuiComponent.blit(var1, 2, 2, 12, 12, 40.0F, 8.0F, 8, 8, 64, 64);
    }
 
    public boolean isEnabled() {
       return true;
    }
