--- net/minecraft/world/level/biome/BiomeManager.java
+++ net/minecraft/world/level/biome/BiomeManager.java
@@ -1,7 +1,8 @@
 package net.minecraft.world.level.biome;
 
+import com.google.common.hash.Hashing;
 import net.minecraft.core.BlockPos;
 
 public class BiomeManager {
    private final BiomeManager.NoiseBiomeSource noiseBiomeSource;
    private final long biomeZoomSeed;
@@ -11,10 +12,14 @@
       this.noiseBiomeSource = var1;
       this.biomeZoomSeed = var2;
       this.zoomer = var4;
    }
 
+   public static long obfuscateSeed(long var0) {
+      return Hashing.sha256().hashLong(var0).asLong();
+   }
+
    public BiomeManager withDifferentSource(BiomeSource var1) {
       return new BiomeManager(var1, this.biomeZoomSeed, this.zoomer);
    }
 
    public Biome getBiome(BlockPos var1) {
