--- net/minecraft/advancements/CriterionTrigger.java
+++ net/minecraft/advancements/CriterionTrigger.java
@@ -1,9 +1,9 @@
 package net.minecraft.advancements;
 
-import com.google.gson.JsonDeserializationContext;
 import com.google.gson.JsonObject;
+import net.minecraft.advancements.critereon.DeserializationContext;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.PlayerAdvancements;
 
 public interface CriterionTrigger<T extends CriterionTriggerInstance> {
    ResourceLocation getId();
@@ -12,11 +12,11 @@
 
    void removePlayerListener(PlayerAdvancements var1, CriterionTrigger.Listener<T> var2);
 
    void removePlayerListeners(PlayerAdvancements var1);
 
-   T createInstance(JsonObject var1, JsonDeserializationContext var2);
+   T createInstance(JsonObject var1, DeserializationContext var2);
 
    public static class Listener<T extends CriterionTriggerInstance> {
       private final T trigger;
       private final Advancement advancement;
       private final String criterion;
