--- net/minecraft/world/level/levelgen/structure/templatesystem/NopProcessor.java
+++ net/minecraft/world/level/levelgen/structure/templatesystem/NopProcessor.java
@@ -1,27 +1,25 @@
 package net.minecraft.world.level.levelgen.structure.templatesystem;
 
-import com.mojang.datafixers.Dynamic;
-import com.mojang.datafixers.types.DynamicOps;
+import com.mojang.serialization.Codec;
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
 import net.minecraft.world.level.LevelReader;
 
 public class NopProcessor extends StructureProcessor {
+   public static final Codec<NopProcessor> CODEC = Codec.unit(() -> {
+      return INSTANCE;
+   });
    public static final NopProcessor INSTANCE = new NopProcessor();
 
    private NopProcessor() {
    }
 
    @Nullable
-   public StructureTemplate.StructureBlockInfo processBlock(LevelReader var1, BlockPos var2, StructureTemplate.StructureBlockInfo var3, StructureTemplate.StructureBlockInfo var4, StructurePlaceSettings var5) {
-      return var4;
+   public StructureTemplate.StructureBlockInfo processBlock(LevelReader var1, BlockPos var2, BlockPos var3, StructureTemplate.StructureBlockInfo var4, StructureTemplate.StructureBlockInfo var5, StructurePlaceSettings var6) {
+      return var5;
    }
 
-   protected StructureProcessorType getType() {
+   protected StructureProcessorType<?> getType() {
       return StructureProcessorType.NOP;
    }
-
-   protected <T> Dynamic<T> getDynamic(DynamicOps<T> var1) {
-      return new Dynamic(var1, var1.emptyMap());
-   }
 }
