--- net/minecraft/commands/arguments/item/ItemPredicateArgument.java
+++ net/minecraft/commands/arguments/item/ItemPredicateArgument.java
@@ -15,10 +15,11 @@
 import net.minecraft.commands.CommandSourceStack;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.nbt.NbtUtils;
 import net.minecraft.network.chat.TranslatableComponent;
 import net.minecraft.resources.ResourceLocation;
+import net.minecraft.tags.ItemTags;
 import net.minecraft.tags.Tag;
 import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 
 public class ItemPredicateArgument implements ArgumentType<ItemPredicateArgument.Result> {
@@ -66,11 +67,11 @@
       try {
          var4.parse();
       } catch (CommandSyntaxException var6) {
       }
 
-      return var4.fillSuggestions(var2);
+      return var4.fillSuggestions(var2, ItemTags.getAllTags());
    }
 
    public Collection<String> getExamples() {
       return EXAMPLES;
    }
