--- net/minecraft/client/resources/sounds/SoundInstance.java
+++ net/minecraft/client/resources/sounds/SoundInstance.java
@@ -24,22 +24,26 @@
 
    float getVolume();
 
    float getPitch();
 
-   float getX();
+   double getX();
 
-   float getY();
+   double getY();
 
-   float getZ();
+   double getZ();
 
    SoundInstance.Attenuation getAttenuation();
 
    default boolean canStartSilent() {
       return false;
    }
 
+   default boolean canPlaySound() {
+      return true;
+   }
+
    public static enum Attenuation {
       NONE,
       LINEAR;
 
       private Attenuation() {
