--- net/minecraft/world/level/block/state/properties/EnumProperty.java
+++ net/minecraft/world/level/block/state/properties/EnumProperty.java
@@ -11,11 +11,11 @@
 import java.util.Optional;
 import java.util.function.Predicate;
 import java.util.stream.Collectors;
 import net.minecraft.util.StringRepresentable;
 
-public class EnumProperty<T extends Enum<T> & StringRepresentable> extends AbstractProperty<T> {
+public class EnumProperty<T extends Enum<T> & StringRepresentable> extends Property<T> {
    private final ImmutableSet<T> values;
    private final Map<String, T> names = Maps.newHashMap();
 
    protected EnumProperty(String var1, Class<T> var2, Collection<T> var3) {
       super(var1, var2);
