--- net/minecraft/network/syncher/EntityDataSerializers.java
+++ net/minecraft/network/syncher/EntityDataSerializers.java
@@ -84,11 +84,11 @@
       public Component read(FriendlyByteBuf var1) {
          return var1.readComponent();
       }
 
       public Component copy(Component var1) {
-         return var1.deepCopy();
+         return var1;
       }
    };
    public static final EntityDataSerializer<Optional<Component>> OPTIONAL_COMPONENT = new EntityDataSerializer<Optional<Component>>() {
       public void write(FriendlyByteBuf var1, Optional<Component> var2) {
          if (var2.isPresent()) {
@@ -103,11 +103,11 @@
       public Optional<Component> read(FriendlyByteBuf var1) {
          return var1.readBoolean() ? Optional.of(var1.readComponent()) : Optional.empty();
       }
 
       public Optional<Component> copy(Optional<Component> var1) {
-         return var1.isPresent() ? Optional.of(((Component)var1.get()).deepCopy()) : Optional.empty();
+         return var1;
       }
    };
    public static final EntityDataSerializer<ItemStack> ITEM_STACK = new EntityDataSerializer<ItemStack>() {
       public void write(FriendlyByteBuf var1, ItemStack var2) {
          var1.writeItem(var2);
