--- net/minecraft/world/level/block/BaseEntityBlock.java
+++ net/minecraft/world/level/block/BaseEntityBlock.java
@@ -3,14 +3,15 @@
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
 import net.minecraft.world.MenuProvider;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 
 public abstract class BaseEntityBlock extends Block implements EntityBlock {
-   protected BaseEntityBlock(Block.Properties var1) {
+   protected BaseEntityBlock(BlockBehaviour.Properties var1) {
       super(var1);
    }
 
    public RenderShape getRenderShape(BlockState var1) {
       return RenderShape.INVISIBLE;
