--- net/minecraft/world/level/block/EndPortalFrameBlock.java
+++ net/minecraft/world/level/block/EndPortalFrameBlock.java
@@ -1,13 +1,14 @@
 package net.minecraft.world.level.block;
 
 import com.google.common.base.Predicates;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
-import net.minecraft.world.item.BlockPlaceContext;
+import net.minecraft.world.item.context.BlockPlaceContext;
 import net.minecraft.world.level.BlockGetter;
 import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.block.state.StateDefinition;
 import net.minecraft.world.level.block.state.pattern.BlockInWorld;
 import net.minecraft.world.level.block.state.pattern.BlockPattern;
 import net.minecraft.world.level.block.state.pattern.BlockPatternBuilder;
@@ -26,11 +27,11 @@
    protected static final VoxelShape BASE_SHAPE;
    protected static final VoxelShape EYE_SHAPE;
    protected static final VoxelShape FULL_SHAPE;
    private static BlockPattern portalShape;
 
-   public EndPortalFrameBlock(Block.Properties var1) {
+   public EndPortalFrameBlock(BlockBehaviour.Properties var1) {
       super(var1);
       this.registerDefaultState((BlockState)((BlockState)((BlockState)this.stateDefinition.any()).setValue(FACING, Direction.NORTH)).setValue(HAS_EYE, false));
    }
 
    public boolean useShapeForLightOcclusion(BlockState var1) {
