--- net/minecraft/world/level/block/GravelBlock.java
+++ net/minecraft/world/level/block/GravelBlock.java
@@ -1,7 +1,9 @@
 package net.minecraft.world.level.block;
 
+import net.minecraft.world.level.block.state.BlockBehaviour;
+
 public class GravelBlock extends FallingBlock {
-   public GravelBlock(Block.Properties var1) {
+   public GravelBlock(BlockBehaviour.Properties var1) {
       super(var1);
    }
 }
