--- net/minecraft/world/level/block/HalfTransparentBlock.java
+++ net/minecraft/world/level/block/HalfTransparentBlock.java
@@ -1,7 +1,9 @@
 package net.minecraft.world.level.block;
 
+import net.minecraft.world.level.block.state.BlockBehaviour;
+
 public class HalfTransparentBlock extends Block {
-   protected HalfTransparentBlock(Block.Properties var1) {
+   protected HalfTransparentBlock(BlockBehaviour.Properties var1) {
       super(var1);
    }
 }
