--- net/minecraft/world/level/block/MyceliumBlock.java
+++ net/minecraft/world/level/block/MyceliumBlock.java
@@ -1,7 +1,9 @@
 package net.minecraft.world.level.block;
 
+import net.minecraft.world.level.block.state.BlockBehaviour;
+
 public class MyceliumBlock extends SpreadingSnowyDirtBlock {
-   public MyceliumBlock(Block.Properties var1) {
+   public MyceliumBlock(BlockBehaviour.Properties var1) {
       super(var1);
    }
 }
