--- net/minecraft/world/level/block/StructureVoidBlock.java
+++ net/minecraft/world/level/block/StructureVoidBlock.java
@@ -1,18 +1,19 @@
 package net.minecraft.world.level.block;
 
 import net.minecraft.core.BlockPos;
 import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.material.PushReaction;
 import net.minecraft.world.phys.shapes.CollisionContext;
 import net.minecraft.world.phys.shapes.VoxelShape;
 
 public class StructureVoidBlock extends Block {
    private static final VoxelShape SHAPE = Block.box(5.0D, 5.0D, 5.0D, 11.0D, 11.0D, 11.0D);
 
-   protected StructureVoidBlock(Block.Properties var1) {
+   protected StructureVoidBlock(BlockBehaviour.Properties var1) {
       super(var1);
    }
 
    public RenderShape getRenderShape(BlockState var1) {
       return RenderShape.INVISIBLE;
