--- net/minecraft/world/level/block/TallSeagrass.java
+++ net/minecraft/world/level/block/TallSeagrass.java
@@ -6,10 +6,11 @@
 import net.minecraft.tags.FluidTags;
 import net.minecraft.world.item.BlockPlaceContext;
 import net.minecraft.world.level.BlockGetter;
 import net.minecraft.world.level.LevelAccessor;
 import net.minecraft.world.level.LevelReader;
+import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.block.state.properties.DoubleBlockHalf;
 import net.minecraft.world.level.block.state.properties.EnumProperty;
 import net.minecraft.world.level.material.Fluid;
 import net.minecraft.world.level.material.FluidState;
@@ -19,20 +20,20 @@
 
 public class TallSeagrass extends ShearableDoublePlantBlock implements LiquidBlockContainer {
    public static final EnumProperty<DoubleBlockHalf> HALF;
    protected static final VoxelShape SHAPE;
 
-   public TallSeagrass(Block.Properties var1) {
+   public TallSeagrass(BlockBehaviour.Properties var1) {
       super(var1);
    }
 
    public VoxelShape getShape(BlockState var1, BlockGetter var2, BlockPos var3, CollisionContext var4) {
       return SHAPE;
    }
 
    protected boolean mayPlaceOn(BlockState var1, BlockGetter var2, BlockPos var3) {
-      return var1.isFaceSturdy(var2, var3, Direction.UP) && var1.getBlock() != Blocks.MAGMA_BLOCK;
+      return var1.isFaceSturdy(var2, var3, Direction.UP) && !var1.is(Blocks.MAGMA_BLOCK);
    }
 
    @Nullable
    public BlockState getStateForPlacement(BlockPlaceContext var1) {
       BlockState var2 = super.getStateForPlacement(var1);
@@ -47,11 +48,11 @@
    }
 
    public boolean canSurvive(BlockState var1, LevelReader var2, BlockPos var3) {
       if (var1.getValue(HALF) == DoubleBlockHalf.UPPER) {
          BlockState var5 = var2.getBlockState(var3.below());
-         return var5.getBlock() == this && var5.getValue(HALF) == DoubleBlockHalf.LOWER;
+         return var5.is(this) && var5.getValue(HALF) == DoubleBlockHalf.LOWER;
       } else {
          FluidState var4 = var2.getFluidState(var3);
          return super.canSurvive(var1, var2, var3) && var4.is(FluidTags.WATER) && var4.getAmount() == 8;
       }
    }
