--- net/minecraft/advancements/critereon/EntityEquipmentPredicate.java
+++ net/minecraft/advancements/critereon/EntityEquipmentPredicate.java
@@ -86,6 +86,52 @@
 
    static {
       ANY = new EntityEquipmentPredicate(ItemPredicate.ANY, ItemPredicate.ANY, ItemPredicate.ANY, ItemPredicate.ANY, ItemPredicate.ANY, ItemPredicate.ANY);
       CAPTAIN = new EntityEquipmentPredicate(ItemPredicate.Builder.item().of((ItemLike)Items.WHITE_BANNER).hasNbt(Raid.getLeaderBannerInstance().getTag()).build(), ItemPredicate.ANY, ItemPredicate.ANY, ItemPredicate.ANY, ItemPredicate.ANY, ItemPredicate.ANY);
    }
+
+   public static class Builder {
+      private ItemPredicate head;
+      private ItemPredicate chest;
+      private ItemPredicate legs;
+      private ItemPredicate feet;
+      private ItemPredicate mainhand;
+      private ItemPredicate offhand;
+
+      public Builder() {
+         this.head = ItemPredicate.ANY;
+         this.chest = ItemPredicate.ANY;
+         this.legs = ItemPredicate.ANY;
+         this.feet = ItemPredicate.ANY;
+         this.mainhand = ItemPredicate.ANY;
+         this.offhand = ItemPredicate.ANY;
+      }
+
+      public static EntityEquipmentPredicate.Builder equipment() {
+         return new EntityEquipmentPredicate.Builder();
+      }
+
+      public EntityEquipmentPredicate.Builder head(ItemPredicate var1) {
+         this.head = var1;
+         return this;
+      }
+
+      public EntityEquipmentPredicate.Builder chest(ItemPredicate var1) {
+         this.chest = var1;
+         return this;
+      }
+
+      public EntityEquipmentPredicate.Builder legs(ItemPredicate var1) {
+         this.legs = var1;
+         return this;
+      }
+
+      public EntityEquipmentPredicate.Builder feet(ItemPredicate var1) {
+         this.feet = var1;
+         return this;
+      }
+
+      public EntityEquipmentPredicate build() {
+         return new EntityEquipmentPredicate(this.head, this.chest, this.legs, this.feet, this.mainhand, this.offhand);
+      }
+   }
 }
