--- net/minecraft/advancements/critereon/MinMaxBounds.java
+++ net/minecraft/advancements/critereon/MinMaxBounds.java
@@ -14,12 +14,12 @@
 import javax.annotation.Nullable;
 import net.minecraft.network.chat.TranslatableComponent;
 import net.minecraft.util.GsonHelper;
 
 public abstract class MinMaxBounds<T extends Number> {
-   public static final SimpleCommandExceptionType ERROR_EMPTY = new SimpleCommandExceptionType(new TranslatableComponent("argument.range.empty", new Object[0]));
-   public static final SimpleCommandExceptionType ERROR_SWAPPED = new SimpleCommandExceptionType(new TranslatableComponent("argument.range.swapped", new Object[0]));
+   public static final SimpleCommandExceptionType ERROR_EMPTY = new SimpleCommandExceptionType(new TranslatableComponent("argument.range.empty"));
+   public static final SimpleCommandExceptionType ERROR_SWAPPED = new SimpleCommandExceptionType(new TranslatableComponent("argument.range.swapped"));
    protected final T min;
    protected final T max;
 
    protected MinMaxBounds(@Nullable T var1, @Nullable T var2) {
       this.min = var1;
