--- net/minecraft/world/item/enchantment/VanishingCurseEnchantment.java
+++ net/minecraft/world/item/enchantment/VanishingCurseEnchantment.java
@@ -2,11 +2,11 @@
 
 import net.minecraft.world.entity.EquipmentSlot;
 
 public class VanishingCurseEnchantment extends Enchantment {
    public VanishingCurseEnchantment(Enchantment.Rarity var1, EquipmentSlot... var2) {
-      super(var1, EnchantmentCategory.ALL, var2);
+      super(var1, EnchantmentCategory.VANISHABLE, var2);
    }
 
    public int getMinCost(int var1) {
       return 25;
    }
