--- net/minecraft/client/particle/DustParticle.java
+++ net/minecraft/client/particle/DustParticle.java
@@ -1,15 +1,15 @@
 package net.minecraft.client.particle;
 
+import net.minecraft.client.multiplayer.ClientLevel;
 import net.minecraft.core.particles.DustParticleOptions;
 import net.minecraft.util.Mth;
-import net.minecraft.world.level.Level;
 
 public class DustParticle extends TextureSheetParticle {
    private final SpriteSet sprites;
 
-   private DustParticle(Level var1, double var2, double var4, double var6, double var8, double var10, double var12, DustParticleOptions var14, SpriteSet var15) {
+   private DustParticle(ClientLevel var1, double var2, double var4, double var6, double var8, double var10, double var12, DustParticleOptions var14, SpriteSet var15) {
       super(var1, var2, var4, var6, var8, var10, var12);
       this.sprites = var15;
       this.xd *= 0.10000000149011612D;
       this.yd *= 0.10000000149011612D;
       this.zd *= 0.10000000149011612D;
@@ -61,10 +61,10 @@
 
       public Provider(SpriteSet var1) {
          this.sprites = var1;
       }
 
-      public Particle createParticle(DustParticleOptions var1, Level var2, double var3, double var5, double var7, double var9, double var11, double var13) {
+      public Particle createParticle(DustParticleOptions var1, ClientLevel var2, double var3, double var5, double var7, double var9, double var11, double var13) {
          return new DustParticle(var2, var3, var5, var7, var9, var11, var13, var1, this.sprites);
       }
    }
 }
