--- net/minecraft/client/particle/SpitParticle.java
+++ net/minecraft/client/particle/SpitParticle.java
@@ -1,12 +1,12 @@
 package net.minecraft.client.particle;
 
+import net.minecraft.client.multiplayer.ClientLevel;
 import net.minecraft.core.particles.SimpleParticleType;
-import net.minecraft.world.level.Level;
 
 public class SpitParticle extends ExplodeParticle {
-   private SpitParticle(Level var1, double var2, double var4, double var6, double var8, double var10, double var12, SpriteSet var14) {
+   private SpitParticle(ClientLevel var1, double var2, double var4, double var6, double var8, double var10, double var12, SpriteSet var14) {
       super(var1, var2, var4, var6, var8, var10, var12, var14);
       this.gravity = 0.5F;
    }
 
    public void tick() {
@@ -19,10 +19,10 @@
 
       public Provider(SpriteSet var1) {
          this.sprites = var1;
       }
 
-      public Particle createParticle(SimpleParticleType var1, Level var2, double var3, double var5, double var7, double var9, double var11, double var13) {
+      public Particle createParticle(SimpleParticleType var1, ClientLevel var2, double var3, double var5, double var7, double var9, double var11, double var13) {
          return new SpitParticle(var2, var3, var5, var7, var9, var11, var13, this.sprites);
       }
    }
 }
