--- net/minecraft/client/particle/TextureSheetParticle.java
+++ net/minecraft/client/particle/TextureSheetParticle.java
@@ -1,18 +1,18 @@
 package net.minecraft.client.particle;
 
+import net.minecraft.client.multiplayer.ClientLevel;
 import net.minecraft.client.renderer.texture.TextureAtlasSprite;
-import net.minecraft.world.level.Level;
 
 public abstract class TextureSheetParticle extends SingleQuadParticle {
    protected TextureAtlasSprite sprite;
 
-   protected TextureSheetParticle(Level var1, double var2, double var4, double var6) {
+   protected TextureSheetParticle(ClientLevel var1, double var2, double var4, double var6) {
       super(var1, var2, var4, var6);
    }
 
-   protected TextureSheetParticle(Level var1, double var2, double var4, double var6, double var8, double var10, double var12) {
+   protected TextureSheetParticle(ClientLevel var1, double var2, double var4, double var6, double var8, double var10, double var12) {
       super(var1, var2, var4, var6, var8, var10, var12);
    }
 
    protected void setSprite(TextureAtlasSprite var1) {
       this.sprite = var1;
