--- net/minecraft/server/commands/DeOpCommands.java
+++ net/minecraft/server/commands/DeOpCommands.java
@@ -13,11 +13,11 @@
 import net.minecraft.commands.arguments.GameProfileArgument;
 import net.minecraft.network.chat.TranslatableComponent;
 import net.minecraft.server.players.PlayerList;
 
 public class DeOpCommands {
-   private static final SimpleCommandExceptionType ERROR_NOT_OP = new SimpleCommandExceptionType(new TranslatableComponent("commands.deop.failed", new Object[0]));
+   private static final SimpleCommandExceptionType ERROR_NOT_OP = new SimpleCommandExceptionType(new TranslatableComponent("commands.deop.failed"));
 
    public static void register(CommandDispatcher<CommandSourceStack> var0) {
       var0.register((LiteralArgumentBuilder)((LiteralArgumentBuilder)Commands.literal("deop").requires((var0x) -> {
          return var0x.hasPermission(3);
       })).then(Commands.argument("targets", GameProfileArgument.gameProfile()).suggests((var0x, var1) -> {
