--- net/minecraft/server/commands/StopSoundCommand.java
+++ net/minecraft/server/commands/StopSoundCommand.java
@@ -57,11 +57,11 @@
             var0.sendSuccess(new TranslatableComponent("commands.stopsound.success.source.any", new Object[]{var2.getName()}), true);
          }
       } else if (var3 != null) {
          var0.sendSuccess(new TranslatableComponent("commands.stopsound.success.sourceless.sound", new Object[]{var3}), true);
       } else {
-         var0.sendSuccess(new TranslatableComponent("commands.stopsound.success.sourceless.any", new Object[0]), true);
+         var0.sendSuccess(new TranslatableComponent("commands.stopsound.success.sourceless.any"), true);
       }
 
       return var1.size();
    }
 }
