--- net/minecraft/world/level/block/BlastFurnaceBlock.java
+++ net/minecraft/world/level/block/BlastFurnaceBlock.java
@@ -11,14 +11,15 @@
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.level.BlockGetter;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.entity.BlastFurnaceBlockEntity;
 import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 
 public class BlastFurnaceBlock extends AbstractFurnaceBlock {
-   protected BlastFurnaceBlock(Block.Properties var1) {
+   protected BlastFurnaceBlock(BlockBehaviour.Properties var1) {
       super(var1);
    }
 
    public BlockEntity newBlockEntity(BlockGetter var1) {
       return new BlastFurnaceBlockEntity();
