--- net/minecraft/world/level/block/CoralBlock.java
+++ net/minecraft/world/level/block/CoralBlock.java
@@ -4,20 +4,21 @@
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.tags.FluidTags;
-import net.minecraft.world.item.BlockPlaceContext;
+import net.minecraft.world.item.context.BlockPlaceContext;
 import net.minecraft.world.level.BlockGetter;
 import net.minecraft.world.level.LevelAccessor;
+import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.material.FluidState;
 
 public class CoralBlock extends Block {
    private final Block deadBlock;
 
-   public CoralBlock(Block var1, Block.Properties var2) {
+   public CoralBlock(Block var1, BlockBehaviour.Properties var2) {
       super(var2);
       this.deadBlock = var1;
    }
 
    public void tick(BlockState var1, ServerLevel var2, BlockPos var3, Random var4) {
