--- net/minecraft/world/level/block/GrassPathBlock.java
+++ net/minecraft/world/level/block/GrassPathBlock.java
@@ -2,23 +2,24 @@
 
 import java.util.Random;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
 import net.minecraft.server.level.ServerLevel;
-import net.minecraft.world.item.BlockPlaceContext;
+import net.minecraft.world.item.context.BlockPlaceContext;
 import net.minecraft.world.level.BlockGetter;
 import net.minecraft.world.level.LevelAccessor;
 import net.minecraft.world.level.LevelReader;
+import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.pathfinder.PathComputationType;
 import net.minecraft.world.phys.shapes.CollisionContext;
 import net.minecraft.world.phys.shapes.VoxelShape;
 
 public class GrassPathBlock extends Block {
    protected static final VoxelShape SHAPE;
 
-   protected GrassPathBlock(Block.Properties var1) {
+   protected GrassPathBlock(BlockBehaviour.Properties var1) {
       super(var1);
    }
 
    public boolean useShapeForLightOcclusion(BlockState var1) {
       return true;
@@ -51,13 +52,9 @@
 
    public boolean isPathfindable(BlockState var1, BlockGetter var2, BlockPos var3, PathComputationType var4) {
       return false;
    }
 
-   public boolean isViewBlocking(BlockState var1, BlockGetter var2, BlockPos var3) {
-      return true;
-   }
-
    static {
       SHAPE = FarmBlock.SHAPE;
    }
 }
