--- net/minecraft/world/level/block/PoweredBlock.java
+++ net/minecraft/world/level/block/PoweredBlock.java
@@ -1,14 +1,15 @@
 package net.minecraft.world.level.block;
 
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
 import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 
 public class PoweredBlock extends Block {
-   public PoweredBlock(Block.Properties var1) {
+   public PoweredBlock(BlockBehaviour.Properties var1) {
       super(var1);
    }
 
    public boolean isSignalSource(BlockState var1) {
       return true;
