--- net/minecraft/world/level/block/RailBlock.java
+++ net/minecraft/world/level/block/RailBlock.java
@@ -1,20 +1,21 @@
 package net.minecraft.world.level.block;
 
 import net.minecraft.core.BlockPos;
 import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.block.state.StateDefinition;
 import net.minecraft.world.level.block.state.properties.BlockStateProperties;
 import net.minecraft.world.level.block.state.properties.EnumProperty;
 import net.minecraft.world.level.block.state.properties.Property;
 import net.minecraft.world.level.block.state.properties.RailShape;
 
 public class RailBlock extends BaseRailBlock {
    public static final EnumProperty<RailShape> SHAPE;
 
-   protected RailBlock(Block.Properties var1) {
+   protected RailBlock(BlockBehaviour.Properties var1) {
       super(false, var1);
       this.registerDefaultState((BlockState)((BlockState)this.stateDefinition.any()).setValue(SHAPE, RailShape.NORTH_SOUTH));
    }
 
    protected void updateState(BlockState var1, Level var2, BlockPos var3, Block var4) {
