--- net/minecraft/world/level/block/Seagrass.java
+++ net/minecraft/world/level/block/Seagrass.java
@@ -4,14 +4,15 @@
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.tags.FluidTags;
-import net.minecraft.world.item.BlockPlaceContext;
+import net.minecraft.world.item.context.BlockPlaceContext;
 import net.minecraft.world.level.BlockGetter;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.LevelAccessor;
+import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.block.state.properties.DoubleBlockHalf;
 import net.minecraft.world.level.material.Fluid;
 import net.minecraft.world.level.material.FluidState;
 import net.minecraft.world.level.material.Fluids;
@@ -19,20 +20,20 @@
 import net.minecraft.world.phys.shapes.VoxelShape;
 
 public class Seagrass extends BushBlock implements BonemealableBlock, LiquidBlockContainer {
    protected static final VoxelShape SHAPE = Block.box(2.0D, 0.0D, 2.0D, 14.0D, 12.0D, 14.0D);
 
-   protected Seagrass(Block.Properties var1) {
+   protected Seagrass(BlockBehaviour.Properties var1) {
       super(var1);
    }
 
    public VoxelShape getShape(BlockState var1, BlockGetter var2, BlockPos var3, CollisionContext var4) {
       return SHAPE;
    }
 
    protected boolean mayPlaceOn(BlockState var1, BlockGetter var2, BlockPos var3) {
-      return var1.isFaceSturdy(var2, var3, Direction.UP) && var1.getBlock() != Blocks.MAGMA_BLOCK;
+      return var1.isFaceSturdy(var2, var3, Direction.UP) && !var1.is(Blocks.MAGMA_BLOCK);
    }
 
    @Nullable
    public BlockState getStateForPlacement(BlockPlaceContext var1) {
       FluidState var2 = var1.getLevel().getFluidState(var1.getClickedPos());
@@ -62,11 +63,11 @@
 
    public void performBonemeal(ServerLevel var1, Random var2, BlockPos var3, BlockState var4) {
       BlockState var5 = Blocks.TALL_SEAGRASS.defaultBlockState();
       BlockState var6 = (BlockState)var5.setValue(TallSeagrass.HALF, DoubleBlockHalf.UPPER);
       BlockPos var7 = var3.above();
-      if (var1.getBlockState(var7).getBlock() == Blocks.WATER) {
+      if (var1.getBlockState(var7).is(Blocks.WATER)) {
          var1.setBlock(var3, var5, 2);
          var1.setBlock(var7, var6, 2);
       }
 
    }
