--- net/minecraft/world/level/block/SpawnerBlock.java
+++ net/minecraft/world/level/block/SpawnerBlock.java
@@ -4,14 +4,15 @@
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.BlockGetter;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.entity.BlockEntity;
 import net.minecraft.world.level.block.entity.SpawnerBlockEntity;
+import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 
 public class SpawnerBlock extends BaseEntityBlock {
-   protected SpawnerBlock(Block.Properties var1) {
+   protected SpawnerBlock(BlockBehaviour.Properties var1) {
       super(var1);
    }
 
    public BlockEntity newBlockEntity(BlockGetter var1) {
       return new SpawnerBlockEntity();
