--- net/minecraft/world/level/block/TrappedChestBlock.java
+++ net/minecraft/world/level/block/TrappedChestBlock.java
@@ -9,14 +9,15 @@
 import net.minecraft.world.level.BlockGetter;
 import net.minecraft.world.level.block.entity.BlockEntity;
 import net.minecraft.world.level.block.entity.BlockEntityType;
 import net.minecraft.world.level.block.entity.ChestBlockEntity;
 import net.minecraft.world.level.block.entity.TrappedChestBlockEntity;
+import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 
 public class TrappedChestBlock extends ChestBlock {
-   public TrappedChestBlock(Block.Properties var1) {
+   public TrappedChestBlock(BlockBehaviour.Properties var1) {
       super(var1, () -> {
          return BlockEntityType.TRAPPED_CHEST;
       });
    }
 
