--- net/minecraft/world/level/block/WeightedPressurePlateBlock.java
+++ net/minecraft/world/level/block/WeightedPressurePlateBlock.java
@@ -6,21 +6,21 @@
 import net.minecraft.util.Mth;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.LevelAccessor;
-import net.minecraft.world.level.LevelReader;
+import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.block.state.StateDefinition;
 import net.minecraft.world.level.block.state.properties.BlockStateProperties;
 import net.minecraft.world.level.block.state.properties.IntegerProperty;
 
 public class WeightedPressurePlateBlock extends BasePressurePlateBlock {
    public static final IntegerProperty POWER;
    private final int maxWeight;
 
-   protected WeightedPressurePlateBlock(int var1, Block.Properties var2) {
+   protected WeightedPressurePlateBlock(int var1, BlockBehaviour.Properties var2) {
       super(var2);
       this.registerDefaultState((BlockState)((BlockState)this.stateDefinition.any()).setValue(POWER, 0));
       this.maxWeight = var1;
    }
 
@@ -48,11 +48,11 @@
 
    protected BlockState setSignalForState(BlockState var1, int var2) {
       return (BlockState)var1.setValue(POWER, var2);
    }
 
-   public int getTickDelay(LevelReader var1) {
+   protected int getPressedTime() {
       return 10;
    }
 
    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> var1) {
       var1.add(POWER);
