--- net/minecraft/world/level/storage/loot/parameters/LootContextParamSets.java
+++ net/minecraft/world/level/storage/loot/parameters/LootContextParamSets.java
@@ -18,21 +18,27 @@
    });
    public static final LootContextParamSet SELECTOR = register("selector", (var0) -> {
       var0.required(LootContextParams.BLOCK_POS).required(LootContextParams.THIS_ENTITY);
    });
    public static final LootContextParamSet FISHING = register("fishing", (var0) -> {
-      var0.required(LootContextParams.BLOCK_POS).required(LootContextParams.TOOL);
+      var0.required(LootContextParams.BLOCK_POS).required(LootContextParams.TOOL).optional(LootContextParams.THIS_ENTITY);
    });
    public static final LootContextParamSet ENTITY = register("entity", (var0) -> {
       var0.required(LootContextParams.THIS_ENTITY).required(LootContextParams.BLOCK_POS).required(LootContextParams.DAMAGE_SOURCE).optional(LootContextParams.KILLER_ENTITY).optional(LootContextParams.DIRECT_KILLER_ENTITY).optional(LootContextParams.LAST_DAMAGE_PLAYER);
    });
    public static final LootContextParamSet GIFT = register("gift", (var0) -> {
       var0.required(LootContextParams.BLOCK_POS).required(LootContextParams.THIS_ENTITY);
    });
+   public static final LootContextParamSet PIGLIN_BARTER = register("barter", (var0) -> {
+      var0.required(LootContextParams.THIS_ENTITY);
+   });
    public static final LootContextParamSet ADVANCEMENT_REWARD = register("advancement_reward", (var0) -> {
       var0.required(LootContextParams.THIS_ENTITY).required(LootContextParams.BLOCK_POS);
    });
+   public static final LootContextParamSet ADVANCEMENT_ENTITY = register("advancement_entity", (var0) -> {
+      var0.required(LootContextParams.THIS_ENTITY).required(LootContextParams.ORIGIN).required(LootContextParams.BLOCK_POS);
+   });
    public static final LootContextParamSet ALL_PARAMS = register("generic", (var0) -> {
       var0.required(LootContextParams.THIS_ENTITY).required(LootContextParams.LAST_DAMAGE_PLAYER).required(LootContextParams.DAMAGE_SOURCE).required(LootContextParams.KILLER_ENTITY).required(LootContextParams.DIRECT_KILLER_ENTITY).required(LootContextParams.BLOCK_POS).required(LootContextParams.BLOCK_STATE).required(LootContextParams.BLOCK_ENTITY).required(LootContextParams.TOOL).required(LootContextParams.EXPLOSION_RADIUS);
    });
    public static final LootContextParamSet BLOCK = register("block", (var0) -> {
       var0.required(LootContextParams.BLOCK_STATE).required(LootContextParams.BLOCK_POS).required(LootContextParams.TOOL).optional(LootContextParams.THIS_ENTITY).optional(LootContextParams.BLOCK_ENTITY).optional(LootContextParams.EXPLOSION_RADIUS);
