--- net/minecraft/world/level/storage/loot/entries/LootItem.java
+++ net/minecraft/world/level/storage/loot/entries/LootItem.java
@@ -20,10 +20,14 @@
    private LootItem(Item var1, int var2, int var3, LootItemCondition[] var4, LootItemFunction[] var5) {
       super(var2, var3, var4, var5);
       this.item = var1;
    }
 
+   public LootPoolEntryType getType() {
+      return LootPoolEntries.ITEM;
+   }
+
    public void createItemStack(Consumer<ItemStack> var1, LootContext var2) {
       var1.accept(new ItemStack(this.item));
    }
 
    public static LootPoolSingletonContainer.Builder<?> lootTableItem(ItemLike var0) {
@@ -32,15 +36,14 @@
       });
    }
 
    public static class Serializer extends LootPoolSingletonContainer.Serializer<LootItem> {
       public Serializer() {
-         super(new ResourceLocation("item"), LootItem.class);
       }
 
-      public void serialize(JsonObject var1, LootItem var2, JsonSerializationContext var3) {
-         super.serialize(var1, (LootPoolSingletonContainer)var2, var3);
+      public void serializeCustom(JsonObject var1, LootItem var2, JsonSerializationContext var3) {
+         super.serializeCustom(var1, (LootPoolSingletonContainer)var2, var3);
          ResourceLocation var4 = Registry.ITEM.getKey(var2.item);
          if (var4 == null) {
             throw new IllegalArgumentException("Can't serialize unknown item " + var2.item);
          } else {
             var1.addProperty("name", var4.toString());
