--- net/minecraft/server/network/LegacyQueryHandler.java
+++ net/minecraft/server/network/LegacyQueryHandler.java
@@ -23,15 +23,11 @@
       ByteBuf var3 = (ByteBuf)var2;
       var3.markReaderIndex();
       boolean var4 = true;
 
       try {
-         try {
-            if (var3.readUnsignedByte() != 254) {
-               return;
-            }
-
+         if (var3.readUnsignedByte() == 254) {
             InetSocketAddress var5 = (InetSocketAddress)var1.channel().remoteAddress();
             MinecraftServer var6 = this.serverConnectionListener.getServer();
             int var7 = var3.readableBytes();
             String var8;
             switch(var7) {
@@ -73,21 +69,23 @@
                }
             }
 
             var3.release();
             var4 = false;
-         } catch (RuntimeException var21) {
+            return;
          }
-
+      } catch (RuntimeException var21) {
+         return;
       } finally {
          if (var4) {
             var3.resetReaderIndex();
             var1.channel().pipeline().remove("legacy_query");
             var1.fireChannelRead(var2);
          }
 
       }
+
    }
 
    private void sendFlushAndClose(ChannelHandlerContext var1, ByteBuf var2) {
       var1.pipeline().firstContext().writeAndFlush(var2).addListener(ChannelFutureListener.CLOSE);
    }
