--- net/minecraft/server/network/ServerStatusPacketListenerImpl.java
+++ net/minecraft/server/network/ServerStatusPacketListenerImpl.java
@@ -9,11 +9,11 @@
 import net.minecraft.network.protocol.status.ServerboundPingRequestPacket;
 import net.minecraft.network.protocol.status.ServerboundStatusRequestPacket;
 import net.minecraft.server.MinecraftServer;
 
 public class ServerStatusPacketListenerImpl implements ServerStatusPacketListener {
-   private static final Component DISCONNECT_REASON = new TranslatableComponent("multiplayer.status.request_handled", new Object[0]);
+   private static final Component DISCONNECT_REASON = new TranslatableComponent("multiplayer.status.request_handled");
    private final MinecraftServer server;
    private final Connection connection;
    private boolean hasRequestedStatus;
 
    public ServerStatusPacketListenerImpl(MinecraftServer var1, Connection var2) {
