--- net/minecraft/world/level/block/entity/RandomizableContainerBlockEntity.java
+++ net/minecraft/world/level/block/entity/RandomizableContainerBlockEntity.java
@@ -1,14 +1,16 @@
 package net.minecraft.world.level.block.entity;
 
 import java.util.Random;
 import javax.annotation.Nullable;
+import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.NonNullList;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.level.ServerLevel;
+import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.world.ContainerHelper;
 import net.minecraft.world.entity.player.Inventory;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.inventory.AbstractContainerMenu;
 import net.minecraft.world.item.ItemStack;
@@ -59,10 +61,14 @@
    }
 
    public void unpackLootTable(@Nullable Player var1) {
       if (this.lootTable != null && this.level.getServer() != null) {
          LootTable var2 = this.level.getServer().getLootTables().get(this.lootTable);
+         if (var1 instanceof ServerPlayer) {
+            CriteriaTriggers.GENERATE_LOOT.trigger((ServerPlayer)var1, this.lootTable);
+         }
+
          this.lootTable = null;
          LootContext.Builder var3 = (new LootContext.Builder((ServerLevel)this.level)).withParameter(LootContextParams.BLOCK_POS, new BlockPos(this.worldPosition)).withOptionalRandomSeed(this.lootTableSeed);
          if (var1 != null) {
             var3.withLuck(var1.getLuck()).withParameter(LootContextParams.THIS_ENTITY, var1);
          }
