--- net/minecraft/world/level/block/entity/SmokerBlockEntity.java
+++ net/minecraft/world/level/block/entity/SmokerBlockEntity.java
@@ -12,11 +12,11 @@
    public SmokerBlockEntity() {
       super(BlockEntityType.SMOKER, RecipeType.SMOKING);
    }
 
    protected Component getDefaultName() {
-      return new TranslatableComponent("container.smoker", new Object[0]);
+      return new TranslatableComponent("container.smoker");
    }
 
    protected int getBurnDuration(ItemStack var1) {
       return super.getBurnDuration(var1) / 2;
    }
