--- net/minecraft/advancements/critereon/SlideDownBlockTrigger.java
+++ net/minecraft/advancements/critereon/SlideDownBlockTrigger.java
@@ -1,9 +1,7 @@
 package net.minecraft.advancements.critereon;
 
-import com.google.gson.JsonDeserializationContext;
-import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonSyntaxException;
 import javax.annotation.Nullable;
 import net.minecraft.core.Registry;
 import net.minecraft.resources.ResourceLocation;
@@ -20,20 +18,20 @@
 
    public ResourceLocation getId() {
       return ID;
    }
 
-   public SlideDownBlockTrigger.TriggerInstance createInstance(JsonObject var1, JsonDeserializationContext var2) {
-      Block var3 = deserializeBlock(var1);
-      StatePropertiesPredicate var4 = StatePropertiesPredicate.fromJson(var1.get("state"));
-      if (var3 != null) {
-         var4.checkState(var3.getStateDefinition(), (var1x) -> {
-            throw new JsonSyntaxException("Block " + var3 + " has no property " + var1x);
+   public SlideDownBlockTrigger.TriggerInstance createInstance(JsonObject var1, EntityPredicate.Composite var2, DeserializationContext var3) {
+      Block var4 = deserializeBlock(var1);
+      StatePropertiesPredicate var5 = StatePropertiesPredicate.fromJson(var1.get("state"));
+      if (var4 != null) {
+         var5.checkState(var4.getStateDefinition(), (var1x) -> {
+            throw new JsonSyntaxException("Block " + var4 + " has no property " + var1x);
          });
       }
 
-      return new SlideDownBlockTrigger.TriggerInstance(var3, var4);
+      return new SlideDownBlockTrigger.TriggerInstance(var2, var4, var5);
    }
 
    @Nullable
    private static Block deserializeBlock(JsonObject var0) {
       if (var0.has("block")) {
@@ -45,41 +43,41 @@
          return null;
       }
    }
 
    public void trigger(ServerPlayer var1, BlockState var2) {
-      this.trigger(var1.getAdvancements(), (var1x) -> {
+      this.trigger(var1, (var1x) -> {
          return var1x.matches(var2);
       });
    }
 
    public static class TriggerInstance extends AbstractCriterionTriggerInstance {
       private final Block block;
       private final StatePropertiesPredicate state;
 
-      public TriggerInstance(@Nullable Block var1, StatePropertiesPredicate var2) {
-         super(SlideDownBlockTrigger.ID);
-         this.block = var1;
-         this.state = var2;
+      public TriggerInstance(EntityPredicate.Composite var1, @Nullable Block var2, StatePropertiesPredicate var3) {
+         super(SlideDownBlockTrigger.ID, var1);
+         this.block = var2;
+         this.state = var3;
       }
 
       public static SlideDownBlockTrigger.TriggerInstance slidesDownBlock(Block var0) {
-         return new SlideDownBlockTrigger.TriggerInstance(var0, StatePropertiesPredicate.ANY);
+         return new SlideDownBlockTrigger.TriggerInstance(EntityPredicate.Composite.ANY, var0, StatePropertiesPredicate.ANY);
       }
 
-      public JsonElement serializeToJson() {
-         JsonObject var1 = new JsonObject();
+      public JsonObject serializeToJson(SerializationContext var1) {
+         JsonObject var2 = super.serializeToJson(var1);
          if (this.block != null) {
-            var1.addProperty("block", Registry.BLOCK.getKey(this.block).toString());
+            var2.addProperty("block", Registry.BLOCK.getKey(this.block).toString());
          }
 
-         var1.add("state", this.state.serializeToJson());
-         return var1;
+         var2.add("state", this.state.serializeToJson());
+         return var2;
       }
 
       public boolean matches(BlockState var1) {
-         if (this.block != null && var1.getBlock() != this.block) {
+         if (this.block != null && !var1.is(this.block)) {
             return false;
          } else {
             return this.state.matches(var1);
          }
       }
