--- net/minecraft/commands/arguments/coordinates/ColumnPosArgument.java
+++ net/minecraft/commands/arguments/coordinates/ColumnPosArgument.java
@@ -18,11 +18,11 @@
 import net.minecraft.network.chat.TranslatableComponent;
 import net.minecraft.server.level.ColumnPos;
 
 public class ColumnPosArgument implements ArgumentType<Coordinates> {
    private static final Collection<String> EXAMPLES = Arrays.asList("0 0", "~ ~", "~1 ~-2", "^ ^", "^-1 ^0");
-   public static final SimpleCommandExceptionType ERROR_NOT_COMPLETE = new SimpleCommandExceptionType(new TranslatableComponent("argument.pos2d.incomplete", new Object[0]));
+   public static final SimpleCommandExceptionType ERROR_NOT_COMPLETE = new SimpleCommandExceptionType(new TranslatableComponent("argument.pos2d.incomplete"));
 
    public ColumnPosArgument() {
    }
 
    public static ColumnPosArgument columnPos() {
