--- net/minecraft/commands/arguments/coordinates/RotationArgument.java
+++ net/minecraft/commands/arguments/coordinates/RotationArgument.java
@@ -10,11 +10,11 @@
 import net.minecraft.commands.CommandSourceStack;
 import net.minecraft.network.chat.TranslatableComponent;
 
 public class RotationArgument implements ArgumentType<Coordinates> {
    private static final Collection<String> EXAMPLES = Arrays.asList("0 0", "~ ~", "~-5 ~5");
-   public static final SimpleCommandExceptionType ERROR_NOT_COMPLETE = new SimpleCommandExceptionType(new TranslatableComponent("argument.rotation.incomplete", new Object[0]));
+   public static final SimpleCommandExceptionType ERROR_NOT_COMPLETE = new SimpleCommandExceptionType(new TranslatableComponent("argument.rotation.incomplete"));
 
    public RotationArgument() {
    }
 
    public static RotationArgument rotation() {
