--- net/minecraft/client/renderer/RenderBuffers.java
+++ net/minecraft/client/renderer/RenderBuffers.java
@@ -10,19 +10,23 @@
    private final ChunkBufferBuilderPack fixedBufferPack = new ChunkBufferBuilderPack();
    private final SortedMap<RenderType, BufferBuilder> fixedBuffers = (SortedMap)Util.make(new Object2ObjectLinkedOpenHashMap(), (var1) -> {
       var1.put(Sheets.solidBlockSheet(), this.fixedBufferPack.builder(RenderType.solid()));
       var1.put(Sheets.cutoutBlockSheet(), this.fixedBufferPack.builder(RenderType.cutout()));
       var1.put(Sheets.bannerSheet(), this.fixedBufferPack.builder(RenderType.cutoutMipped()));
-      var1.put(Sheets.translucentBlockSheet(), this.fixedBufferPack.builder(RenderType.translucent()));
+      var1.put(Sheets.translucentCullBlockSheet(), this.fixedBufferPack.builder(RenderType.translucent()));
       put(var1, Sheets.shieldSheet());
       put(var1, Sheets.bedSheet());
       put(var1, Sheets.shulkerBoxSheet());
       put(var1, Sheets.signSheet());
       put(var1, Sheets.chestSheet());
       put(var1, RenderType.translucentNoCrumbling());
+      put(var1, RenderType.armorGlint());
+      put(var1, RenderType.armorEntityGlint());
       put(var1, RenderType.glint());
+      put(var1, RenderType.glintDirect());
       put(var1, RenderType.entityGlint());
+      put(var1, RenderType.entityGlintDirect());
       put(var1, RenderType.waterMask());
       ModelBakery.DESTROY_TYPES.forEach((var1x) -> {
          put(var1, var1x);
       });
    });
