--- net/minecraft/realms/RealmsLabelProxy.java
+++ net/minecraft/realms/RealmsLabelProxy.java
@@ -1,15 +1,0 @@
-package net.minecraft.realms;
-
-import net.minecraft.client.gui.components.events.GuiEventListener;
-
-public class RealmsLabelProxy implements GuiEventListener {
-   private final RealmsLabel label;
-
-   public RealmsLabelProxy(RealmsLabel var1) {
-      this.label = var1;
-   }
-
-   public RealmsLabel getLabel() {
-      return this.label;
-   }
-}
