--- net/minecraft/realms/RealmsSharedConstants.java
+++ net/minecraft/realms/RealmsSharedConstants.java
@@ -1,15 +1,0 @@
-package net.minecraft.realms;
-
-import net.minecraft.SharedConstants;
-
-public class RealmsSharedConstants {
-   public static final int TICKS_PER_SECOND = 20;
-   public static final char[] ILLEGAL_FILE_CHARACTERS;
-
-   public RealmsSharedConstants() {
-   }
-
-   static {
-      ILLEGAL_FILE_CHARACTERS = SharedConstants.ILLEGAL_FILE_CHARACTERS;
-   }
-}
