--- net/minecraft/core/particles/BlockParticleOption.java
+++ net/minecraft/core/particles/BlockParticleOption.java
@@ -1,9 +1,10 @@
 package net.minecraft.core.particles;
 
 import com.mojang.brigadier.StringReader;
 import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.serialization.Codec;
 import net.minecraft.commands.arguments.blocks.BlockStateParser;
 import net.minecraft.core.Registry;
 import net.minecraft.network.FriendlyByteBuf;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.state.BlockState;
@@ -20,10 +21,18 @@
       }
    };
    private final ParticleType<BlockParticleOption> type;
    private final BlockState state;
 
+   public static Codec<BlockParticleOption> codec(ParticleType<BlockParticleOption> var0) {
+      return BlockState.CODEC.xmap((var1) -> {
+         return new BlockParticleOption(var0, var1);
+      }, (var0x) -> {
+         return var0x.state;
+      });
+   }
+
    public BlockParticleOption(ParticleType<BlockParticleOption> var1, BlockState var2) {
       this.type = var1;
       this.state = var2;
    }
 
