--- net/minecraft/world/level/material/WaterFluid.java
+++ net/minecraft/world/level/material/WaterFluid.java
@@ -42,11 +42,11 @@
       if (!var3.isSource() && !(Boolean)var3.getValue(FALLING)) {
          if (var4.nextInt(64) == 0) {
             var1.playLocalSound((double)var2.getX() + 0.5D, (double)var2.getY() + 0.5D, (double)var2.getZ() + 0.5D, SoundEvents.WATER_AMBIENT, SoundSource.BLOCKS, var4.nextFloat() * 0.25F + 0.75F, var4.nextFloat() + 0.5F, false);
          }
       } else if (var4.nextInt(10) == 0) {
-         var1.addParticle(ParticleTypes.UNDERWATER, (double)var2.getX() + (double)var4.nextFloat(), (double)var2.getY() + (double)var4.nextFloat(), (double)var2.getZ() + (double)var4.nextFloat(), 0.0D, 0.0D, 0.0D);
+         var1.addParticle(ParticleTypes.UNDERWATER, (double)var2.getX() + var4.nextDouble(), (double)var2.getY() + var4.nextDouble(), (double)var2.getZ() + var4.nextDouble(), 0.0D, 0.0D, 0.0D);
       }
 
    }
 
    @Nullable
