--- net/minecraft/world/level/chunk/ChunkSource.java
+++ net/minecraft/world/level/chunk/ChunkSource.java
@@ -1,9 +1,8 @@
 package net.minecraft.world.level.chunk;
 
 import java.io.IOException;
-import java.util.function.BooleanSupplier;
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.level.BlockGetter;
 import net.minecraft.world.level.ChunkPos;
@@ -33,12 +32,10 @@
    }
 
    @Nullable
    public abstract ChunkAccess getChunk(int var1, int var2, ChunkStatus var3, boolean var4);
 
-   public abstract void tick(BooleanSupplier var1);
-
    public abstract String gatherStats();
 
    public void close() throws IOException {
    }
 
