--- net/minecraft/world/entity/ai/behavior/SetEntityLookTarget.java
+++ net/minecraft/world/entity/ai/behavior/SetEntityLookTarget.java
@@ -25,10 +25,16 @@
       this((var1x) -> {
          return var1.equals(var1x.getType());
       }, var2);
    }
 
+   public SetEntityLookTarget(float var1) {
+      this((var0) -> {
+         return true;
+      }, var1);
+   }
+
    public SetEntityLookTarget(Predicate<LivingEntity> var1, float var2) {
       super(ImmutableMap.of(MemoryModuleType.LOOK_TARGET, MemoryStatus.VALUE_ABSENT, MemoryModuleType.VISIBLE_LIVING_ENTITIES, MemoryStatus.VALUE_PRESENT));
       this.predicate = var1;
       this.maxDistSqr = var2 * var2;
    }
@@ -41,10 +47,10 @@
       Brain var5 = var2.getBrain();
       var5.getMemory(MemoryModuleType.VISIBLE_LIVING_ENTITIES).ifPresent((var3x) -> {
          var3x.stream().filter(this.predicate).filter((var2x) -> {
             return var2x.distanceToSqr(var2) <= (double)this.maxDistSqr;
          }).findFirst().ifPresent((var1) -> {
-            var5.setMemory(MemoryModuleType.LOOK_TARGET, (Object)(new EntityPosWrapper(var1)));
+            var5.setMemory(MemoryModuleType.LOOK_TARGET, (Object)(new EntityTracker(var1, true)));
          });
       });
    }
 }
