--- net/minecraft/world/entity/ai/behavior/StopBeingAngryIfTargetDead.java
+++ net/minecraft/world/entity/ai/behavior/StopBeingAngryIfTargetDead.java
@@ -1,0 +1,27 @@
+package net.minecraft.world.entity.ai.behavior;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Optional;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.Mob;
+import net.minecraft.world.entity.ai.memory.MemoryModuleType;
+import net.minecraft.world.entity.ai.memory.MemoryStatus;
+
+public class StopBeingAngryIfTargetDead<E extends Mob> extends Behavior<E> {
+   public StopBeingAngryIfTargetDead() {
+      super(ImmutableMap.of(MemoryModuleType.ANGRY_AT, MemoryStatus.VALUE_PRESENT));
+   }
+
+   protected void start(ServerLevel var1, E var2, long var3) {
+      if (this.isCurrentTargetDeadOrRemoved(var2)) {
+         var2.getBrain().eraseMemory(MemoryModuleType.ANGRY_AT);
+      }
+
+   }
+
+   private boolean isCurrentTargetDeadOrRemoved(E var1) {
+      Optional var2 = BehaviorUtils.getLivingEntityFromUUIDMemory(var1, MemoryModuleType.ANGRY_AT);
+      return !var2.isPresent() || !((LivingEntity)var2.get()).isAlive();
+   }
+}
