--- net/minecraft/client/renderer/chunk/ChunkRenderDispatcher.java
+++ net/minecraft/client/renderer/chunk/ChunkRenderDispatcher.java
@@ -38,11 +38,10 @@
 import net.minecraft.client.renderer.block.ModelBlockRenderer;
 import net.minecraft.client.renderer.blockentity.BlockEntityRenderDispatcher;
 import net.minecraft.client.renderer.blockentity.BlockEntityRenderer;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
-import net.minecraft.core.Vec3i;
 import net.minecraft.util.thread.ProcessorMailbox;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.RenderShape;
 import net.minecraft.world.level.block.entity.BlockEntity;
@@ -321,11 +320,11 @@
             Direction[] var4 = Direction.values();
             int var5 = var4.length;
 
             for(int var6 = 0; var6 < var5; ++var6) {
                Direction var7 = var4[var6];
-               this.relativeOrigins[var7.ordinal()].set((Vec3i)this.origin).move(var7, 16);
+               this.relativeOrigins[var7.ordinal()].set(this.origin).move(var7, 16);
             }
 
          }
       }
 
