--- net/minecraft/client/renderer/chunk/VisGraph.java
+++ net/minecraft/client/renderer/chunk/VisGraph.java
@@ -67,14 +67,10 @@
       }
 
       return var1;
    }
 
-   public Set<Direction> floodFill(BlockPos var1) {
-      return this.floodFill(getIndex(var1));
-   }
-
    private Set<Direction> floodFill(int var1) {
       EnumSet var2 = EnumSet.noneOf(Direction.class);
       IntArrayFIFOQueue var3 = new IntArrayFIFOQueue();
       var3.enqueue(var1);
       this.bitSet.set(var1, true);
