--- net/minecraft/client/resources/metadata/language/LanguageMetadataSectionSerializer.java
+++ net/minecraft/client/resources/metadata/language/LanguageMetadataSectionSerializer.java
@@ -5,11 +5,11 @@
 import com.google.gson.JsonObject;
 import com.google.gson.JsonParseException;
 import java.util.HashSet;
 import java.util.Iterator;
 import java.util.Map.Entry;
-import net.minecraft.client.resources.language.Language;
+import net.minecraft.client.resources.language.LanguageInfo;
 import net.minecraft.server.packs.metadata.MetadataSectionSerializer;
 import net.minecraft.util.GsonHelper;
 
 public class LanguageMetadataSectionSerializer implements MetadataSectionSerializer<LanguageMetadataSection> {
    public LanguageMetadataSectionSerializer() {
@@ -43,11 +43,11 @@
          }
 
          if (var8.isEmpty()) {
             throw new JsonParseException("Invalid language->'" + var5 + "'->name: empty value");
          }
-      } while(var2.add(new Language(var5, var7, var8, var9)));
+      } while(var2.add(new LanguageInfo(var5, var7, var8, var9)));
 
       throw new JsonParseException("Duplicate language->'" + var5 + "' defined");
    }
 
    public String getMetadataSectionName() {
