--- net/minecraft/client/resources/metadata/language/LanguageMetadataSection.java
+++ net/minecraft/client/resources/metadata/language/LanguageMetadataSection.java
@@ -1,17 +1,17 @@
 package net.minecraft.client.resources.metadata.language;
 
 import java.util.Collection;
-import net.minecraft.client.resources.language.Language;
+import net.minecraft.client.resources.language.LanguageInfo;
 
 public class LanguageMetadataSection {
    public static final LanguageMetadataSectionSerializer SERIALIZER = new LanguageMetadataSectionSerializer();
-   private final Collection<Language> languages;
+   private final Collection<LanguageInfo> languages;
 
-   public LanguageMetadataSection(Collection<Language> var1) {
+   public LanguageMetadataSection(Collection<LanguageInfo> var1) {
       this.languages = var1;
    }
 
-   public Collection<Language> getLanguages() {
+   public Collection<LanguageInfo> getLanguages() {
       return this.languages;
    }
 }
