--- net/minecraft/world/entity/animal/Chicken.java
+++ net/minecraft/world/entity/animal/Chicken.java
@@ -9,20 +9,22 @@
 import net.minecraft.world.entity.AgableMob;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.EntityDimensions;
 import net.minecraft.world.entity.EntityType;
 import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.Mob;
 import net.minecraft.world.entity.Pose;
+import net.minecraft.world.entity.ai.attributes.AttributeSupplier;
+import net.minecraft.world.entity.ai.attributes.Attributes;
 import net.minecraft.world.entity.ai.goal.BreedGoal;
 import net.minecraft.world.entity.ai.goal.FloatGoal;
 import net.minecraft.world.entity.ai.goal.FollowParentGoal;
 import net.minecraft.world.entity.ai.goal.LookAtPlayerGoal;
 import net.minecraft.world.entity.ai.goal.PanicGoal;
 import net.minecraft.world.entity.ai.goal.RandomLookAroundGoal;
 import net.minecraft.world.entity.ai.goal.TemptGoal;
 import net.minecraft.world.entity.ai.goal.WaterAvoidingRandomStrollGoal;
-import net.minecraft.world.entity.monster.SharedMonsterAttributes;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
 import net.minecraft.world.item.crafting.Ingredient;
 import net.minecraft.world.level.Level;
@@ -59,14 +61,12 @@
 
    protected float getStandingEyeHeight(Pose var1, EntityDimensions var2) {
       return this.isBaby() ? var2.height * 0.85F : var2.height * 0.92F;
    }
 
-   protected void registerAttributes() {
-      super.registerAttributes();
-      this.getAttribute(SharedMonsterAttributes.MAX_HEALTH).setBaseValue(4.0D);
-      this.getAttribute(SharedMonsterAttributes.MOVEMENT_SPEED).setBaseValue(0.25D);
+   public static AttributeSupplier.Builder createAttributes() {
+      return Mob.createMobAttributes().add(Attributes.MAX_HEALTH, 4.0D).add(Attributes.MOVEMENT_SPEED, 0.25D);
    }
 
    public void aiStep() {
       super.aiStep();
       this.oFlap = this.flap;
@@ -138,20 +138,20 @@
       var1.putBoolean("IsChickenJockey", this.isChickenJockey);
       var1.putInt("EggLayTime", this.eggTime);
    }
 
    public boolean removeWhenFarAway(double var1) {
-      return this.isChickenJockey() && !this.isVehicle();
+      return this.isChickenJockey();
    }
 
    public void positionRider(Entity var1) {
       super.positionRider(var1);
       float var2 = Mth.sin(this.yBodyRot * 0.017453292F);
       float var3 = Mth.cos(this.yBodyRot * 0.017453292F);
       float var4 = 0.1F;
       float var5 = 0.0F;
-      var1.setPos(this.getX() + (double)(0.1F * var2), this.getY(0.5D) + var1.getRidingHeight() + 0.0D, this.getZ() - (double)(0.1F * var3));
+      var1.setPos(this.getX() + (double)(0.1F * var2), this.getY(0.5D) + var1.getMyRidingOffset() + 0.0D, this.getZ() - (double)(0.1F * var3));
       if (var1 instanceof LivingEntity) {
          ((LivingEntity)var1).yBodyRot = this.yBodyRot;
       }
 
    }
