--- net/minecraft/world/entity/animal/Pufferfish.java
+++ net/minecraft/world/entity/animal/Pufferfish.java
@@ -130,11 +130,14 @@
    }
 
    public void playerTouch(Player var1) {
       int var2 = this.getPuffState();
       if (var1 instanceof ServerPlayer && var2 > 0 && var1.hurt(DamageSource.mobAttack(this), (float)(1 + var2))) {
-         ((ServerPlayer)var1).connection.send(new ClientboundGameEventPacket(9, 0.0F));
+         if (!this.isSilent()) {
+            ((ServerPlayer)var1).connection.send(new ClientboundGameEventPacket(ClientboundGameEventPacket.PUFFER_FISH_STING, 0.0F));
+         }
+
          var1.addEffect(new MobEffectInstance(MobEffects.POISON, 60 * var2, 0));
       }
 
    }
 
