--- net/minecraft/world/entity/animal/TropicalFish.java
+++ net/minecraft/world/entity/animal/TropicalFish.java
@@ -1,9 +1,10 @@
 package net.minecraft.world.entity.animal;
 
 import java.util.Locale;
 import javax.annotation.Nullable;
+import net.minecraft.Util;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.network.syncher.EntityDataAccessor;
 import net.minecraft.network.syncher.EntityDataSerializers;
 import net.minecraft.network.syncher.SynchedEntityData;
 import net.minecraft.resources.ResourceLocation;
@@ -16,11 +17,11 @@
 import net.minecraft.world.entity.SpawnGroupData;
 import net.minecraft.world.item.DyeColor;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
 import net.minecraft.world.level.Level;
-import net.minecraft.world.level.LevelAccessor;
+import net.minecraft.world.level.ServerLevelAccessor;
 
 public class TropicalFish extends AbstractSchoolingFish {
    private static final EntityDataAccessor<Integer> DATA_ID_TYPE_VARIANT;
    private static final ResourceLocation[] BASE_TEXTURE_LOCATIONS;
    private static final ResourceLocation[] PATTERN_A_TEXTURE_LOCATIONS;
@@ -142,11 +143,11 @@
    public ResourceLocation getBaseTextureLocation() {
       return BASE_TEXTURE_LOCATIONS[getBaseVariant(this.getVariant())];
    }
 
    @Nullable
-   public SpawnGroupData finalizeSpawn(LevelAccessor var1, DifficultyInstance var2, MobSpawnType var3, @Nullable SpawnGroupData var4, @Nullable CompoundTag var5) {
+   public SpawnGroupData finalizeSpawn(ServerLevelAccessor var1, DifficultyInstance var2, MobSpawnType var3, @Nullable SpawnGroupData var4, @Nullable CompoundTag var5) {
       Object var11 = super.finalizeSpawn(var1, var2, var3, var4, var5);
       if (var5 != null && var5.contains("BucketVariantTag", 3)) {
          this.setVariant(var5.getInt("BucketVariantTag"));
          return (SpawnGroupData)var11;
       } else {
@@ -159,11 +160,11 @@
             var6 = var10.base;
             var7 = var10.pattern;
             var8 = var10.baseColor;
             var9 = var10.patternColor;
          } else if ((double)this.random.nextFloat() < 0.9D) {
-            int var12 = COMMON_VARIANTS[this.random.nextInt(COMMON_VARIANTS.length)];
+            int var12 = Util.getRandom(COMMON_VARIANTS, this.random);
             var6 = var12 & 255;
             var7 = (var12 & '\uff00') >> 8;
             var8 = (var12 & 16711680) >> 16;
             var9 = (var12 & -16777216) >> 24;
             var11 = new TropicalFish.TropicalFishGroupData(this, var6, var7, var8, var9);
