--- net/minecraft/world/entity/ai/sensing/InteractableDoorsSensor.java
+++ net/minecraft/world/entity/ai/sensing/InteractableDoorsSensor.java
@@ -4,24 +4,24 @@
 import com.google.common.collect.Lists;
 import java.util.ArrayList;
 import java.util.Set;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.GlobalPos;
+import net.minecraft.resources.ResourceKey;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.tags.BlockTags;
 import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.ai.Brain;
 import net.minecraft.world.entity.ai.memory.MemoryModuleType;
-import net.minecraft.world.level.dimension.DimensionType;
 
 public class InteractableDoorsSensor extends Sensor<LivingEntity> {
    public InteractableDoorsSensor() {
    }
 
    protected void doTick(ServerLevel var1, LivingEntity var2) {
-      DimensionType var3 = var1.getDimension().getType();
-      BlockPos var4 = new BlockPos(var2);
+      ResourceKey var3 = var1.dimension();
+      BlockPos var4 = var2.blockPosition();
       ArrayList var5 = Lists.newArrayList();
 
       for(int var6 = -1; var6 <= 1; ++var6) {
          for(int var7 = -1; var7 <= 1; ++var7) {
             for(int var8 = -1; var8 <= 1; ++var8) {
