--- net/minecraft/commands/arguments/coordinates/SwizzleArgument.java
+++ net/minecraft/commands/arguments/coordinates/SwizzleArgument.java
@@ -12,11 +12,11 @@
 import net.minecraft.core.Direction;
 import net.minecraft.network.chat.TranslatableComponent;
 
 public class SwizzleArgument implements ArgumentType<EnumSet<Direction.Axis>> {
    private static final Collection<String> EXAMPLES = Arrays.asList("xyz", "x");
-   private static final SimpleCommandExceptionType ERROR_INVALID = new SimpleCommandExceptionType(new TranslatableComponent("arguments.swizzle.invalid", new Object[0]));
+   private static final SimpleCommandExceptionType ERROR_INVALID = new SimpleCommandExceptionType(new TranslatableComponent("arguments.swizzle.invalid"));
 
    public SwizzleArgument() {
    }
 
    public static SwizzleArgument swizzle() {
