--- net/minecraft/util/ExceptionCollector.java
+++ net/minecraft/util/ExceptionCollector.java
@@ -1,0 +1,26 @@
+package net.minecraft.util;
+
+import javax.annotation.Nullable;
+
+public class ExceptionCollector<T extends Throwable> {
+   @Nullable
+   private T result;
+
+   public ExceptionCollector() {
+   }
+
+   public void add(T var1) {
+      if (this.result == null) {
+         this.result = var1;
+      } else {
+         this.result.addSuppressed(var1);
+      }
+
+   }
+
+   public void throwIfPresent() throws T {
+      if (this.result != null) {
+         throw this.result;
+      }
+   }
+}
