--- net/minecraft/world/entity/boss/enderdragon/phases/DragonHoldingPatternPhase.java
+++ net/minecraft/world/entity/boss/enderdragon/phases/DragonHoldingPatternPhase.java
@@ -1,9 +1,10 @@
 package net.minecraft.world.entity.boss.enderdragon.phases;
 
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
+import net.minecraft.core.Vec3i;
 import net.minecraft.util.Mth;
 import net.minecraft.world.damagesource.DamageSource;
 import net.minecraft.world.entity.ai.targeting.TargetingConditions;
 import net.minecraft.world.entity.boss.enderdragon.EndCrystal;
 import net.minecraft.world.entity.boss.enderdragon.EnderDragon;
@@ -107,19 +108,19 @@
       ((DragonStrafePlayerPhase)this.dragon.getPhaseManager().getPhase(EnderDragonPhase.STRAFE_PLAYER)).setTarget(var1);
    }
 
    private void navigateToNextPathNode() {
       if (this.currentPath != null && !this.currentPath.isDone()) {
-         Vec3 var1 = this.currentPath.currentPos();
+         Vec3i var1 = this.currentPath.currentPos();
          this.currentPath.next();
-         double var2 = var1.x;
-         double var4 = var1.z;
+         double var2 = (double)var1.getX();
+         double var4 = (double)var1.getZ();
 
          double var6;
          do {
-            var6 = var1.y + (double)(this.dragon.getRandom().nextFloat() * 20.0F);
-         } while(var6 < var1.y);
+            var6 = (double)((float)var1.getY() + this.dragon.getRandom().nextFloat() * 20.0F);
+         } while(var6 < (double)var1.getY());
 
          this.targetLocation = new Vec3(var2, var6, var4);
       }
 
    }
