--- net/minecraft/world/item/BlockItem.java
+++ net/minecraft/world/item/BlockItem.java
@@ -11,10 +11,12 @@
 import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.sounds.SoundEvent;
 import net.minecraft.sounds.SoundSource;
 import net.minecraft.world.InteractionResult;
 import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.context.BlockPlaceContext;
+import net.minecraft.world.item.context.UseOnContext;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.SoundType;
 import net.minecraft.world.level.block.entity.BlockEntity;
 import net.minecraft.world.level.block.state.BlockState;
@@ -31,11 +33,11 @@
       this.block = var1;
    }
 
    public InteractionResult useOn(UseOnContext var1) {
       InteractionResult var2 = this.place(new BlockPlaceContext(var1));
-      return var2 != InteractionResult.SUCCESS && this.isEdible() ? this.use(var1.level, var1.player, var1.hand).getResult() : var2;
+      return !var2.consumesAction() && this.isEdible() ? this.use(var1.getLevel(), var1.getPlayer(), var1.getHand()).getResult() : var2;
    }
 
    public InteractionResult place(BlockPlaceContext var1) {
       if (!var1.canPlace()) {
          return InteractionResult.FAIL;
@@ -65,12 +67,15 @@
                   }
                }
 
                SoundType var10 = var8.getSoundType();
                var5.playSound(var6, var4, this.getPlaceSound(var8), SoundSource.BLOCKS, (var10.getVolume() + 1.0F) / 2.0F, var10.getPitch() * 0.8F);
-               var7.shrink(1);
-               return InteractionResult.SUCCESS;
+               if (var6 == null || !var6.abilities.instabuild) {
+                  var7.shrink(1);
+               }
+
+               return InteractionResult.sidedSuccess(var5.isClientSide);
             }
          }
       }
    }
 
@@ -156,11 +161,11 @@
                var7.merge(var5);
                var7.putInt("x", var2.getX());
                var7.putInt("y", var2.getY());
                var7.putInt("z", var2.getZ());
                if (!var7.equals(var8)) {
-                  var6.load(var7);
+                  var6.load(var0.getBlockState(var2), var7);
                   var6.setChanged();
                   return true;
                }
             }
          }
