--- net/minecraft/world/item/KnowledgeBookItem.java
+++ net/minecraft/world/item/KnowledgeBookItem.java
@@ -48,11 +48,11 @@
 
             var2.awardRecipes(var7);
             var2.awardStat(Stats.ITEM_USED.get(this));
          }
 
-         return InteractionResultHolder.success(var4);
+         return InteractionResultHolder.sidedSuccess(var4, var1.isClientSide());
       } else {
          LOGGER.error((String)"Tag not valid: {}", (Object)var5);
          return InteractionResultHolder.fail(var4);
       }
    }
