--- net/minecraft/world/item/RecordItem.java
+++ net/minecraft/world/item/RecordItem.java
@@ -5,10 +5,11 @@
 import net.minecraft.core.BlockPos;
 import net.minecraft.sounds.SoundEvent;
 import net.minecraft.stats.Stats;
 import net.minecraft.world.InteractionResult;
 import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.context.UseOnContext;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.JukeboxBlock;
 import net.minecraft.world.level.block.state.BlockState;
 
@@ -26,11 +27,11 @@
 
    public InteractionResult useOn(UseOnContext var1) {
       Level var2 = var1.getLevel();
       BlockPos var3 = var1.getClickedPos();
       BlockState var4 = var2.getBlockState(var3);
-      if (var4.getBlock() == Blocks.JUKEBOX && !(Boolean)var4.getValue(JukeboxBlock.HAS_RECORD)) {
+      if (var4.is(Blocks.JUKEBOX) && !(Boolean)var4.getValue(JukeboxBlock.HAS_RECORD)) {
          ItemStack var5 = var1.getItemInHand();
          if (!var2.isClientSide) {
             ((JukeboxBlock)Blocks.JUKEBOX).setRecord(var2, var3, var4, var5);
             var2.levelEvent((Player)null, 1010, var3, Item.getId(this));
             var5.shrink(1);
@@ -38,11 +39,11 @@
             if (var6 != null) {
                var6.awardStat(Stats.PLAY_RECORD);
             }
          }
 
-         return InteractionResult.SUCCESS;
+         return InteractionResult.sidedSuccess(var2.isClientSide);
       } else {
          return InteractionResult.PASS;
       }
    }
 
