--- net/minecraft/world/item/Tiers.java
+++ net/minecraft/world/item/Tiers.java
@@ -1,28 +1,31 @@
 package net.minecraft.world.item;
 
 import java.util.function.Supplier;
 import net.minecraft.tags.ItemTags;
+import net.minecraft.tags.Tag;
 import net.minecraft.util.LazyLoadedValue;
 import net.minecraft.world.item.crafting.Ingredient;
-import net.minecraft.world.level.block.Blocks;
 
 public enum Tiers implements Tier {
    WOOD(0, 59, 2.0F, 0.0F, 15, () -> {
-      return Ingredient.of(ItemTags.PLANKS);
+      return Ingredient.of((Tag)ItemTags.PLANKS);
    }),
    STONE(1, 131, 4.0F, 1.0F, 5, () -> {
-      return Ingredient.of(Blocks.COBBLESTONE);
+      return Ingredient.of((Tag)ItemTags.STONE_TOOL_MATERIALS);
    }),
    IRON(2, 250, 6.0F, 2.0F, 14, () -> {
       return Ingredient.of(Items.IRON_INGOT);
    }),
    DIAMOND(3, 1561, 8.0F, 3.0F, 10, () -> {
       return Ingredient.of(Items.DIAMOND);
    }),
    GOLD(0, 32, 12.0F, 0.0F, 22, () -> {
       return Ingredient.of(Items.GOLD_INGOT);
+   }),
+   NETHERITE(4, 2031, 9.0F, 4.0F, 15, () -> {
+      return Ingredient.of(Items.NETHERITE_INGOT);
    });
 
    private final int level;
    private final int uses;
    private final float speed;
