--- net/minecraft/network/protocol/game/ServerboundClientInformationPacket.java
+++ net/minecraft/network/protocol/game/ServerboundClientInformationPacket.java
@@ -37,14 +37,10 @@
 
    public void handle(ServerGamePacketListener var1) {
       var1.handleClientInformation(this);
    }
 
-   public String getLanguage() {
-      return this.language;
-   }
-
    public ChatVisiblity getChatVisibility() {
       return this.chatVisibility;
    }
 
    public boolean getChatColors() {
