--- net/minecraft/network/protocol/game/ServerboundInteractPacket.java
+++ net/minecraft/network/protocol/game/ServerboundInteractPacket.java
@@ -12,10 +12,11 @@
 public class ServerboundInteractPacket implements Packet<ServerGamePacketListener> {
    private int entityId;
    private ServerboundInteractPacket.Action action;
    private Vec3 location;
    private InteractionHand hand;
+   private boolean usingSecondaryAction;
 
    public ServerboundInteractPacket() {
    }
 
    public ServerboundInteractPacket(Entity var1) {
@@ -30,10 +31,11 @@
          this.location = new Vec3((double)var1.readFloat(), (double)var1.readFloat(), (double)var1.readFloat());
       }
 
       if (this.action == ServerboundInteractPacket.Action.INTERACT || this.action == ServerboundInteractPacket.Action.INTERACT_AT) {
          this.hand = (InteractionHand)var1.readEnum(InteractionHand.class);
+         this.usingSecondaryAction = var1.readBoolean();
       }
 
    }
 
    public void write(FriendlyByteBuf var1) throws IOException {
@@ -45,10 +47,11 @@
          var1.writeFloat((float)this.location.z);
       }
 
       if (this.action == ServerboundInteractPacket.Action.INTERACT || this.action == ServerboundInteractPacket.Action.INTERACT_AT) {
          var1.writeEnum(this.hand);
+         var1.writeBoolean(this.usingSecondaryAction);
       }
 
    }
 
    public void handle(ServerGamePacketListener var1) {
@@ -70,10 +73,14 @@
 
    public Vec3 getLocation() {
       return this.location;
    }
 
+   public boolean isUsingSecondaryAction() {
+      return this.usingSecondaryAction;
+   }
+
    public static enum Action {
       INTERACT,
       ATTACK,
       INTERACT_AT;
 
