--- net/minecraft/world/level/storage/loot/predicates/BonusLevelTableCondition.java
+++ net/minecraft/world/level/storage/loot/predicates/BonusLevelTableCondition.java
@@ -23,10 +23,14 @@
    private BonusLevelTableCondition(Enchantment var1, float[] var2) {
       this.enchantment = var1;
       this.values = var2;
    }
 
+   public LootItemConditionType getType() {
+      return LootItemConditions.TABLE_BONUS;
+   }
+
    public Set<LootContextParam<?>> getReferencedContextParams() {
       return ImmutableSet.of(LootContextParams.TOOL);
    }
 
    public boolean test(LootContext var1) {
@@ -40,13 +44,12 @@
       return () -> {
          return new BonusLevelTableCondition(var0, var1);
       };
    }
 
-   public static class Serializer extends LootItemCondition.Serializer<BonusLevelTableCondition> {
+   public static class Serializer implements net.minecraft.world.level.storage.loot.Serializer<BonusLevelTableCondition> {
       public Serializer() {
-         super(new ResourceLocation("table_bonus"), BonusLevelTableCondition.class);
       }
 
       public void serialize(JsonObject var1, BonusLevelTableCondition var2, JsonSerializationContext var3) {
          var1.addProperty("enchantment", Registry.ENCHANTMENT.getKey(var2.enchantment).toString());
          var1.add("chances", var3.serialize(var2.values));
