--- net/minecraft/world/level/storage/loot/predicates/LootItemKilledByPlayerCondition.java
+++ net/minecraft/world/level/storage/loot/predicates/LootItemKilledByPlayerCondition.java
@@ -3,21 +3,24 @@
 import com.google.common.collect.ImmutableSet;
 import com.google.gson.JsonDeserializationContext;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonSerializationContext;
 import java.util.Set;
-import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.level.storage.loot.LootContext;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParam;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParams;
 
 public class LootItemKilledByPlayerCondition implements LootItemCondition {
    private static final LootItemKilledByPlayerCondition INSTANCE = new LootItemKilledByPlayerCondition();
 
    private LootItemKilledByPlayerCondition() {
    }
 
+   public LootItemConditionType getType() {
+      return LootItemConditions.KILLED_BY_PLAYER;
+   }
+
    public Set<LootContextParam<?>> getReferencedContextParams() {
       return ImmutableSet.of(LootContextParams.LAST_DAMAGE_PLAYER);
    }
 
    public boolean test(LootContext var1) {
@@ -28,13 +31,12 @@
       return () -> {
          return INSTANCE;
       };
    }
 
-   public static class Serializer extends LootItemCondition.Serializer<LootItemKilledByPlayerCondition> {
-      protected Serializer() {
-         super(new ResourceLocation("killed_by_player"), LootItemKilledByPlayerCondition.class);
+   public static class Serializer implements net.minecraft.world.level.storage.loot.Serializer<LootItemKilledByPlayerCondition> {
+      public Serializer() {
       }
 
       public void serialize(JsonObject var1, LootItemKilledByPlayerCondition var2, JsonSerializationContext var3) {
       }
 
