--- net/minecraft/util/datafix/schemas/V1125.java
+++ net/minecraft/util/datafix/schemas/V1125.java
@@ -22,12 +22,12 @@
       super.registerTypes(var1, var2, var3);
       var1.registerType(false, References.ADVANCEMENTS, () -> {
          return DSL.optionalFields("minecraft:adventure/adventuring_time", DSL.optionalFields("criteria", DSL.compoundList(References.BIOME.in(var1), DSL.constType(DSL.string()))), "minecraft:adventure/kill_a_mob", DSL.optionalFields("criteria", DSL.compoundList(References.ENTITY_NAME.in(var1), DSL.constType(DSL.string()))), "minecraft:adventure/kill_all_mobs", DSL.optionalFields("criteria", DSL.compoundList(References.ENTITY_NAME.in(var1), DSL.constType(DSL.string()))), "minecraft:husbandry/bred_all_animals", DSL.optionalFields("criteria", DSL.compoundList(References.ENTITY_NAME.in(var1), DSL.constType(DSL.string()))));
       });
       var1.registerType(false, References.BIOME, () -> {
-         return DSL.constType(DSL.namespacedString());
+         return DSL.constType(namespacedString());
       });
       var1.registerType(false, References.ENTITY_NAME, () -> {
-         return DSL.constType(DSL.namespacedString());
+         return DSL.constType(namespacedString());
       });
    }
 }
