--- net/minecraft/network/protocol/game/ClientGamePacketListener.java
+++ net/minecraft/network/protocol/game/ClientGamePacketListener.java
@@ -109,11 +109,11 @@
 
    void handleSetPlayerTeamPacket(ClientboundSetPlayerTeamPacket var1);
 
    void handleSetScore(ClientboundSetScorePacket var1);
 
-   void handleSetSpawn(ClientboundSetSpawnPositionPacket var1);
+   void handleSetSpawn(ClientboundSetDefaultSpawnPositionPacket var1);
 
    void handleSetTime(ClientboundSetTimePacket var1);
 
    void handleSoundEvent(ClientboundSoundPacket var1);
 
