--- net/minecraft/network/protocol/game/ServerboundSignUpdatePacket.java
+++ net/minecraft/network/protocol/game/ServerboundSignUpdatePacket.java
@@ -1,23 +1,22 @@
 package net.minecraft.network.protocol.game;
 
 import java.io.IOException;
 import net.minecraft.core.BlockPos;
 import net.minecraft.network.FriendlyByteBuf;
-import net.minecraft.network.chat.Component;
 import net.minecraft.network.protocol.Packet;
 
 public class ServerboundSignUpdatePacket implements Packet<ServerGamePacketListener> {
    private BlockPos pos;
    private String[] lines;
 
    public ServerboundSignUpdatePacket() {
    }
 
-   public ServerboundSignUpdatePacket(BlockPos var1, Component var2, Component var3, Component var4, Component var5) {
+   public ServerboundSignUpdatePacket(BlockPos var1, String var2, String var3, String var4, String var5) {
       this.pos = var1;
-      this.lines = new String[]{var2.getString(), var3.getString(), var4.getString(), var5.getString()};
+      this.lines = new String[]{var2, var3, var4, var5};
    }
 
    public void read(FriendlyByteBuf var1) throws IOException {
       this.pos = var1.readBlockPos();
       this.lines = new String[4];
