--- net/minecraft/world/entity/projectile/EvokerFangs.java
+++ net/minecraft/world/entity/projectile/EvokerFangs.java
@@ -56,20 +56,20 @@
       return this.owner;
    }
 
    protected void readAdditionalSaveData(CompoundTag var1) {
       this.warmupDelayTicks = var1.getInt("Warmup");
-      if (var1.hasUUID("OwnerUUID")) {
-         this.ownerUUID = var1.getUUID("OwnerUUID");
+      if (var1.hasUUID("Owner")) {
+         this.ownerUUID = var1.getUUID("Owner");
       }
 
    }
 
    protected void addAdditionalSaveData(CompoundTag var1) {
       var1.putInt("Warmup", this.warmupDelayTicks);
       if (this.ownerUUID != null) {
-         var1.putUUID("OwnerUUID", this.ownerUUID);
+         var1.putUUID("Owner", this.ownerUUID);
       }
 
    }
 
    public void tick() {
