--- net/minecraft/world/level/chunk/LevelChunkSection.java
+++ net/minecraft/world/level/chunk/LevelChunkSection.java
@@ -1,7 +1,8 @@
 package net.minecraft.world.level.chunk;
 
+import java.util.function.Predicate;
 import javax.annotation.Nullable;
 import net.minecraft.nbt.NbtUtils;
 import net.minecraft.network.FriendlyByteBuf;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.Blocks;
@@ -141,11 +142,11 @@
 
    public int getSerializedSize() {
       return 2 + this.states.getSerializedSize();
    }
 
-   public boolean maybeHas(BlockState var1) {
+   public boolean maybeHas(Predicate<BlockState> var1) {
       return this.states.maybeHas(var1);
    }
 
    static {
       GLOBAL_BLOCKSTATE_PALETTE = new GlobalPalette(Block.BLOCK_STATE_REGISTRY, Blocks.AIR.defaultBlockState());
