--- net/minecraft/world/level/chunk/Palette.java
+++ net/minecraft/world/level/chunk/Palette.java
@@ -1,15 +1,16 @@
 package net.minecraft.world.level.chunk;
 
+import java.util.function.Predicate;
 import javax.annotation.Nullable;
 import net.minecraft.nbt.ListTag;
 import net.minecraft.network.FriendlyByteBuf;
 
 public interface Palette<T> {
    int idFor(T var1);
 
-   boolean maybeHas(T var1);
+   boolean maybeHas(Predicate<T> var1);
 
    @Nullable
    T valueFor(int var1);
 
    void write(FriendlyByteBuf var1);
