--- net/minecraft/world/level/storage/loot/entries/SequentialEntry.java
+++ net/minecraft/world/level/storage/loot/entries/SequentialEntry.java
@@ -5,10 +5,14 @@
 public class SequentialEntry extends CompositeEntryBase {
    SequentialEntry(LootPoolEntryContainer[] var1, LootItemCondition[] var2) {
       super(var1, var2);
    }
 
+   public LootPoolEntryType getType() {
+      return LootPoolEntries.SEQUENCE;
+   }
+
    protected ComposableEntryContainer compose(ComposableEntryContainer[] var1) {
       switch(var1.length) {
       case 0:
          return ALWAYS_TRUE;
       case 1:
