--- net/minecraft/world/item/BannerPatternItem.java
+++ net/minecraft/world/item/BannerPatternItem.java
@@ -2,10 +2,11 @@
 
 import java.util.List;
 import javax.annotation.Nullable;
 import net.minecraft.ChatFormatting;
 import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
 import net.minecraft.network.chat.TranslatableComponent;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.entity.BannerPattern;
 
 public class BannerPatternItem extends Item {
@@ -22,9 +23,9 @@
 
    public void appendHoverText(ItemStack var1, @Nullable Level var2, List<Component> var3, TooltipFlag var4) {
       var3.add(this.getDisplayName().withStyle(ChatFormatting.GRAY));
    }
 
-   public Component getDisplayName() {
-      return new TranslatableComponent(this.getDescriptionId() + ".desc", new Object[0]);
+   public MutableComponent getDisplayName() {
+      return new TranslatableComponent(this.getDescriptionId() + ".desc");
    }
 }
