--- net/minecraft/world/item/BedItem.java
+++ net/minecraft/world/item/BedItem.java
@@ -1,7 +1,8 @@
 package net.minecraft.world.item;
 
+import net.minecraft.world.item.context.BlockPlaceContext;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.state.BlockState;
 
 public class BedItem extends BlockItem {
    public BedItem(Block var1, Item.Properties var2) {
