--- net/minecraft/world/item/GameMasterBlockItem.java
+++ net/minecraft/world/item/GameMasterBlockItem.java
@@ -1,9 +1,10 @@
 package net.minecraft.world.item;
 
 import javax.annotation.Nullable;
 import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.context.BlockPlaceContext;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.state.BlockState;
 
 public class GameMasterBlockItem extends BlockItem {
    public GameMasterBlockItem(Block var1, Item.Properties var2) {
