--- net/minecraft/world/item/MinecartItem.java
+++ net/minecraft/world/item/MinecartItem.java
@@ -3,13 +3,15 @@
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.BlockSource;
 import net.minecraft.core.Direction;
 import net.minecraft.core.dispenser.DefaultDispenseItemBehavior;
 import net.minecraft.core.dispenser.DispenseItemBehavior;
+import net.minecraft.server.level.ServerLevel;
 import net.minecraft.tags.BlockTags;
 import net.minecraft.world.InteractionResult;
 import net.minecraft.world.entity.vehicle.AbstractMinecart;
+import net.minecraft.world.item.context.UseOnContext;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.BaseRailBlock;
 import net.minecraft.world.level.block.DispenserBlock;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.block.state.properties.RailShape;
@@ -18,11 +20,11 @@
    private static final DispenseItemBehavior DISPENSE_ITEM_BEHAVIOR = new DefaultDispenseItemBehavior() {
       private final DefaultDispenseItemBehavior defaultDispenseItemBehavior = new DefaultDispenseItemBehavior();
 
       public ItemStack execute(BlockSource var1, ItemStack var2) {
          Direction var3 = (Direction)var1.getBlockState().getValue(DispenserBlock.FACING);
-         Level var4 = var1.getLevel();
+         ServerLevel var4 = var1.getLevel();
          double var5 = var1.x() + (double)var3.getStepX() * 1.125D;
          double var7 = Math.floor(var1.y()) + (double)var3.getStepY();
          double var9 = var1.z() + (double)var3.getStepZ() * 1.125D;
          BlockPos var11 = var1.getPos().relative(var3);
          BlockState var12 = var4.getBlockState(var11);
@@ -92,9 +94,9 @@
 
             var2.addFreshEntity(var9);
          }
 
          var5.shrink(1);
-         return InteractionResult.SUCCESS;
+         return InteractionResult.sidedSuccess(var2.isClientSide);
       }
    }
 }
