--- net/minecraft/world/item/StandingAndWallBlockItem.java
+++ net/minecraft/world/item/StandingAndWallBlockItem.java
@@ -2,10 +2,11 @@
 
 import java.util.Map;
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
+import net.minecraft.world.item.context.BlockPlaceContext;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.phys.shapes.CollisionContext;
 
