--- net/minecraft/world/level/levelgen/PhantomSpawner.java
+++ net/minecraft/world/level/levelgen/PhantomSpawner.java
@@ -13,16 +13,17 @@
 import net.minecraft.world.entity.EntityType;
 import net.minecraft.world.entity.MobSpawnType;
 import net.minecraft.world.entity.SpawnGroupData;
 import net.minecraft.world.entity.monster.Phantom;
 import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.CustomSpawner;
 import net.minecraft.world.level.GameRules;
 import net.minecraft.world.level.NaturalSpawner;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.material.FluidState;
 
-public class PhantomSpawner {
+public class PhantomSpawner implements CustomSpawner {
    private int nextTick;
 
    public PhantomSpawner() {
    }
 
@@ -36,11 +37,11 @@
          --this.nextTick;
          if (this.nextTick > 0) {
             return 0;
          } else {
             this.nextTick += (60 + var4.nextInt(60)) * 20;
-            if (var1.getSkyDarken() < 5 && var1.dimension.isHasSkyLight()) {
+            if (var1.getSkyDarken() < 5 && var1.dimensionType().hasSkyLight()) {
                return 0;
             } else {
                int var5 = 0;
                Iterator var6 = var1.players().iterator();
 
@@ -62,12 +63,12 @@
                                  }
 
                                  var7 = (Player)var6.next();
                               } while(var7.isSpectator());
 
-                              var8 = new BlockPos(var7);
-                           } while(var1.dimension.isHasSkyLight() && (var8.getY() < var1.getSeaLevel() || !var1.canSeeSky(var8)));
+                              var8 = var7.blockPosition();
+                           } while(var1.dimensionType().hasSkyLight() && (var8.getY() < var1.getSeaLevel() || !var1.canSeeSky(var8)));
 
                            var9 = var1.getCurrentDifficultyAt(var8);
                         } while(!var9.isHarderThan(var4.nextFloat() * 3.0F));
 
                         ServerStatsCounter var10 = ((ServerPlayer)var7).getStats();
@@ -76,11 +77,11 @@
                      } while(var4.nextInt(var11) < 72000);
 
                      var13 = var8.above(20 + var4.nextInt(15)).east(-10 + var4.nextInt(21)).south(-10 + var4.nextInt(21));
                      var14 = var1.getBlockState(var13);
                      var15 = var1.getFluidState(var13);
-                  } while(!NaturalSpawner.isValidEmptySpawnBlock(var1, var13, var14, var15));
+                  } while(!NaturalSpawner.isValidEmptySpawnBlock(var1, var13, var14, var15, EntityType.PHANTOM));
 
                   SpawnGroupData var16 = null;
                   int var17 = 1 + var4.nextInt(var9.getDifficulty().getId() + 1);
 
                   for(int var18 = 0; var18 < var17; ++var18) {
