--- net/minecraft/util/datafix/fixes/ChunkBiomeFix.java
+++ net/minecraft/util/datafix/fixes/ChunkBiomeFix.java
@@ -19,11 +19,11 @@
       Type var1 = this.getInputSchema().getType(References.CHUNK);
       OpticFinder var2 = var1.findField("Level");
       return this.fixTypeEverywhereTyped("Leaves fix", var1, (var1x) -> {
          return var1x.updateTyped(var2, (var0) -> {
             return var0.update(DSL.remainderFinder(), (var0x) -> {
-               Optional var1 = var0x.get("Biomes").asIntStreamOpt();
+               Optional var1 = var0x.get("Biomes").asIntStreamOpt().result();
                if (!var1.isPresent()) {
                   return var0x;
                } else {
                   int[] var2 = ((IntStream)var1.get()).toArray();
                   int[] var3 = new int[1024];
