--- net/minecraft/util/datafix/fixes/EntityRedundantChanceTagsFix.java
+++ net/minecraft/util/datafix/fixes/EntityRedundantChanceTagsFix.java
@@ -1,12 +1,12 @@
 package net.minecraft.util.datafix.fixes;
 
 import com.mojang.datafixers.DSL;
 import com.mojang.datafixers.DataFix;
-import com.mojang.datafixers.Dynamic;
 import com.mojang.datafixers.TypeRewriteRule;
 import com.mojang.datafixers.schemas.Schema;
+import com.mojang.serialization.Dynamic;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.stream.Stream;
 
 public class EntityRedundantChanceTagsFix extends DataFix {
