--- net/minecraft/util/datafix/fixes/EntityCatSplitFix.java
+++ net/minecraft/util/datafix/fixes/EntityCatSplitFix.java
@@ -1,10 +1,10 @@
 package net.minecraft.util.datafix.fixes;
 
-import com.mojang.datafixers.Dynamic;
 import com.mojang.datafixers.schemas.Schema;
 import com.mojang.datafixers.util.Pair;
+import com.mojang.serialization.Dynamic;
 import java.util.Objects;
 
 public class EntityCatSplitFix extends SimpleEntityRenameFix {
    public EntityCatSplitFix(Schema var1, boolean var2) {
       super("EntityCatSplitFix", var1, var2);
