--- net/minecraft/util/datafix/fixes/ObjectiveRenderTypeFix.java
+++ net/minecraft/util/datafix/fixes/ObjectiveRenderTypeFix.java
@@ -24,11 +24,11 @@
          throw new IllegalStateException("Objective type is not what was expected.");
       } else {
          return this.fixTypeEverywhere("ObjectiveRenderTypeFix", var1, (var0) -> {
             return (var0x) -> {
                return var0x.mapSecond((var0) -> {
-                  Optional var1 = var0.get("RenderType").asString();
+                  Optional var1 = var0.get("RenderType").asString().result();
                   if (!var1.isPresent()) {
                      String var2 = var0.get("CriteriaName").asString("");
                      ObjectiveCriteria.RenderType var3 = getRenderType(var2);
                      return var0.set("RenderType", var0.createString(var3.getId()));
                   } else {
