--- net/minecraft/util/datafix/fixes/OptionsKeyLwjgl3Fix.java
+++ net/minecraft/util/datafix/fixes/OptionsKeyLwjgl3Fix.java
@@ -1,14 +1,14 @@
 package net.minecraft.util.datafix.fixes;
 
 import com.mojang.datafixers.DSL;
 import com.mojang.datafixers.DataFix;
 import com.mojang.datafixers.DataFixUtils;
-import com.mojang.datafixers.Dynamic;
 import com.mojang.datafixers.TypeRewriteRule;
 import com.mojang.datafixers.schemas.Schema;
 import com.mojang.datafixers.util.Pair;
+import com.mojang.serialization.Dynamic;
 import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
 import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
 import java.util.Map;
 import java.util.stream.Collectors;
 
@@ -159,10 +159,10 @@
                      }
                   } else {
                      return Pair.of(var0.getKey(), var0.getValue());
                   }
                }).collect(Collectors.toMap(Pair::getFirst, Pair::getSecond)));
-            }).orElse(var0x);
+            }).result().orElse(var0x);
          });
       });
    }
 }
