--- net/minecraft/world/level/levelgen/synth/NormalNoise.java
+++ net/minecraft/world/level/levelgen/synth/NormalNoise.java
@@ -1,0 +1,35 @@
+package net.minecraft.world.level.levelgen.synth;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.stream.IntStream;
+import net.minecraft.world.level.levelgen.WorldgenRandom;
+
+public class NormalNoise {
+   private final double valueFactor;
+   private final PerlinNoise first;
+   private final PerlinNoise second;
+
+   public NormalNoise(WorldgenRandom var1, IntStream var2) {
+      this(var1, (List)var2.boxed().collect(ImmutableList.toImmutableList()));
+   }
+
+   public NormalNoise(WorldgenRandom var1, List<Integer> var2) {
+      this.first = new PerlinNoise(var1, var2);
+      this.second = new PerlinNoise(var1, var2);
+      int var3 = (Integer)var2.stream().min(Integer::compareTo).orElse(0);
+      int var4 = (Integer)var2.stream().max(Integer::compareTo).orElse(0);
+      this.valueFactor = 0.16666666666666666D / expectedDeviation(var4 - var3);
+   }
+
+   private static double expectedDeviation(int var0) {
+      return 0.1D * (1.0D + 1.0D / (double)(var0 + 1));
+   }
+
+   public double getValue(double var1, double var3, double var5) {
+      double var7 = var1 * 1.0181268882175227D;
+      double var9 = var3 * 1.0181268882175227D;
+      double var11 = var5 * 1.0181268882175227D;
+      return (this.first.getValue(var1, var3, var5) + this.second.getValue(var7, var9, var11)) * this.valueFactor;
+   }
+}
