--- net/minecraft/client/sounds/WeighedSoundEvents.java
+++ net/minecraft/client/sounds/WeighedSoundEvents.java
@@ -12,15 +12,16 @@
 
 public class WeighedSoundEvents implements Weighted<Sound> {
    private final List<Weighted<Sound>> list = Lists.newArrayList();
    private final Random random = new Random();
    private final ResourceLocation location;
+   @Nullable
    private final Component subtitle;
 
    public WeighedSoundEvents(ResourceLocation var1, @Nullable String var2) {
       this.location = var1;
-      this.subtitle = var2 == null ? null : new TranslatableComponent(var2, new Object[0]);
+      this.subtitle = var2 == null ? null : new TranslatableComponent(var2);
    }
 
    public int getWeight() {
       int var1 = 0;
 
