--- net/minecraft/world/level/levelgen/structure/templatesystem/AlwaysTrueTest.java
+++ net/minecraft/world/level/levelgen/structure/templatesystem/AlwaysTrueTest.java
@@ -1,25 +1,23 @@
 package net.minecraft.world.level.levelgen.structure.templatesystem;
 
-import com.mojang.datafixers.Dynamic;
-import com.mojang.datafixers.types.DynamicOps;
+import com.mojang.serialization.Codec;
 import java.util.Random;
 import net.minecraft.world.level.block.state.BlockState;
 
 public class AlwaysTrueTest extends RuleTest {
+   public static final Codec<AlwaysTrueTest> CODEC = Codec.unit(() -> {
+      return INSTANCE;
+   });
    public static final AlwaysTrueTest INSTANCE = new AlwaysTrueTest();
 
    private AlwaysTrueTest() {
    }
 
    public boolean test(BlockState var1, Random var2) {
       return true;
    }
 
-   protected RuleTestType getType() {
+   protected RuleTestType<?> getType() {
       return RuleTestType.ALWAYS_TRUE_TEST;
    }
-
-   protected <T> Dynamic<T> getDynamic(DynamicOps<T> var1) {
-      return new Dynamic(var1, var1.emptyMap());
-   }
 }
