--- net/minecraft/world/level/levelgen/placement/EmeraldPlacementDecorator.java
+++ net/minecraft/world/level/levelgen/placement/EmeraldPlacementDecorator.java
@@ -1,17 +1,16 @@
 package net.minecraft.world.level.levelgen.placement;
 
-import com.mojang.datafixers.Dynamic;
+import com.mojang.serialization.Codec;
 import java.util.Random;
-import java.util.function.Function;
 import java.util.stream.IntStream;
 import java.util.stream.Stream;
 import net.minecraft.core.BlockPos;
 import net.minecraft.world.level.levelgen.feature.configurations.NoneDecoratorConfiguration;
 
 public class EmeraldPlacementDecorator extends SimpleFeatureDecorator<NoneDecoratorConfiguration> {
-   public EmeraldPlacementDecorator(Function<Dynamic<?>, ? extends NoneDecoratorConfiguration> var1) {
+   public EmeraldPlacementDecorator(Codec<NoneDecoratorConfiguration> var1) {
       super(var1);
    }
 
    public Stream<BlockPos> place(Random var1, NoneDecoratorConfiguration var2, BlockPos var3) {
       int var4 = 3 + var1.nextInt(6);
