--- net/minecraft/world/level/levelgen/placement/NopePlacementDecorator.java
+++ net/minecraft/world/level/levelgen/placement/NopePlacementDecorator.java
@@ -1,16 +1,15 @@
 package net.minecraft.world.level.levelgen.placement;
 
-import com.mojang.datafixers.Dynamic;
+import com.mojang.serialization.Codec;
 import java.util.Random;
-import java.util.function.Function;
 import java.util.stream.Stream;
 import net.minecraft.core.BlockPos;
 import net.minecraft.world.level.levelgen.feature.configurations.NoneDecoratorConfiguration;
 
 public class NopePlacementDecorator extends SimpleFeatureDecorator<NoneDecoratorConfiguration> {
-   public NopePlacementDecorator(Function<Dynamic<?>, ? extends NoneDecoratorConfiguration> var1) {
+   public NopePlacementDecorator(Codec<NoneDecoratorConfiguration> var1) {
       super(var1);
    }
 
    public Stream<BlockPos> place(Random var1, NoneDecoratorConfiguration var2, BlockPos var3) {
       return Stream.of(var3);
