--- net/minecraft/world/level/material/Fluid.java
+++ net/minecraft/world/level/material/Fluid.java
@@ -20,11 +20,11 @@
    private FluidState defaultFluidState;
 
    protected Fluid() {
       StateDefinition.Builder var1 = new StateDefinition.Builder(this);
       this.createFluidStateDefinition(var1);
-      this.stateDefinition = var1.create(FluidStateImpl::new);
+      this.stateDefinition = var1.create(Fluid::defaultFluidState, FluidState::new);
       this.registerDefaultState((FluidState)this.stateDefinition.any());
    }
 
    protected void createFluidStateDefinition(StateDefinition.Builder<Fluid, FluidState> var1) {
    }
