--- net/minecraft/tags/SerializationTags.java
+++ net/minecraft/tags/SerializationTags.java
@@ -1,0 +1,13 @@
+package net.minecraft.tags;
+
+public class SerializationTags {
+   private static volatile TagContainer instance = TagContainer.of(BlockTags.getAllTags(), ItemTags.getAllTags(), FluidTags.getAllTags(), EntityTypeTags.getAllTags());
+
+   public static TagContainer getInstance() {
+      return instance;
+   }
+
+   public static void bind(TagContainer var0) {
+      instance = var0;
+   }
+}
