--- net/minecraft/world/entity/ai/goal/RangedCrossbowAttackGoal.java
+++ net/minecraft/world/entity/ai/goal/RangedCrossbowAttackGoal.java
@@ -1,8 +1,9 @@
 package net.minecraft.world.entity.ai.goal;
 
 import java.util.EnumSet;
+import net.minecraft.util.IntRange;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.monster.CrossbowAttackMob;
 import net.minecraft.world.entity.monster.Monster;
 import net.minecraft.world.entity.monster.RangedAttackMob;
@@ -10,16 +11,18 @@
 import net.minecraft.world.item.CrossbowItem;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
 
 public class RangedCrossbowAttackGoal<T extends Monster & RangedAttackMob & CrossbowAttackMob> extends Goal {
+   public static final IntRange PATHFINDING_DELAY_RANGE = new IntRange(20, 40);
    private final T mob;
    private RangedCrossbowAttackGoal.CrossbowState crossbowState;
    private final double speedModifier;
    private final float attackRadiusSqr;
    private int seeTime;
    private int attackDelay;
+   private int updatePathDelay;
 
    public RangedCrossbowAttackGoal(T var1, double var2, float var4) {
       this.crossbowState = RangedCrossbowAttackGoal.CrossbowState.UNCHARGED;
       this.mob = var1;
       this.speedModifier = var2;
@@ -72,12 +75,17 @@
          }
 
          double var4 = this.mob.distanceToSqr(var1);
          boolean var6 = (var4 > (double)this.attackRadiusSqr || this.seeTime < 5) && this.attackDelay == 0;
          if (var6) {
-            this.mob.getNavigation().moveTo((Entity)var1, this.canRun() ? this.speedModifier : this.speedModifier * 0.5D);
+            --this.updatePathDelay;
+            if (this.updatePathDelay <= 0) {
+               this.mob.getNavigation().moveTo((Entity)var1, this.canRun() ? this.speedModifier : this.speedModifier * 0.5D);
+               this.updatePathDelay = PATHFINDING_DELAY_RANGE.randomValue(this.mob.getRandom());
+            }
          } else {
+            this.updatePathDelay = 0;
             this.mob.getNavigation().stop();
          }
 
          this.mob.getLookControl().setLookAt(var1, 30.0F, 30.0F);
          if (this.crossbowState == RangedCrossbowAttackGoal.CrossbowState.UNCHARGED) {
