--- net/minecraft/world/entity/ai/goal/TradeWithPlayerGoal.java
+++ net/minecraft/world/entity/ai/goal/TradeWithPlayerGoal.java
@@ -15,11 +15,11 @@
    public boolean canUse() {
       if (!this.mob.isAlive()) {
          return false;
       } else if (this.mob.isInWater()) {
          return false;
-      } else if (!this.mob.onGround) {
+      } else if (!this.mob.isOnGround()) {
          return false;
       } else if (this.mob.hurtMarked) {
          return false;
       } else {
          Player var1 = this.mob.getTradingPlayer();
