--- net/minecraft/world/entity/ai/goal/ZombieAttackGoal.java
+++ net/minecraft/world/entity/ai/goal/ZombieAttackGoal.java
@@ -22,11 +22,11 @@
    }
 
    public void tick() {
       super.tick();
       ++this.raiseArmTicks;
-      if (this.raiseArmTicks >= 5 && this.attackTime < 10) {
+      if (this.raiseArmTicks >= 5 && this.getTicksUntilNextAttack() < this.getAttackInterval() / 2) {
          this.zombie.setAggressive(true);
       } else {
          this.zombie.setAggressive(false);
       }
 
