--- net/minecraft/world/level/block/state/properties/NoteBlockInstrument.java
+++ net/minecraft/world/level/block/state/properties/NoteBlockInstrument.java
@@ -2,11 +2,10 @@
 
 import net.minecraft.sounds.SoundEvent;
 import net.minecraft.sounds.SoundEvents;
 import net.minecraft.tags.BlockTags;
 import net.minecraft.util.StringRepresentable;
-import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.material.Material;
 
 public enum NoteBlockInstrument implements StringRepresentable {
@@ -42,42 +41,41 @@
    public SoundEvent getSoundEvent() {
       return this.soundEvent;
    }
 
    public static NoteBlockInstrument byState(BlockState var0) {
-      Block var1 = var0.getBlock();
-      if (var1 == Blocks.CLAY) {
+      if (var0.is(Blocks.CLAY)) {
          return FLUTE;
-      } else if (var1 == Blocks.GOLD_BLOCK) {
+      } else if (var0.is(Blocks.GOLD_BLOCK)) {
          return BELL;
-      } else if (var1.is(BlockTags.WOOL)) {
+      } else if (var0.is(BlockTags.WOOL)) {
          return GUITAR;
-      } else if (var1 == Blocks.PACKED_ICE) {
+      } else if (var0.is(Blocks.PACKED_ICE)) {
          return CHIME;
-      } else if (var1 == Blocks.BONE_BLOCK) {
+      } else if (var0.is(Blocks.BONE_BLOCK)) {
          return XYLOPHONE;
-      } else if (var1 == Blocks.IRON_BLOCK) {
+      } else if (var0.is(Blocks.IRON_BLOCK)) {
          return IRON_XYLOPHONE;
-      } else if (var1 == Blocks.SOUL_SAND) {
+      } else if (var0.is(Blocks.SOUL_SAND)) {
          return COW_BELL;
-      } else if (var1 == Blocks.PUMPKIN) {
+      } else if (var0.is(Blocks.PUMPKIN)) {
          return DIDGERIDOO;
-      } else if (var1 == Blocks.EMERALD_BLOCK) {
+      } else if (var0.is(Blocks.EMERALD_BLOCK)) {
          return BIT;
-      } else if (var1 == Blocks.HAY_BLOCK) {
+      } else if (var0.is(Blocks.HAY_BLOCK)) {
          return BANJO;
-      } else if (var1 == Blocks.GLOWSTONE) {
+      } else if (var0.is(Blocks.GLOWSTONE)) {
          return PLING;
       } else {
-         Material var2 = var0.getMaterial();
-         if (var2 == Material.STONE) {
+         Material var1 = var0.getMaterial();
+         if (var1 == Material.STONE) {
             return BASEDRUM;
-         } else if (var2 == Material.SAND) {
+         } else if (var1 == Material.SAND) {
             return SNARE;
-         } else if (var2 == Material.GLASS) {
+         } else if (var1 == Material.GLASS) {
             return HAT;
          } else {
-            return var2 == Material.WOOD ? BASS : HARP;
+            return var1 != Material.WOOD && var1 != Material.NETHER_WOOD ? HARP : BASS;
          }
       }
    }
 }
