--- net/minecraft/world/level/biome/BirchForestHillsBiome.java
+++ net/minecraft/world/level/biome/BirchForestHillsBiome.java
@@ -1,22 +1,18 @@
 package net.minecraft.world.level.biome;
 
+import com.google.common.collect.ImmutableList;
 import net.minecraft.world.entity.EntityType;
 import net.minecraft.world.entity.MobCategory;
-import net.minecraft.world.level.levelgen.feature.Feature;
-import net.minecraft.world.level.levelgen.feature.MineshaftFeature;
-import net.minecraft.world.level.levelgen.feature.configurations.FeatureConfiguration;
-import net.minecraft.world.level.levelgen.feature.configurations.MineshaftConfiguration;
 import net.minecraft.world.level.levelgen.surfacebuilders.SurfaceBuilder;
 
 public final class BirchForestHillsBiome extends Biome {
    public BirchForestHillsBiome() {
-      super((new Biome.BiomeBuilder()).surfaceBuilder(SurfaceBuilder.DEFAULT, SurfaceBuilder.CONFIG_GRASS).precipitation(Biome.Precipitation.RAIN).biomeCategory(Biome.BiomeCategory.FOREST).depth(0.45F).scale(0.3F).temperature(0.6F).downfall(0.6F).waterColor(4159204).waterFogColor(329011).parent((String)null));
-      this.addStructureStart(Feature.MINESHAFT.configured(new MineshaftConfiguration(0.004D, MineshaftFeature.Type.NORMAL)));
-      this.addStructureStart(Feature.STRONGHOLD.configured(FeatureConfiguration.NONE));
+      super((new Biome.BiomeBuilder()).surfaceBuilder(SurfaceBuilder.DEFAULT, SurfaceBuilder.CONFIG_GRASS).precipitation(Biome.Precipitation.RAIN).biomeCategory(Biome.BiomeCategory.FOREST).depth(0.45F).scale(0.3F).temperature(0.6F).downfall(0.6F).specialEffects((new BiomeSpecialEffects.Builder()).waterColor(4159204).waterFogColor(329011).fogColor(12638463).ambientMoodSound(AmbientMoodSettings.LEGACY_CAVE_SETTINGS).build()).parent((String)null).optimalParameters(ImmutableList.of(new Biome.ClimateParameters(-0.1F, 0.2F, 0.25F, 0.0F, 1.0F))));
+      BiomeDefaultFeatures.addDefaultOverworldLandStructures(this);
+      this.addStructureStart(BiomeDefaultFeatures.RUINED_PORTAL_STANDARD);
       BiomeDefaultFeatures.addDefaultCarvers(this);
-      BiomeDefaultFeatures.addStructureFeaturePlacement(this);
       BiomeDefaultFeatures.addDefaultLakes(this);
       BiomeDefaultFeatures.addDefaultMonsterRoom(this);
       BiomeDefaultFeatures.addForestFlowers(this);
       BiomeDefaultFeatures.addDefaultUndergroundVariety(this);
       BiomeDefaultFeatures.addDefaultOres(this);
