--- net/minecraft/world/level/biome/CheckerboardColumnBiomeSource.java
+++ net/minecraft/world/level/biome/CheckerboardColumnBiomeSource.java
@@ -1,18 +1,39 @@
 package net.minecraft.world.level.biome;
 
-import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ImmutableList;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import java.util.List;
+import java.util.function.Supplier;
 
 public class CheckerboardColumnBiomeSource extends BiomeSource {
-   private final Biome[] allowedBiomes;
+   public static final Codec<CheckerboardColumnBiomeSource> CODEC = RecordCodecBuilder.create((var0) -> {
+      return var0.group(Biome.CODEC.listOf().fieldOf("biomes").forGetter((var0x) -> {
+         return var0x.allowedBiomes;
+      }), Codec.intRange(0, 62).fieldOf("scale").orElse(2).forGetter((var0x) -> {
+         return var0x.size;
+      })).apply(var0, CheckerboardColumnBiomeSource::new);
+   });
+   private final List<Supplier<Biome>> allowedBiomes;
    private final int bitShift;
+   private final int size;
 
-   public CheckerboardColumnBiomeSource(CheckerboardBiomeSourceSettings var1) {
-      super(ImmutableSet.copyOf(var1.getAllowedBiomes()));
-      this.allowedBiomes = var1.getAllowedBiomes();
-      this.bitShift = var1.getSize() + 2;
+   public CheckerboardColumnBiomeSource(List<Supplier<Biome>> var1, int var2) {
+      super((List)var1.stream().map(Supplier::get).collect(ImmutableList.toImmutableList()));
+      this.allowedBiomes = var1;
+      this.bitShift = var2 + 2;
+      this.size = var2;
    }
 
+   protected Codec<? extends BiomeSource> codec() {
+      return CODEC;
+   }
+
+   public BiomeSource withSeed(long var1) {
+      return this;
+   }
+
    public Biome getNoiseBiome(int var1, int var2, int var3) {
-      return this.allowedBiomes[Math.abs(((var1 >> this.bitShift) + (var3 >> this.bitShift)) % this.allowedBiomes.length)];
+      return (Biome)((Supplier)this.allowedBiomes.get(Math.floorMod((var1 >> this.bitShift) + (var3 >> this.bitShift), this.allowedBiomes.size()))).get();
    }
 }
