--- net/minecraft/world/level/biome/SnowyTaigaBiome.java
+++ net/minecraft/world/level/biome/SnowyTaigaBiome.java
@@ -1,23 +1,18 @@
 package net.minecraft.world.level.biome;
 
 import net.minecraft.world.entity.EntityType;
 import net.minecraft.world.entity.MobCategory;
-import net.minecraft.world.level.levelgen.feature.Feature;
-import net.minecraft.world.level.levelgen.feature.MineshaftFeature;
-import net.minecraft.world.level.levelgen.feature.configurations.FeatureConfiguration;
-import net.minecraft.world.level.levelgen.feature.configurations.MineshaftConfiguration;
 import net.minecraft.world.level.levelgen.surfacebuilders.SurfaceBuilder;
 
 public final class SnowyTaigaBiome extends Biome {
    public SnowyTaigaBiome() {
-      super((new Biome.BiomeBuilder()).surfaceBuilder(SurfaceBuilder.DEFAULT, SurfaceBuilder.CONFIG_GRASS).precipitation(Biome.Precipitation.SNOW).biomeCategory(Biome.BiomeCategory.TAIGA).depth(0.2F).scale(0.2F).temperature(-0.5F).downfall(0.4F).waterColor(4020182).waterFogColor(329011).parent((String)null));
-      this.addStructureStart(Feature.IGLOO.configured(FeatureConfiguration.NONE));
-      this.addStructureStart(Feature.MINESHAFT.configured(new MineshaftConfiguration(0.004D, MineshaftFeature.Type.NORMAL)));
-      this.addStructureStart(Feature.STRONGHOLD.configured(FeatureConfiguration.NONE));
+      super((new Biome.BiomeBuilder()).surfaceBuilder(SurfaceBuilder.DEFAULT, SurfaceBuilder.CONFIG_GRASS).precipitation(Biome.Precipitation.SNOW).biomeCategory(Biome.BiomeCategory.TAIGA).depth(0.2F).scale(0.2F).temperature(-0.5F).downfall(0.4F).specialEffects((new BiomeSpecialEffects.Builder()).waterColor(4020182).waterFogColor(329011).fogColor(12638463).ambientMoodSound(AmbientMoodSettings.LEGACY_CAVE_SETTINGS).build()).parent((String)null));
+      this.addStructureStart(BiomeDefaultFeatures.IGLOO);
+      BiomeDefaultFeatures.addDefaultOverworldLandStructures(this);
+      this.addStructureStart(BiomeDefaultFeatures.RUINED_PORTAL_STANDARD);
       BiomeDefaultFeatures.addDefaultCarvers(this);
-      BiomeDefaultFeatures.addStructureFeaturePlacement(this);
       BiomeDefaultFeatures.addDefaultLakes(this);
       BiomeDefaultFeatures.addDefaultMonsterRoom(this);
       BiomeDefaultFeatures.addFerns(this);
       BiomeDefaultFeatures.addDefaultUndergroundVariety(this);
       BiomeDefaultFeatures.addDefaultOres(this);
