--- net/minecraft/world/level/biome/SnowyBeachBiome.java
+++ net/minecraft/world/level/biome/SnowyBeachBiome.java
@@ -1,24 +1,19 @@
 package net.minecraft.world.level.biome;
 
 import net.minecraft.world.entity.EntityType;
 import net.minecraft.world.entity.MobCategory;
-import net.minecraft.world.level.levelgen.feature.Feature;
-import net.minecraft.world.level.levelgen.feature.MineshaftFeature;
-import net.minecraft.world.level.levelgen.feature.configurations.BuriedTreasureConfiguration;
-import net.minecraft.world.level.levelgen.feature.configurations.MineshaftConfiguration;
-import net.minecraft.world.level.levelgen.feature.configurations.ShipwreckConfiguration;
 import net.minecraft.world.level.levelgen.surfacebuilders.SurfaceBuilder;
 
 public final class SnowyBeachBiome extends Biome {
    public SnowyBeachBiome() {
-      super((new Biome.BiomeBuilder()).surfaceBuilder(SurfaceBuilder.DEFAULT, SurfaceBuilder.CONFIG_DESERT).precipitation(Biome.Precipitation.SNOW).biomeCategory(Biome.BiomeCategory.BEACH).depth(0.0F).scale(0.025F).temperature(0.05F).downfall(0.3F).waterColor(4020182).waterFogColor(329011).parent((String)null));
-      this.addStructureStart(Feature.MINESHAFT.configured(new MineshaftConfiguration(0.004D, MineshaftFeature.Type.NORMAL)));
-      this.addStructureStart(Feature.BURIED_TREASURE.configured(new BuriedTreasureConfiguration(0.01F)));
-      this.addStructureStart(Feature.SHIPWRECK.configured(new ShipwreckConfiguration(true)));
+      super((new Biome.BiomeBuilder()).surfaceBuilder(SurfaceBuilder.DEFAULT, SurfaceBuilder.CONFIG_DESERT).precipitation(Biome.Precipitation.SNOW).biomeCategory(Biome.BiomeCategory.BEACH).depth(0.0F).scale(0.025F).temperature(0.05F).downfall(0.3F).specialEffects((new BiomeSpecialEffects.Builder()).waterColor(4020182).waterFogColor(329011).fogColor(12638463).ambientMoodSound(AmbientMoodSettings.LEGACY_CAVE_SETTINGS).build()).parent((String)null));
+      this.addStructureStart(BiomeDefaultFeatures.MINESHAFT);
+      this.addStructureStart(BiomeDefaultFeatures.BURIED_TREASURE);
+      this.addStructureStart(BiomeDefaultFeatures.SHIPWRECH_BEACHED);
+      this.addStructureStart(BiomeDefaultFeatures.RUINED_PORTAL_STANDARD);
       BiomeDefaultFeatures.addDefaultCarvers(this);
-      BiomeDefaultFeatures.addStructureFeaturePlacement(this);
       BiomeDefaultFeatures.addDefaultLakes(this);
       BiomeDefaultFeatures.addDefaultMonsterRoom(this);
       BiomeDefaultFeatures.addDefaultUndergroundVariety(this);
       BiomeDefaultFeatures.addDefaultOres(this);
       BiomeDefaultFeatures.addDefaultSoftDisks(this);
