--- net/minecraft/world/level/biome/WoodedBadlandsBiome.java
+++ net/minecraft/world/level/biome/WoodedBadlandsBiome.java
@@ -1,22 +1,17 @@
 package net.minecraft.world.level.biome;
 
 import net.minecraft.world.entity.EntityType;
 import net.minecraft.world.entity.MobCategory;
-import net.minecraft.world.level.levelgen.feature.Feature;
-import net.minecraft.world.level.levelgen.feature.MineshaftFeature;
-import net.minecraft.world.level.levelgen.feature.configurations.FeatureConfiguration;
-import net.minecraft.world.level.levelgen.feature.configurations.MineshaftConfiguration;
 import net.minecraft.world.level.levelgen.surfacebuilders.SurfaceBuilder;
 
 public final class WoodedBadlandsBiome extends Biome {
    public WoodedBadlandsBiome() {
-      super((new Biome.BiomeBuilder()).surfaceBuilder(SurfaceBuilder.WOODED_BADLANDS, SurfaceBuilder.CONFIG_BADLANDS).precipitation(Biome.Precipitation.NONE).biomeCategory(Biome.BiomeCategory.MESA).depth(1.5F).scale(0.025F).temperature(2.0F).downfall(0.0F).waterColor(4159204).waterFogColor(329011).parent((String)null));
-      this.addStructureStart(Feature.MINESHAFT.configured(new MineshaftConfiguration(0.004D, MineshaftFeature.Type.MESA)));
-      this.addStructureStart(Feature.STRONGHOLD.configured(FeatureConfiguration.NONE));
+      super((new Biome.BiomeBuilder()).surfaceBuilder(SurfaceBuilder.WOODED_BADLANDS, SurfaceBuilder.CONFIG_BADLANDS).precipitation(Biome.Precipitation.NONE).biomeCategory(Biome.BiomeCategory.MESA).depth(1.5F).scale(0.025F).temperature(2.0F).downfall(0.0F).specialEffects((new BiomeSpecialEffects.Builder()).waterColor(4159204).waterFogColor(329011).fogColor(12638463).ambientMoodSound(AmbientMoodSettings.LEGACY_CAVE_SETTINGS).build()).parent((String)null));
+      BiomeDefaultFeatures.addDefaultOverworldLandMesaStructures(this);
+      this.addStructureStart(BiomeDefaultFeatures.RUINED_PORTAL_MOUNTAIN);
       BiomeDefaultFeatures.addDefaultCarvers(this);
-      BiomeDefaultFeatures.addStructureFeaturePlacement(this);
       BiomeDefaultFeatures.addDefaultLakes(this);
       BiomeDefaultFeatures.addDefaultMonsterRoom(this);
       BiomeDefaultFeatures.addDefaultUndergroundVariety(this);
       BiomeDefaultFeatures.addDefaultOres(this);
       BiomeDefaultFeatures.addExtraGold(this);
