--- net/minecraft/world/level/levelgen/structure/BoundingBox.java
+++ net/minecraft/world/level/levelgen/structure/BoundingBox.java
@@ -31,10 +31,14 @@
 
    public static BoundingBox getUnknownBox() {
       return new BoundingBox(Integer.MAX_VALUE, Integer.MAX_VALUE, Integer.MAX_VALUE, Integer.MIN_VALUE, Integer.MIN_VALUE, Integer.MIN_VALUE);
    }
 
+   public static BoundingBox infinite() {
+      return new BoundingBox(Integer.MIN_VALUE, Integer.MIN_VALUE, Integer.MIN_VALUE, Integer.MAX_VALUE, Integer.MAX_VALUE, Integer.MAX_VALUE);
+   }
+
    public static BoundingBox orientBox(int var0, int var1, int var2, int var3, int var4, int var5, int var6, int var7, int var8, Direction var9) {
       switch(var9) {
       case NORTH:
          return new BoundingBox(var0 + var3, var1 + var4, var2 - var8 + 1 + var5, var0 + var6 - 1 + var3, var1 + var7 - 1 + var4, var2 + var5);
       case SOUTH:
@@ -116,10 +120,14 @@
 
    public BoundingBox moved(int var1, int var2, int var3) {
       return new BoundingBox(this.x0 + var1, this.y0 + var2, this.z0 + var3, this.x1 + var1, this.y1 + var2, this.z1 + var3);
    }
 
+   public void move(Vec3i var1) {
+      this.move(var1.getX(), var1.getY(), var1.getZ());
+   }
+
    public boolean isInside(Vec3i var1) {
       return var1.getX() >= this.x0 && var1.getX() <= this.x1 && var1.getZ() >= this.z0 && var1.getZ() <= this.z1 && var1.getY() >= this.y0 && var1.getY() <= this.y1;
    }
 
    public Vec3i getLength() {
