--- net/minecraft/server/players/GameProfileCache.java
+++ net/minecraft/server/players/GameProfileCache.java
@@ -11,10 +11,11 @@
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonParseException;
 import com.google.gson.JsonSerializationContext;
 import com.google.gson.JsonSerializer;
+import com.google.gson.reflect.TypeToken;
 import com.mojang.authlib.Agent;
 import com.mojang.authlib.GameProfile;
 import com.mojang.authlib.GameProfileRepository;
 import com.mojang.authlib.ProfileLookupCallback;
 import java.io.BufferedReader;
@@ -22,11 +23,10 @@
 import java.io.File;
 import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.io.Reader;
 import java.io.Writer;
-import java.lang.reflect.ParameterizedType;
 import java.lang.reflect.Type;
 import java.nio.charset.StandardCharsets;
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
 import java.util.ArrayList;
@@ -50,22 +50,11 @@
    private final Map<UUID, GameProfileCache.GameProfileInfo> profilesByUUID = Maps.newHashMap();
    private final Deque<GameProfile> profileMRUList = Lists.newLinkedList();
    private final GameProfileRepository profileRepository;
    protected final Gson gson;
    private final File file;
-   private static final ParameterizedType GAMEPROFILE_ENTRY_TYPE = new ParameterizedType() {
-      public Type[] getActualTypeArguments() {
-         return new Type[]{GameProfileCache.GameProfileInfo.class};
-      }
-
-      public Type getRawType() {
-         return List.class;
-      }
-
-      public Type getOwnerType() {
-         return null;
-      }
+   private static final TypeToken<List<GameProfileCache.GameProfileInfo>> GAMEPROFILE_ENTRY_TYPE = new TypeToken<List<GameProfileCache.GameProfileInfo>>() {
    };
 
    public GameProfileCache(GameProfileRepository var1, File var2) {
       this.profileRepository = var1;
       this.file = var2;
@@ -178,11 +167,11 @@
    public void load() {
       BufferedReader var1 = null;
 
       try {
          var1 = Files.newReader(this.file, StandardCharsets.UTF_8);
-         List var2 = (List)GsonHelper.fromJson(this.gson, (Reader)var1, (Type)GAMEPROFILE_ENTRY_TYPE);
+         List var2 = (List)GsonHelper.fromJson(this.gson, (Reader)var1, (TypeToken)GAMEPROFILE_ENTRY_TYPE);
          this.profilesByName.clear();
          this.profilesByUUID.clear();
          this.profileMRUList.clear();
          if (var2 != null) {
             Iterator var3 = Lists.reverse(var2).iterator();
@@ -209,12 +198,12 @@
       try {
          var2 = Files.newWriter(this.file, StandardCharsets.UTF_8);
          var2.write(var1);
          return;
       } catch (FileNotFoundException var8) {
-         return;
       } catch (IOException var9) {
+         return;
       } finally {
          IOUtils.closeQuietly((Writer)var2);
       }
 
    }
