--- net/minecraft/world/level/storage/DerivedLevelData.java
+++ net/minecraft/world/level/storage/DerivedLevelData.java
@@ -1,34 +1,29 @@
 package net.minecraft.world.level.storage;
 
-import javax.annotation.Nullable;
+import java.util.UUID;
 import net.minecraft.CrashReportCategory;
 import net.minecraft.core.BlockPos;
-import net.minecraft.nbt.CompoundTag;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.world.Difficulty;
 import net.minecraft.world.level.GameRules;
 import net.minecraft.world.level.GameType;
-import net.minecraft.world.level.LevelType;
+import net.minecraft.world.level.border.WorldBorder;
 import net.minecraft.world.level.dimension.DimensionType;
 import net.minecraft.world.level.timers.TimerQueue;
 
-public class DerivedLevelData extends LevelData {
-   private final LevelData wrapped;
+public class DerivedLevelData implements ServerLevelData {
+   private final DimensionType dimensionType;
+   private final WorldData worldData;
+   private final ServerLevelData wrapped;
 
-   public DerivedLevelData(LevelData var1) {
-      this.wrapped = var1;
+   public DerivedLevelData(DimensionType var1, WorldData var2, ServerLevelData var3) {
+      this.dimensionType = var1;
+      this.worldData = var2;
+      this.wrapped = var3;
    }
 
-   public CompoundTag createTag(@Nullable CompoundTag var1) {
-      return this.wrapped.createTag(var1);
-   }
-
-   public long getSeed() {
-      return this.wrapped.getSeed();
-   }
-
    public int getXSpawn() {
       return this.wrapped.getXSpawn();
    }
 
    public int getYSpawn() {
@@ -45,20 +40,19 @@
 
    public long getDayTime() {
       return this.wrapped.getDayTime();
    }
 
-   public CompoundTag getLoadedPlayerTag() {
-      return this.wrapped.getLoadedPlayerTag();
+   public String getLevelName() {
+      return this.worldData.getLevelName();
    }
 
-   public String getLevelName() {
-      return this.wrapped.getLevelName();
+   public int getClearWeatherTime() {
+      return this.wrapped.getClearWeatherTime();
    }
 
-   public int getVersion() {
-      return this.wrapped.getVersion();
+   public void setClearWeatherTime(int var1) {
    }
 
    public boolean isThundering() {
       return this.wrapped.isThundering();
    }
@@ -74,28 +68,31 @@
    public int getRainTime() {
       return this.wrapped.getRainTime();
    }
 
    public GameType getGameType() {
-      return this.wrapped.getGameType();
+      return this.worldData.getGameType();
    }
 
-   public void setGameTime(long var1) {
+   public void setXSpawn(int var1) {
    }
 
-   public void setDayTime(long var1) {
+   public void setYSpawn(int var1) {
    }
 
-   public void setSpawn(BlockPos var1) {
+   public void setZSpawn(int var1) {
    }
 
-   public void setLevelName(String var1) {
+   public void setGameTime(long var1) {
    }
 
-   public void setVersion(int var1) {
+   public void setDayTime(long var1) {
    }
 
+   public void setSpawn(BlockPos var1) {
+   }
+
    public void setThundering(boolean var1) {
    }
 
    public void setThunderTime(int var1) {
    }
@@ -104,69 +101,68 @@
    }
 
    public void setRainTime(int var1) {
    }
 
-   public boolean isGenerateMapFeatures() {
-      return this.wrapped.isGenerateMapFeatures();
+   public void setGameType(GameType var1) {
    }
 
    public boolean isHardcore() {
-      return this.wrapped.isHardcore();
+      return this.worldData.isHardcore();
    }
 
-   public LevelType getGeneratorType() {
-      return this.wrapped.getGeneratorType();
-   }
-
-   public void setGenerator(LevelType var1) {
-   }
-
    public boolean getAllowCommands() {
-      return this.wrapped.getAllowCommands();
+      return this.worldData.getAllowCommands();
    }
 
-   public void setAllowCommands(boolean var1) {
-   }
-
    public boolean isInitialized() {
       return this.wrapped.isInitialized();
    }
 
    public void setInitialized(boolean var1) {
    }
 
    public GameRules getGameRules() {
-      return this.wrapped.getGameRules();
+      return this.worldData.getGameRules();
    }
 
-   public Difficulty getDifficulty() {
-      return this.wrapped.getDifficulty();
+   public WorldBorder.Settings getWorldBorder() {
+      return this.wrapped.getWorldBorder();
    }
 
-   public void setDifficulty(Difficulty var1) {
+   public void setWorldBorder(WorldBorder.Settings var1) {
    }
 
-   public boolean isDifficultyLocked() {
-      return this.wrapped.isDifficultyLocked();
+   public Difficulty getDifficulty() {
+      return this.worldData.getDifficulty();
    }
 
-   public void setDifficultyLocked(boolean var1) {
+   public boolean isDifficultyLocked() {
+      return this.worldData.isDifficultyLocked();
    }
 
    public TimerQueue<MinecraftServer> getScheduledEvents() {
       return this.wrapped.getScheduledEvents();
    }
 
-   public void setDimensionData(DimensionType var1, CompoundTag var2) {
-      this.wrapped.setDimensionData(var1, var2);
+   public int getWanderingTraderSpawnDelay() {
+      return 0;
    }
 
-   public CompoundTag getDimensionData(DimensionType var1) {
-      return this.wrapped.getDimensionData(var1);
+   public void setWanderingTraderSpawnDelay(int var1) {
    }
 
+   public int getWanderingTraderSpawnChance() {
+      return 0;
+   }
+
+   public void setWanderingTraderSpawnChance(int var1) {
+   }
+
+   public void setWanderingTraderId(UUID var1) {
+   }
+
    public void fillCrashReportCategory(CrashReportCategory var1) {
       var1.setDetail("Derived", (Object)true);
       this.wrapped.fillCrashReportCategory(var1);
    }
 }
