--- net/minecraft/client/multiplayer/ServerList.java
+++ net/minecraft/client/multiplayer/ServerList.java
@@ -2,10 +2,11 @@
 
 import com.google.common.collect.Lists;
 import java.io.File;
 import java.util.Iterator;
 import java.util.List;
+import net.minecraft.Util;
 import net.minecraft.client.Minecraft;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.nbt.ListTag;
 import net.minecraft.nbt.NbtIo;
 import org.apache.logging.log4j.LogManager;
@@ -48,15 +49,19 @@
          while(var2.hasNext()) {
             ServerData var3 = (ServerData)var2.next();
             var1.add(var3.write());
          }
 
-         CompoundTag var5 = new CompoundTag();
-         var5.put("servers", var1);
-         NbtIo.safeWrite(var5, new File(this.minecraft.gameDirectory, "servers.dat"));
-      } catch (Exception var4) {
-         LOGGER.error("Couldn't save server list", var4);
+         CompoundTag var7 = new CompoundTag();
+         var7.put("servers", var1);
+         File var8 = File.createTempFile("servers", ".dat", this.minecraft.gameDirectory);
+         NbtIo.write(var7, var8);
+         File var4 = new File(this.minecraft.gameDirectory, "servers.dat_old");
+         File var5 = new File(this.minecraft.gameDirectory, "servers.dat");
+         Util.safeReplaceFile(var5, var8, var4);
+      } catch (Exception var6) {
+         LOGGER.error("Couldn't save server list", var6);
       }
 
    }
 
    public ServerData get(int var1) {
