--- net/minecraft/client/multiplayer/ServerData.java
+++ net/minecraft/client/multiplayer/ServerData.java
@@ -1,24 +1,28 @@
 package net.minecraft.client.multiplayer;
 
+import java.util.Collections;
+import java.util.List;
 import javax.annotation.Nullable;
 import net.minecraft.SharedConstants;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
 import net.minecraft.network.chat.TranslatableComponent;
 
 public class ServerData {
    public String name;
    public String ip;
-   public String status;
-   public String motd;
+   public Component status;
+   public Component motd;
    public long ping;
    public int protocol = SharedConstants.getCurrentVersion().getProtocolVersion();
-   public String version = SharedConstants.getCurrentVersion().getName();
+   public Component version = new TextComponent(SharedConstants.getCurrentVersion().getName());
    public boolean pinged;
-   public String playerList;
+   public List<Component> playerList = Collections.emptyList();
    private ServerData.ServerPackStatus packStatus;
+   @Nullable
    private String iconB64;
    private boolean lan;
 
    public ServerData(String var1, String var2, boolean var3) {
       this.packStatus = ServerData.ServerPackStatus.PROMPT;
@@ -98,11 +102,11 @@
       PROMPT("prompt");
 
       private final Component name;
 
       private ServerPackStatus(String var3) {
-         this.name = new TranslatableComponent("addServer.resourcePack." + var3, new Object[0]);
+         this.name = new TranslatableComponent("addServer.resourcePack." + var3);
       }
 
       public Component getName() {
          return this.name;
       }
