--- net/minecraft/world/inventory/FurnaceResultSlot.java
+++ net/minecraft/world/inventory/FurnaceResultSlot.java
@@ -38,11 +38,11 @@
    }
 
    protected void checkTakeAchievements(ItemStack var1) {
       var1.onCraftedBy(this.player.level, this.player, this.removeCount);
       if (!this.player.level.isClientSide && this.container instanceof AbstractFurnaceBlockEntity) {
-         ((AbstractFurnaceBlockEntity)this.container).awardResetAndExperience(this.player);
+         ((AbstractFurnaceBlockEntity)this.container).awardUsedRecipesAndPopExperience(this.player);
       }
 
       this.removeCount = 0;
    }
 }
