--- net/minecraft/world/inventory/HorseInventoryMenu.java
+++ net/minecraft/world/inventory/HorseInventoryMenu.java
@@ -19,24 +19,24 @@
       boolean var5 = true;
       var3.startOpen(var2.player);
       boolean var6 = true;
       this.addSlot(new Slot(var3, 0, 8, 18) {
          public boolean mayPlace(ItemStack var1) {
-            return var1.getItem() == Items.SADDLE && !this.hasItem() && var4.canBeSaddled();
+            return var1.getItem() == Items.SADDLE && !this.hasItem() && var4.isSaddleable();
          }
 
          public boolean isActive() {
-            return var4.canBeSaddled();
+            return var4.isSaddleable();
          }
       });
       this.addSlot(new Slot(var3, 1, 8, 36) {
          public boolean mayPlace(ItemStack var1) {
             return var4.isArmor(var1);
          }
 
          public boolean isActive() {
-            return var4.wearsArmor();
+            return var4.canWearArmor();
          }
 
          public int getMaxStackSize() {
             return 1;
          }
