--- net/minecraft/world/inventory/StonecutterMenu.java
+++ net/minecraft/world/inventory/StonecutterMenu.java
@@ -108,18 +108,22 @@
    public boolean stillValid(Player var1) {
       return stillValid(this.access, var1, Blocks.STONECUTTER);
    }
 
    public boolean clickMenuButton(Player var1, int var2) {
-      if (var2 >= 0 && var2 < this.recipes.size()) {
+      if (this.isValidRecipeIndex(var2)) {
          this.selectedRecipeIndex.set(var2);
          this.setupResultSlot();
       }
 
       return true;
    }
 
+   private boolean isValidRecipeIndex(int var1) {
+      return var1 >= 0 && var1 < this.recipes.size();
+   }
+
    public void slotsChanged(Container var1) {
       ItemStack var2 = this.inputSlot.getItem();
       if (var2.getItem() != this.input.getItem()) {
          this.input = var2.copy();
          this.setupRecipeList(var1, var2);
@@ -136,11 +140,11 @@
       }
 
    }
 
    private void setupResultSlot() {
-      if (!this.recipes.isEmpty()) {
+      if (!this.recipes.isEmpty() && this.isValidRecipeIndex(this.selectedRecipeIndex.get())) {
          StonecutterRecipe var1 = (StonecutterRecipe)this.recipes.get(this.selectedRecipeIndex.get());
          this.resultSlot.set(var1.assemble(this.container));
       } else {
          this.resultSlot.set(ItemStack.EMPTY);
       }
