--- net/minecraft/stats/ServerRecipeBook.java
+++ net/minecraft/stats/ServerRecipeBook.java
@@ -21,14 +21,12 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class ServerRecipeBook extends RecipeBook {
    private static final Logger LOGGER = LogManager.getLogger();
-   private final RecipeManager manager;
 
-   public ServerRecipeBook(RecipeManager var1) {
-      this.manager = var1;
+   public ServerRecipeBook() {
    }
 
    public int addRecipes(Collection<Recipe<?>> var1, ServerPlayer var2) {
       ArrayList var3 = Lists.newArrayList();
       int var4 = 0;
@@ -68,19 +66,16 @@
       this.sendRecipes(ClientboundRecipePacket.State.REMOVE, var2, var3);
       return var4;
    }
 
    private void sendRecipes(ClientboundRecipePacket.State var1, ServerPlayer var2, List<ResourceLocation> var3) {
-      var2.connection.send(new ClientboundRecipePacket(var1, var3, Collections.emptyList(), this.guiOpen, this.filteringCraftable, this.furnaceGuiOpen, this.furnaceFilteringCraftable));
+      var2.connection.send(new ClientboundRecipePacket(var1, var3, Collections.emptyList(), this.getBookSettings()));
    }
 
    public CompoundTag toNbt() {
       CompoundTag var1 = new CompoundTag();
-      var1.putBoolean("isGuiOpen", this.guiOpen);
-      var1.putBoolean("isFilteringCraftable", this.filteringCraftable);
-      var1.putBoolean("isFurnaceGuiOpen", this.furnaceGuiOpen);
-      var1.putBoolean("isFurnaceFilteringCraftable", this.furnaceFilteringCraftable);
+      this.getBookSettings().write(var1);
       ListTag var2 = new ListTag();
       Iterator var3 = this.known.iterator();
 
       while(var3.hasNext()) {
          ResourceLocation var4 = (ResourceLocation)var3.next();
@@ -98,39 +93,36 @@
 
       var1.put("toBeDisplayed", var6);
       return var1;
    }
 
-   public void fromNbt(CompoundTag var1) {
-      this.guiOpen = var1.getBoolean("isGuiOpen");
-      this.filteringCraftable = var1.getBoolean("isFilteringCraftable");
-      this.furnaceGuiOpen = var1.getBoolean("isFurnaceGuiOpen");
-      this.furnaceFilteringCraftable = var1.getBoolean("isFurnaceFilteringCraftable");
-      ListTag var2 = var1.getList("recipes", 8);
-      this.loadRecipes(var2, this::add);
-      ListTag var3 = var1.getList("toBeDisplayed", 8);
-      this.loadRecipes(var3, this::addHighlight);
+   public void fromNbt(CompoundTag var1, RecipeManager var2) {
+      this.setBookSettings(RecipeBookSettings.read(var1));
+      ListTag var3 = var1.getList("recipes", 8);
+      this.loadRecipes(var3, this::add, var2);
+      ListTag var4 = var1.getList("toBeDisplayed", 8);
+      this.loadRecipes(var4, this::addHighlight, var2);
    }
 
-   private void loadRecipes(ListTag var1, Consumer<Recipe<?>> var2) {
-      for(int var3 = 0; var3 < var1.size(); ++var3) {
-         String var4 = var1.getString(var3);
+   private void loadRecipes(ListTag var1, Consumer<Recipe<?>> var2, RecipeManager var3) {
+      for(int var4 = 0; var4 < var1.size(); ++var4) {
+         String var5 = var1.getString(var4);
 
          try {
-            ResourceLocation var5 = new ResourceLocation(var4);
-            Optional var6 = this.manager.byKey(var5);
-            if (!var6.isPresent()) {
-               LOGGER.error((String)"Tried to load unrecognized recipe: {} removed now.", (Object)var5);
+            ResourceLocation var6 = new ResourceLocation(var5);
+            Optional var7 = var3.byKey(var6);
+            if (!var7.isPresent()) {
+               LOGGER.error((String)"Tried to load unrecognized recipe: {} removed now.", (Object)var6);
             } else {
-               var2.accept(var6.get());
+               var2.accept(var7.get());
             }
-         } catch (ResourceLocationException var7) {
-            LOGGER.error((String)"Tried to load improperly formatted recipe: {} removed now.", (Object)var4);
+         } catch (ResourceLocationException var8) {
+            LOGGER.error((String)"Tried to load improperly formatted recipe: {} removed now.", (Object)var5);
          }
       }
 
    }
 
    public void sendInitialRecipeBook(ServerPlayer var1) {
-      var1.connection.send(new ClientboundRecipePacket(ClientboundRecipePacket.State.INIT, this.known, this.highlight, this.guiOpen, this.filteringCraftable, this.furnaceGuiOpen, this.furnaceFilteringCraftable));
+      var1.connection.send(new ClientboundRecipePacket(ClientboundRecipePacket.State.INIT, this.known, this.highlight, this.getBookSettings()));
    }
 }
