--- net/minecraft/util/Mth.java
+++ net/minecraft/util/Mth.java
@@ -208,11 +208,11 @@
       var1 |= var1 >> 8;
       var1 |= var1 >> 16;
       return var1 + 1;
    }
 
-   private static boolean isPowerOfTwo(int var0) {
+   public static boolean isPowerOfTwo(int var0) {
       return var0 != 0 && (var0 & var0 - 1) == 0;
    }
 
    public static int ceillog2(int var0) {
       var0 = isPowerOfTwo(var0) ? var0 : smallestEncompassingPowerOfTwo(var0);
@@ -236,10 +236,14 @@
          int var2 = var0 % var1;
          return var2 == 0 ? var0 : var0 + var1 - var2;
       }
    }
 
+   public static float frac(float var0) {
+      return var0 - (float)floor(var0);
+   }
+
    public static double frac(double var0) {
       return var0 - (double)lfloor(var0);
    }
 
    public static long getSeed(Vec3i var0) {
@@ -260,11 +264,11 @@
 
    public static UUID createInsecureUUID() {
       return createInsecureUUID(RANDOM);
    }
 
-   public static double pct(double var0, double var2, double var4) {
+   public static double inverseLerp(double var0, double var2, double var4) {
       return (var0 - var2) / (var4 - var2);
    }
 
    public static double atan2(double var0, double var2) {
       double var4 = var2 * var2 + var0 * var0;
@@ -440,10 +444,14 @@
       }
 
       return var0 + var2 * var3;
    }
 
+   public static float square(float var0) {
+      return var0 * var0;
+   }
+
    static {
       for(int var0 = 0; var0 < 257; ++var0) {
          double var1 = (double)var0 / 256.0D;
          double var3 = Math.asin(var1);
          COS_TAB[var0] = Math.cos(var3);
