--- net/minecraft/client/gui/chat/OverlayChatListener.java
+++ net/minecraft/client/gui/chat/OverlayChatListener.java
@@ -1,7 +1,8 @@
 package net.minecraft.client.gui.chat;
 
+import java.util.UUID;
 import net.minecraft.client.Minecraft;
 import net.minecraft.network.chat.ChatType;
 import net.minecraft.network.chat.Component;
 
 public class OverlayChatListener implements ChatListener {
@@ -9,9 +10,11 @@
 
    public OverlayChatListener(Minecraft var1) {
       this.minecraft = var1;
    }
 
-   public void handle(ChatType var1, Component var2) {
-      this.minecraft.gui.setOverlayMessage(var2, false);
+   public void handle(ChatType var1, Component var2, UUID var3) {
+      if (!this.minecraft.isBlocked(var3)) {
+         this.minecraft.gui.setOverlayMessage(var2, false);
+      }
    }
 }
