--- net/minecraft/client/gui/chat/StandardChatListener.java
+++ net/minecraft/client/gui/chat/StandardChatListener.java
@@ -1,7 +1,8 @@
 package net.minecraft.client.gui.chat;
 
+import java.util.UUID;
 import net.minecraft.client.Minecraft;
 import net.minecraft.network.chat.ChatType;
 import net.minecraft.network.chat.Component;
 
 public class StandardChatListener implements ChatListener {
@@ -9,9 +10,16 @@
 
    public StandardChatListener(Minecraft var1) {
       this.minecraft = var1;
    }
 
-   public void handle(ChatType var1, Component var2) {
-      this.minecraft.gui.getChat().addMessage(var2);
+   public void handle(ChatType var1, Component var2, UUID var3) {
+      if (!this.minecraft.isBlocked(var3)) {
+         if (var1 != ChatType.CHAT) {
+            this.minecraft.gui.getChat().addMessage(var2);
+         } else {
+            this.minecraft.gui.getChat().enqueueMessage(var2);
+         }
+
+      }
    }
 }
