--- net/minecraft/world/scores/Objective.java
+++ net/minecraft/world/scores/Objective.java
@@ -9,17 +9,19 @@
 public class Objective {
    private final Scoreboard scoreboard;
    private final String name;
    private final ObjectiveCriteria criteria;
    private Component displayName;
+   private Component formattedDisplayName;
    private ObjectiveCriteria.RenderType renderType;
 
    public Objective(Scoreboard var1, String var2, ObjectiveCriteria var3, Component var4, ObjectiveCriteria.RenderType var5) {
       this.scoreboard = var1;
       this.name = var2;
       this.criteria = var3;
       this.displayName = var4;
+      this.formattedDisplayName = this.createFormattedDisplayName();
       this.renderType = var5;
    }
 
    public Scoreboard getScoreboard() {
       return this.scoreboard;
@@ -35,18 +37,23 @@
 
    public Component getDisplayName() {
       return this.displayName;
    }
 
-   public Component getFormattedDisplayName() {
-      return ComponentUtils.wrapInSquareBrackets(this.displayName.deepCopy().withStyle((var1) -> {
-         var1.setHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, new TextComponent(this.getName())));
+   private Component createFormattedDisplayName() {
+      return ComponentUtils.wrapInSquareBrackets(this.displayName.mutableCopy().withStyle((var1) -> {
+         return var1.withHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, new TextComponent(this.name)));
       }));
    }
 
+   public Component getFormattedDisplayName() {
+      return this.formattedDisplayName;
+   }
+
    public void setDisplayName(Component var1) {
       this.displayName = var1;
+      this.formattedDisplayName = this.createFormattedDisplayName();
       this.scoreboard.onObjectiveChanged(this);
    }
 
    public ObjectiveCriteria.RenderType getRenderType() {
       return this.renderType;
