--- net/minecraft/world/level/dimension/end/DragonRespawnAnimation.java
+++ net/minecraft/world/level/dimension/end/DragonRespawnAnimation.java
@@ -65,11 +65,11 @@
                      var1.removeBlock(var14, false);
                   }
 
                   var1.explode((Entity)null, (double)((float)var11.getCenterX() + 0.5F), (double)var11.getHeight(), (double)((float)var11.getCenterZ() + 0.5F), 5.0F, Explosion.BlockInteraction.DESTROY);
                   SpikeConfiguration var17 = new SpikeConfiguration(true, ImmutableList.of(var11), new BlockPos(0, 128, 0));
-                  Feature.END_SPIKE.configured(var17).place(var1, var1.getChunkSource().getGenerator(), new Random(), new BlockPos(var11.getCenterX(), 45, var11.getCenterZ()));
+                  Feature.END_SPIKE.configured(var17).place(var1, var1.structureFeatureManager(), var1.getChunkSource().getGenerator(), new Random(), new BlockPos(var11.getCenterX(), 45, var11.getCenterZ()));
                }
             } else if (var7) {
                var2.setRespawnStage(SUMMONING_DRAGON);
             }
          }
