--- net/minecraft/DetectedVersion.java
+++ net/minecraft/DetectedVersion.java
@@ -14,31 +14,32 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class DetectedVersion implements GameVersion {
    private static final Logger LOGGER = LogManager.getLogger();
+   public static final GameVersion BUILT_IN = new DetectedVersion();
    private final String id;
    private final String name;
    private final boolean stable;
    private final int worldVersion;
    private final int protocolVersion;
    private final int packVersion;
    private final Date buildTime;
    private final String releaseTarget;
 
-   public DetectedVersion() {
+   private DetectedVersion() {
       this.id = UUID.randomUUID().toString().replaceAll("-", "");
-      this.name = "1.15.2";
-      this.stable = true;
-      this.worldVersion = 2230;
-      this.protocolVersion = 578;
+      this.name = "1.16-pre7";
+      this.stable = false;
+      this.worldVersion = 2563;
+      this.protocolVersion = 732;
       this.packVersion = 5;
       this.buildTime = new Date();
-      this.releaseTarget = "1.15.2";
+      this.releaseTarget = "1.16";
    }
 
-   protected DetectedVersion(JsonObject var1) {
+   private DetectedVersion(JsonObject var1) {
       this.id = GsonHelper.getAsString(var1, "id");
       this.name = GsonHelper.getAsString(var1, "name");
       this.releaseTarget = GsonHelper.getAsString(var1, "release_target");
       this.stable = GsonHelper.getAsBoolean(var1, "stable");
       this.worldVersion = GsonHelper.getAsInt(var1, "world_version");
@@ -54,11 +55,11 @@
 
          Object var4;
          try {
             if (var0 == null) {
                LOGGER.warn("Missing version information!");
-               DetectedVersion var35 = new DetectedVersion();
+               GameVersion var35 = BUILT_IN;
                return var35;
             }
 
             InputStreamReader var2 = new InputStreamReader(var0);
             Throwable var3 = null;
