--- com/mojang/realmsclient/dto/RealmsDescriptionDto.java
+++ com/mojang/realmsclient/dto/RealmsDescriptionDto.java
@@ -1,9 +1,13 @@
 package com.mojang.realmsclient.dto;
 
-public class RealmsDescriptionDto extends ValueObject {
+import com.google.gson.annotations.SerializedName;
+
+public class RealmsDescriptionDto extends ValueObject implements ReflectionBasedSerialization {
+   @SerializedName("name")
    public String name;
+   @SerializedName("description")
    public String description;
 
    public RealmsDescriptionDto(String var1, String var2) {
       this.name = var1;
       this.description = var2;
