--- com/mojang/realmsclient/dto/RegionPingResult.java
+++ com/mojang/realmsclient/dto/RegionPingResult.java
@@ -1,11 +1,14 @@
 package com.mojang.realmsclient.dto;
 
+import com.google.gson.annotations.SerializedName;
 import java.util.Locale;
 
-public class RegionPingResult extends ValueObject {
+public class RegionPingResult extends ValueObject implements ReflectionBasedSerialization {
+   @SerializedName("regionName")
    private final String regionName;
+   @SerializedName("ping")
    private final int ping;
 
    public RegionPingResult(String var1, int var2) {
       this.regionName = var1;
       this.ping = var2;
