--- com/mojang/realmsclient/dto/WorldTemplate.java
+++ com/mojang/realmsclient/dto/WorldTemplate.java
@@ -1,25 +1,28 @@
 package com.mojang.realmsclient.dto;
 
 import com.google.gson.JsonObject;
 import com.mojang.realmsclient.util.JsonUtils;
+import javax.annotation.Nullable;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class WorldTemplate extends ValueObject {
    private static final Logger LOGGER = LogManager.getLogger();
-   public String id;
-   public String name;
-   public String version;
-   public String author;
-   public String link;
+   public String id = "";
+   public String name = "";
+   public String version = "";
+   public String author = "";
+   public String link = "";
+   @Nullable
    public String image;
-   public String trailer;
-   public String recommendedPlayers;
+   public String trailer = "";
+   public String recommendedPlayers = "";
    public WorldTemplate.WorldTemplateType type;
 
    public WorldTemplate() {
+      this.type = WorldTemplate.WorldTemplateType.WORLD_TEMPLATE;
    }
 
    public static WorldTemplate parse(JsonObject var0) {
       WorldTemplate var1 = new WorldTemplate();
 
