--- net/minecraft/client/gui/screens/MenuScreens.java
+++ net/minecraft/client/gui/screens/MenuScreens.java
@@ -20,10 +20,11 @@
 import net.minecraft.client.gui.screens.inventory.LecternScreen;
 import net.minecraft.client.gui.screens.inventory.LoomScreen;
 import net.minecraft.client.gui.screens.inventory.MenuAccess;
 import net.minecraft.client.gui.screens.inventory.MerchantScreen;
 import net.minecraft.client.gui.screens.inventory.ShulkerBoxScreen;
+import net.minecraft.client.gui.screens.inventory.SmithingScreen;
 import net.minecraft.client.gui.screens.inventory.SmokerScreen;
 import net.minecraft.client.gui.screens.inventory.StonecutterScreen;
 import net.minecraft.core.Registry;
 import net.minecraft.network.chat.Component;
 import net.minecraft.world.entity.player.Inventory;
@@ -95,10 +96,11 @@
       register(MenuType.HOPPER, HopperScreen::new);
       register(MenuType.LECTERN, LecternScreen::new);
       register(MenuType.LOOM, LoomScreen::new);
       register(MenuType.MERCHANT, MerchantScreen::new);
       register(MenuType.SHULKER_BOX, ShulkerBoxScreen::new);
+      register(MenuType.SMITHING, SmithingScreen::new);
       register(MenuType.SMOKER, SmokerScreen::new);
       register(MenuType.CARTOGRAPHY_TABLE, CartographyTableScreen::new);
       register(MenuType.STONECUTTER, StonecutterScreen::new);
    }
 
