--- net/minecraft/client/gui/screens/ProgressScreen.java
+++ net/minecraft/client/gui/screens/ProgressScreen.java
@@ -1,16 +1,20 @@
 package net.minecraft.client.gui.screens;
 
-import java.util.Objects;
+import com.mojang.blaze3d.vertex.PoseStack;
+import javax.annotation.Nullable;
 import net.minecraft.client.gui.chat.NarratorChatListener;
 import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
 import net.minecraft.network.chat.TranslatableComponent;
 import net.minecraft.util.ProgressListener;
 
 public class ProgressScreen extends Screen implements ProgressListener {
-   private String title = "";
-   private String stage = "";
+   @Nullable
+   private Component header;
+   @Nullable
+   private Component stage;
    private int progress;
    private boolean stop;
 
    public ProgressScreen() {
       super(NarratorChatListener.NO_TITLE);
@@ -23,16 +27,16 @@
    public void progressStartNoAbort(Component var1) {
       this.progressStart(var1);
    }
 
    public void progressStart(Component var1) {
-      this.title = var1.getColoredString();
-      this.progressStage(new TranslatableComponent("progress.working", new Object[0]));
+      this.header = var1;
+      this.progressStage(new TranslatableComponent("progress.working"));
    }
 
    public void progressStage(Component var1) {
-      this.stage = var1.getColoredString();
+      this.stage = var1;
       this.progressStagePercentage(0);
    }
 
    public void progressStagePercentage(int var1) {
       this.progress = var1;
@@ -40,22 +44,25 @@
 
    public void stop() {
       this.stop = true;
    }
 
-   public void render(int var1, int var2, float var3) {
+   public void render(PoseStack var1, int var2, int var3, float var4) {
       if (this.stop) {
          if (!this.minecraft.isConnectedToRealms()) {
             this.minecraft.setScreen((Screen)null);
          }
 
       } else {
-         this.renderBackground();
-         this.drawCenteredString(this.font, this.title, this.width / 2, 70, 16777215);
-         if (!Objects.equals(this.stage, "") && this.progress != 0) {
-            this.drawCenteredString(this.font, this.stage + " " + this.progress + "%", this.width / 2, 90, 16777215);
+         this.renderBackground(var1);
+         if (this.header != null) {
+            this.drawCenteredString(var1, this.font, this.header, this.width / 2, 70, 16777215);
          }
 
-         super.render(var1, var2, var3);
+         if (this.stage != null && this.progress != 0) {
+            this.drawCenteredString(var1, this.font, (new TextComponent("")).append(this.stage).append(" " + this.progress + "%"), this.width / 2, 90, 16777215);
+         }
+
+         super.render(var1, var2, var3, var4);
       }
    }
 }
