--- net/minecraft/nbt/ByteArrayTag.java
+++ net/minecraft/nbt/ByteArrayTag.java
@@ -4,10 +4,11 @@
 import java.io.DataOutput;
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.List;
 import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
 import net.minecraft.network.chat.TextComponent;
 import org.apache.commons.lang3.ArrayUtils;
 
 public class ByteArrayTag extends CollectionTag<ByteTag> {
    public static final TagType<ByteArrayTag> TYPE = new TagType<ByteArrayTag>() {
@@ -93,16 +94,16 @@
    public int hashCode() {
       return Arrays.hashCode(this.data);
    }
 
    public Component getPrettyDisplay(String var1, int var2) {
-      Component var3 = (new TextComponent("B")).withStyle(SYNTAX_HIGHLIGHTING_NUMBER_TYPE);
-      Component var4 = (new TextComponent("[")).append(var3).append(";");
+      MutableComponent var3 = (new TextComponent("B")).withStyle(SYNTAX_HIGHLIGHTING_NUMBER_TYPE);
+      MutableComponent var4 = (new TextComponent("[")).append(var3).append(";");
 
       for(int var5 = 0; var5 < this.data.length; ++var5) {
-         Component var6 = (new TextComponent(String.valueOf(this.data[var5]))).withStyle(SYNTAX_HIGHLIGHTING_NUMBER);
-         var4.append(" ").append(var6).append(var3);
+         MutableComponent var6 = (new TextComponent(String.valueOf(this.data[var5]))).withStyle(SYNTAX_HIGHLIGHTING_NUMBER);
+         var4.append(" ").append((Component)var6).append((Component)var3);
          if (var5 != this.data.length - 1) {
             var4.append(",");
          }
       }
 
@@ -154,9 +155,13 @@
       byte var2 = this.data[var1];
       this.data = ArrayUtils.remove(this.data, var1);
       return ByteTag.valueOf(var2);
    }
 
+   public byte getElementType() {
+      return 1;
+   }
+
    public void clear() {
       this.data = new byte[0];
    }
 }
