--- com/mojang/realmsclient/gui/screens/RealmsInviteScreen.java
+++ com/mojang/realmsclient/gui/screens/RealmsInviteScreen.java
@@ -1,103 +1,99 @@
 package com.mojang.realmsclient.gui.screens;
 
+import com.mojang.blaze3d.vertex.PoseStack;
 import com.mojang.realmsclient.client.RealmsClient;
 import com.mojang.realmsclient.dto.RealmsServer;
-import com.mojang.realmsclient.gui.RealmsConstants;
-import net.minecraft.realms.Realms;
-import net.minecraft.realms.RealmsButton;
-import net.minecraft.realms.RealmsEditBox;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.components.EditBox;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.resources.language.I18n;
+import net.minecraft.network.chat.CommonComponents;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.realms.NarrationHelper;
 import net.minecraft.realms.RealmsScreen;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class RealmsInviteScreen extends RealmsScreen {
    private static final Logger LOGGER = LogManager.getLogger();
-   private RealmsEditBox profileName;
+   private EditBox profileName;
    private final RealmsServer serverData;
    private final RealmsConfigureWorldScreen configureScreen;
-   private final RealmsScreen lastScreen;
-   private final int BUTTON_INVITE_ID = 0;
-   private final int BUTTON_CANCEL_ID = 1;
-   private RealmsButton inviteButton;
-   private final int PROFILENAME_EDIT_BOX = 2;
+   private final Screen lastScreen;
    private String errorMsg;
    private boolean showError;
 
-   public RealmsInviteScreen(RealmsConfigureWorldScreen var1, RealmsScreen var2, RealmsServer var3) {
+   public RealmsInviteScreen(RealmsConfigureWorldScreen var1, Screen var2, RealmsServer var3) {
       this.configureScreen = var1;
       this.lastScreen = var2;
       this.serverData = var3;
    }
 
    public void tick() {
       this.profileName.tick();
    }
 
    public void init() {
-      this.setKeyboardHandlerSendRepeatsToGui(true);
-      this.buttonsAdd(this.inviteButton = new RealmsButton(0, this.width() / 2 - 100, RealmsConstants.row(10), getLocalizedString("mco.configure.world.buttons.invite")) {
-         public void onPress() {
-            RealmsInviteScreen.this.onInvite();
-         }
-      });
-      this.buttonsAdd(new RealmsButton(1, this.width() / 2 - 100, RealmsConstants.row(12), getLocalizedString("gui.cancel")) {
-         public void onPress() {
-            Realms.setScreen(RealmsInviteScreen.this.lastScreen);
-         }
-      });
-      this.profileName = this.newEditBox(2, this.width() / 2 - 100, RealmsConstants.row(2), 200, 20, getLocalizedString("mco.configure.world.invite.profile.name"));
-      this.focusOn(this.profileName);
+      this.minecraft.keyboardHandler.setSendRepeatsToGui(true);
+      this.profileName = new EditBox(this.minecraft.font, this.width / 2 - 100, row(2), 200, 20, (EditBox)null, new TranslatableComponent("mco.configure.world.invite.profile.name"));
       this.addWidget(this.profileName);
+      this.setInitialFocus(this.profileName);
+      this.addButton(new Button(this.width / 2 - 100, row(10), 200, 20, new TranslatableComponent("mco.configure.world.buttons.invite"), (var1) -> {
+         this.onInvite();
+      }));
+      this.addButton(new Button(this.width / 2 - 100, row(12), 200, 20, CommonComponents.GUI_CANCEL, (var1) -> {
+         this.minecraft.setScreen(this.lastScreen);
+      }));
    }
 
    public void removed() {
-      this.setKeyboardHandlerSendRepeatsToGui(false);
+      this.minecraft.keyboardHandler.setSendRepeatsToGui(false);
    }
 
    private void onInvite() {
-      RealmsClient var1 = RealmsClient.createRealmsClient();
+      RealmsClient var1 = RealmsClient.create();
       if (this.profileName.getValue() != null && !this.profileName.getValue().isEmpty()) {
          try {
             RealmsServer var2 = var1.invite(this.serverData.id, this.profileName.getValue().trim());
             if (var2 != null) {
                this.serverData.players = var2.players;
-               Realms.setScreen(new RealmsPlayerScreen(this.configureScreen, this.serverData));
+               this.minecraft.setScreen(new RealmsPlayerScreen(this.configureScreen, this.serverData));
             } else {
-               this.showError(getLocalizedString("mco.configure.world.players.error"));
+               this.showError(I18n.get("mco.configure.world.players.error"));
             }
          } catch (Exception var3) {
             LOGGER.error("Couldn't invite user");
-            this.showError(getLocalizedString("mco.configure.world.players.error"));
+            this.showError(I18n.get("mco.configure.world.players.error"));
          }
 
       } else {
-         this.showError(getLocalizedString("mco.configure.world.players.error"));
+         this.showError(I18n.get("mco.configure.world.players.error"));
       }
    }
 
    private void showError(String var1) {
       this.showError = true;
       this.errorMsg = var1;
-      Realms.narrateNow(var1);
+      NarrationHelper.now(var1);
    }
 
    public boolean keyPressed(int var1, int var2, int var3) {
       if (var1 == 256) {
-         Realms.setScreen(this.lastScreen);
+         this.minecraft.setScreen(this.lastScreen);
          return true;
       } else {
          return super.keyPressed(var1, var2, var3);
       }
    }
 
-   public void render(int var1, int var2, float var3) {
-      this.renderBackground();
-      this.drawString(getLocalizedString("mco.configure.world.invite.profile.name"), this.width() / 2 - 100, RealmsConstants.row(1), 10526880);
+   public void render(PoseStack var1, int var2, int var3, float var4) {
+      this.renderBackground(var1);
+      this.font.draw(var1, I18n.get("mco.configure.world.invite.profile.name"), (float)(this.width / 2 - 100), (float)row(1), 10526880);
       if (this.showError) {
-         this.drawCenteredString(this.errorMsg, this.width() / 2, RealmsConstants.row(5), 16711680);
+         this.drawCenteredString(var1, this.font, this.errorMsg, this.width / 2, row(5), 16711680);
       }
 
-      this.profileName.render(var1, var2, var3);
-      super.render(var1, var2, var3);
+      this.profileName.render(var1, var2, var3, var4);
+      super.render(var1, var2, var3, var4);
    }
 }
