--- net/minecraft/world/level/storage/loot/functions/SetItemCountFunction.java
+++ net/minecraft/world/level/storage/loot/functions/SetItemCountFunction.java
@@ -1,11 +1,10 @@
 package net.minecraft.world.level.storage.loot.functions;
 
 import com.google.gson.JsonDeserializationContext;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonSerializationContext;
-import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.storage.loot.LootContext;
 import net.minecraft.world.level.storage.loot.RandomIntGenerator;
 import net.minecraft.world.level.storage.loot.RandomIntGenerators;
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
@@ -16,10 +15,14 @@
    private SetItemCountFunction(LootItemCondition[] var1, RandomIntGenerator var2) {
       super(var1);
       this.value = var2;
    }
 
+   public LootItemFunctionType getType() {
+      return LootItemFunctions.SET_COUNT;
+   }
+
    public ItemStack run(ItemStack var1, LootContext var2) {
       var1.setCount(this.value.getInt(var2.getRandom()));
       return var1;
    }
 
@@ -28,12 +31,11 @@
          return new SetItemCountFunction(var1, var0);
       });
    }
 
    public static class Serializer extends LootItemConditionalFunction.Serializer<SetItemCountFunction> {
-      protected Serializer() {
-         super(new ResourceLocation("set_count"), SetItemCountFunction.class);
+      public Serializer() {
       }
 
       public void serialize(JsonObject var1, SetItemCountFunction var2, JsonSerializationContext var3) {
          super.serialize(var1, (LootItemConditionalFunction)var2, var3);
          var1.add("count", RandomIntGenerators.serialize(var2.value, var3));
