--- net/minecraft/world/level/storage/loot/functions/SetLoreFunction.java
+++ net/minecraft/world/level/storage/loot/functions/SetLoreFunction.java
@@ -15,11 +15,10 @@
 import javax.annotation.Nullable;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.nbt.ListTag;
 import net.minecraft.nbt.StringTag;
 import net.minecraft.network.chat.Component;
-import net.minecraft.resources.ResourceLocation;
 import net.minecraft.util.GsonHelper;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.storage.loot.LootContext;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParam;
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
@@ -35,10 +34,14 @@
       this.replace = var2;
       this.lore = ImmutableList.copyOf((Collection)var3);
       this.resolutionContext = var4;
    }
 
+   public LootItemFunctionType getType() {
+      return LootItemFunctions.SET_LORE;
+   }
+
    public Set<LootContextParam<?>> getReferencedContextParams() {
       return this.resolutionContext != null ? ImmutableSet.of(this.resolutionContext.getParam()) : ImmutableSet.of();
    }
 
    public ItemStack run(ItemStack var1, LootContext var2) {
@@ -92,11 +95,10 @@
       }
    }
 
    public static class Serializer extends LootItemConditionalFunction.Serializer<SetLoreFunction> {
       public Serializer() {
-         super(new ResourceLocation("set_lore"), SetLoreFunction.class);
       }
 
       public void serialize(JsonObject var1, SetLoreFunction var2, JsonSerializationContext var3) {
          super.serialize(var1, (LootItemConditionalFunction)var2, var3);
          var1.addProperty("replace", var2.replace);
