--- net/minecraft/world/level/storage/loot/functions/SetNameFunction.java
+++ net/minecraft/world/level/storage/loot/functions/SetNameFunction.java
@@ -9,11 +9,11 @@
 import java.util.function.UnaryOperator;
 import javax.annotation.Nullable;
 import net.minecraft.commands.CommandSourceStack;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.ComponentUtils;
-import net.minecraft.resources.ResourceLocation;
+import net.minecraft.network.chat.MutableComponent;
 import net.minecraft.util.GsonHelper;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.storage.loot.LootContext;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParam;
@@ -31,10 +31,14 @@
       super(var1);
       this.name = var2;
       this.resolutionContext = var3;
    }
 
+   public LootItemFunctionType getType() {
+      return LootItemFunctions.SET_NAME;
+   }
+
    public Set<LootContextParam<?>> getReferencedContextParams() {
       return this.resolutionContext != null ? ImmutableSet.of(this.resolutionContext.getParam()) : ImmutableSet.of();
    }
 
    public static UnaryOperator<Component> createResolver(LootContext var0, @Nullable LootContext.EntityTarget var1) {
@@ -66,11 +70,10 @@
       return var1;
    }
 
    public static class Serializer extends LootItemConditionalFunction.Serializer<SetNameFunction> {
       public Serializer() {
-         super(new ResourceLocation("set_name"), SetNameFunction.class);
       }
 
       public void serialize(JsonObject var1, SetNameFunction var2, JsonSerializationContext var3) {
          super.serialize(var1, (LootItemConditionalFunction)var2, var3);
          if (var2.name != null) {
@@ -82,11 +85,11 @@
          }
 
       }
 
       public SetNameFunction deserialize(JsonObject var1, JsonDeserializationContext var2, LootItemCondition[] var3) {
-         Component var4 = Component.Serializer.fromJson(var1.get("name"));
+         MutableComponent var4 = Component.Serializer.fromJson(var1.get("name"));
          LootContext.EntityTarget var5 = (LootContext.EntityTarget)GsonHelper.getAsObject(var1, "entity", (Object)null, var2, LootContext.EntityTarget.class);
          return new SetNameFunction(var3, var4, var5);
       }
    }
 }
