--- net/minecraft/world/entity/ai/goal/LandOnOwnersShoulderGoal.java
+++ net/minecraft/world/entity/ai/goal/LandOnOwnersShoulderGoal.java
@@ -13,11 +13,11 @@
    }
 
    public boolean canUse() {
       ServerPlayer var1 = (ServerPlayer)this.entity.getOwner();
       boolean var2 = var1 != null && !var1.isSpectator() && !var1.abilities.flying && !var1.isInWater();
-      return !this.entity.isSitting() && var2 && this.entity.canSitOnShoulder();
+      return !this.entity.isOrderedToSit() && var2 && this.entity.canSitOnShoulder();
    }
 
    public boolean isInterruptable() {
       return !this.isSittingOnShoulder;
    }
@@ -26,11 +26,11 @@
       this.owner = (ServerPlayer)this.entity.getOwner();
       this.isSittingOnShoulder = false;
    }
 
    public void tick() {
-      if (!this.isSittingOnShoulder && !this.entity.isSitting() && !this.entity.isLeashed()) {
+      if (!this.isSittingOnShoulder && !this.entity.isInSittingPose() && !this.entity.isLeashed()) {
          if (this.entity.getBoundingBox().intersects(this.owner.getBoundingBox())) {
             this.isSittingOnShoulder = this.entity.setEntityOnShoulder(this.owner);
          }
 
       }
