--- net/minecraft/world/entity/ai/goal/MoveThroughVillageGoal.java
+++ net/minecraft/world/entity/ai/goal/MoveThroughVillageGoal.java
@@ -12,10 +12,11 @@
 import net.minecraft.world.entity.PathfinderMob;
 import net.minecraft.world.entity.ai.navigation.GroundPathNavigation;
 import net.minecraft.world.entity.ai.util.RandomPos;
 import net.minecraft.world.entity.ai.village.poi.PoiManager;
 import net.minecraft.world.entity.ai.village.poi.PoiType;
+import net.minecraft.world.level.block.DoorBlock;
 import net.minecraft.world.level.pathfinder.Node;
 import net.minecraft.world.level.pathfinder.Path;
 import net.minecraft.world.phys.Vec3;
 
 public class MoveThroughVillageGoal extends Goal {
@@ -44,12 +45,12 @@
       this.updateVisited();
       if (this.onlyAtNight && this.mob.level.isDay()) {
          return false;
       } else {
          ServerLevel var1 = (ServerLevel)this.mob.level;
-         BlockPos var2 = new BlockPos(this.mob);
-         if (!var1.closeToVillage(var2, 6)) {
+         BlockPos var2 = this.mob.blockPosition();
+         if (!var1.isCloseToVillage(var2, 6)) {
             return false;
          } else {
             Vec3 var3 = RandomPos.getLandPos(this.mob, 15, 7, (var3x) -> {
                if (!var1.isVillage(var3x)) {
                   return Double.NEGATIVE_INFINITY;
@@ -70,11 +71,11 @@
                   boolean var6 = var5.canOpenDoors();
                   var5.setCanOpenDoors(this.canDealWithDoors.getAsBoolean());
                   this.path = var5.createPath((BlockPos)this.poiPos, 0);
                   var5.setCanOpenDoors(var6);
                   if (this.path == null) {
-                     Vec3 var7 = RandomPos.getPosTowards(this.mob, 10, 7, new Vec3(this.poiPos));
+                     Vec3 var7 = RandomPos.getPosTowards(this.mob, 10, 7, Vec3.atBottomCenterOf(this.poiPos));
                      if (var7 == null) {
                         return false;
                      }
 
                      var5.setCanOpenDoors(this.canDealWithDoors.getAsBoolean());
@@ -86,11 +87,11 @@
                   }
 
                   for(int var10 = 0; var10 < this.path.getSize(); ++var10) {
                      Node var8 = this.path.get(var10);
                      BlockPos var9 = new BlockPos(var8.x, var8.y + 1, var8.z);
-                     if (DoorInteractGoal.isDoor(this.mob.level, var9)) {
+                     if (DoorBlock.isWoodenDoor(this.mob.level, var9)) {
                         this.path = this.mob.getNavigation().createPath((double)var8.x, (double)var8.y, (double)var8.z, 0);
                         break;
                      }
                   }
 
