--- net/minecraft/world/entity/ai/goal/StrollThroughVillageGoal.java
+++ net/minecraft/world/entity/ai/goal/StrollThroughVillageGoal.java
@@ -31,12 +31,12 @@
          return false;
       } else if (this.mob.getRandom().nextInt(this.interval) != 0) {
          return false;
       } else {
          ServerLevel var1 = (ServerLevel)this.mob.level;
-         BlockPos var2 = new BlockPos(this.mob);
-         if (!var1.closeToVillage(var2, 6)) {
+         BlockPos var2 = this.mob.blockPosition();
+         if (!var1.isCloseToVillage(var2, 6)) {
             return false;
          } else {
             Vec3 var3 = RandomPos.getLandPos(this.mob, 15, 7, (var1x) -> {
                return (double)(-var1.sectionsToVillage(SectionPos.of(var1x)));
             });
@@ -52,11 +52,11 @@
 
    public void tick() {
       if (this.wantedPos != null) {
          PathNavigation var1 = this.mob.getNavigation();
          if (var1.isDone() && !this.wantedPos.closerThan(this.mob.position(), 10.0D)) {
-            Vec3 var2 = new Vec3(this.wantedPos);
+            Vec3 var2 = Vec3.atBottomCenterOf(this.wantedPos);
             Vec3 var3 = this.mob.position();
             Vec3 var4 = var3.subtract(var2);
             var2 = var4.scale(0.4D).add(var2);
             Vec3 var5 = var2.subtract(var3).normalize().scale(10.0D).add(var3);
             BlockPos var6 = new BlockPos(var5);
@@ -69,9 +69,9 @@
       }
    }
 
    private void moveRandomly() {
       Random var1 = this.mob.getRandom();
-      BlockPos var2 = this.mob.level.getHeightmapPos(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, (new BlockPos(this.mob)).offset(-8 + var1.nextInt(16), 0, -8 + var1.nextInt(16)));
+      BlockPos var2 = this.mob.level.getHeightmapPos(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, this.mob.blockPosition().offset(-8 + var1.nextInt(16), 0, -8 + var1.nextInt(16)));
       this.mob.getNavigation().moveTo((double)var2.getX(), (double)var2.getY(), (double)var2.getZ(), 1.0D);
    }
 }
