--- net/minecraft/network/Varint21FrameDecoder.java
+++ net/minecraft/network/Varint21FrameDecoder.java
@@ -25,16 +25,16 @@
          if (var4[var5] >= 0) {
             FriendlyByteBuf var6 = new FriendlyByteBuf(Unpooled.wrappedBuffer(var4));
 
             try {
                int var7 = var6.readVarInt();
-               if (var2.readableBytes() >= var7) {
-                  var3.add(var2.readBytes(var7));
+               if (var2.readableBytes() < var7) {
+                  var2.resetReaderIndex();
                   return;
                }
 
-               var2.resetReaderIndex();
+               var3.add(var2.readBytes(var7));
             } finally {
                var6.release();
             }
 
             return;
