--- net/minecraft/client/gui/screens/inventory/BeaconScreen.java
+++ net/minecraft/client/gui/screens/inventory/BeaconScreen.java
@@ -1,17 +1,21 @@
 package net.minecraft.client.gui.screens.inventory;
 
 import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
 import java.util.Iterator;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.components.AbstractButton;
 import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.screens.Screen;
 import net.minecraft.client.renderer.texture.TextureAtlasSprite;
 import net.minecraft.client.resources.language.I18n;
 import net.minecraft.core.NonNullList;
+import net.minecraft.network.chat.CommonComponents;
 import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.network.chat.TranslatableComponent;
 import net.minecraft.network.protocol.Packet;
 import net.minecraft.network.protocol.game.ServerboundContainerClosePacket;
 import net.minecraft.network.protocol.game.ServerboundSetBeaconPacket;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.effect.MobEffect;
@@ -112,43 +116,44 @@
       }
 
       this.confirmButton.active = ((BeaconMenu)this.menu).hasPayment() && this.primary != null;
    }
 
-   protected void renderLabels(int var1, int var2) {
-      this.drawCenteredString(this.font, I18n.get("block.minecraft.beacon.primary"), 62, 10, 14737632);
-      this.drawCenteredString(this.font, I18n.get("block.minecraft.beacon.secondary"), 169, 10, 14737632);
-      Iterator var3 = this.buttons.iterator();
+   protected void renderLabels(PoseStack var1, int var2, int var3) {
+      this.drawCenteredString(var1, this.font, I18n.get("block.minecraft.beacon.primary"), 62, 10, 14737632);
+      this.drawCenteredString(var1, this.font, I18n.get("block.minecraft.beacon.secondary"), 169, 10, 14737632);
+      Iterator var4 = this.buttons.iterator();
 
-      while(var3.hasNext()) {
-         AbstractWidget var4 = (AbstractWidget)var3.next();
-         if (var4.isHovered()) {
-            var4.renderToolTip(var1 - this.leftPos, var2 - this.topPos);
+      while(var4.hasNext()) {
+         AbstractWidget var5 = (AbstractWidget)var4.next();
+         if (var5.isHovered()) {
+            var5.renderToolTip(var1, var2 - this.leftPos, var3 - this.topPos);
             break;
          }
       }
 
    }
 
-   protected void renderBg(float var1, int var2, int var3) {
+   protected void renderBg(PoseStack var1, float var2, int var3, int var4) {
       RenderSystem.color4f(1.0F, 1.0F, 1.0F, 1.0F);
       this.minecraft.getTextureManager().bind(BEACON_LOCATION);
-      int var4 = (this.width - this.imageWidth) / 2;
-      int var5 = (this.height - this.imageHeight) / 2;
-      this.blit(var4, var5, 0, 0, this.imageWidth, this.imageHeight);
+      int var5 = (this.width - this.imageWidth) / 2;
+      int var6 = (this.height - this.imageHeight) / 2;
+      this.blit(var1, var5, var6, 0, 0, this.imageWidth, this.imageHeight);
       this.itemRenderer.blitOffset = 100.0F;
-      this.itemRenderer.renderAndDecorateItem(new ItemStack(Items.EMERALD), var4 + 42, var5 + 109);
-      this.itemRenderer.renderAndDecorateItem(new ItemStack(Items.DIAMOND), var4 + 42 + 22, var5 + 109);
-      this.itemRenderer.renderAndDecorateItem(new ItemStack(Items.GOLD_INGOT), var4 + 42 + 44, var5 + 109);
-      this.itemRenderer.renderAndDecorateItem(new ItemStack(Items.IRON_INGOT), var4 + 42 + 66, var5 + 109);
+      this.itemRenderer.renderAndDecorateItem(new ItemStack(Items.NETHERITE_INGOT), var5 + 20, var6 + 109);
+      this.itemRenderer.renderAndDecorateItem(new ItemStack(Items.EMERALD), var5 + 41, var6 + 109);
+      this.itemRenderer.renderAndDecorateItem(new ItemStack(Items.DIAMOND), var5 + 41 + 22, var6 + 109);
+      this.itemRenderer.renderAndDecorateItem(new ItemStack(Items.GOLD_INGOT), var5 + 42 + 44, var6 + 109);
+      this.itemRenderer.renderAndDecorateItem(new ItemStack(Items.IRON_INGOT), var5 + 42 + 66, var6 + 109);
       this.itemRenderer.blitOffset = 0.0F;
    }
 
-   public void render(int var1, int var2, float var3) {
-      this.renderBackground();
-      super.render(var1, var2, var3);
-      this.renderTooltip(var1, var2);
+   public void render(PoseStack var1, int var2, int var3, float var4) {
+      this.renderBackground(var1);
+      super.render(var1, var2, var3, var4);
+      this.renderTooltip(var1, var2, var3);
    }
 
    class BeaconCancelButton extends BeaconScreen.BeaconSpriteScreenButton {
       public BeaconCancelButton(int var2, int var3) {
          super(var2, var3, 112, 220);
@@ -157,12 +162,12 @@
       public void onPress() {
          BeaconScreen.this.minecraft.player.connection.send((Packet)(new ServerboundContainerClosePacket(BeaconScreen.this.minecraft.player.containerMenu.containerId)));
          BeaconScreen.this.minecraft.setScreen((Screen)null);
       }
 
-      public void renderToolTip(int var1, int var2) {
-         BeaconScreen.this.renderTooltip(I18n.get("gui.cancel"), var1, var2);
+      public void renderToolTip(PoseStack var1, int var2, int var3) {
+         BeaconScreen.this.renderTooltip(var1, CommonComponents.GUI_CANCEL, var2, var3);
       }
    }
 
    class BeaconConfirmButton extends BeaconScreen.BeaconSpriteScreenButton {
       public BeaconConfirmButton(int var2, int var3) {
@@ -173,12 +178,12 @@
          BeaconScreen.this.minecraft.getConnection().send((Packet)(new ServerboundSetBeaconPacket(MobEffect.getId(BeaconScreen.this.primary), MobEffect.getId(BeaconScreen.this.secondary))));
          BeaconScreen.this.minecraft.player.connection.send((Packet)(new ServerboundContainerClosePacket(BeaconScreen.this.minecraft.player.containerMenu.containerId)));
          BeaconScreen.this.minecraft.setScreen((Screen)null);
       }
 
-      public void renderToolTip(int var1, int var2) {
-         BeaconScreen.this.renderTooltip(I18n.get("gui.done"), var1, var2);
+      public void renderToolTip(PoseStack var1, int var2, int var3) {
+         BeaconScreen.this.renderTooltip(var1, CommonComponents.GUI_DONE, var2, var3);
       }
    }
 
    abstract static class BeaconSpriteScreenButton extends BeaconScreen.BeaconScreenButton {
       private final int iconX;
@@ -188,12 +193,12 @@
          super(var1, var2);
          this.iconX = var3;
          this.iconY = var4;
       }
 
-      protected void renderIcon() {
-         this.blit(this.x + 2, this.y + 2, this.iconX, this.iconY, 18, 18);
+      protected void renderIcon(PoseStack var1) {
+         this.blit(var1, this.x + 2, this.y + 2, this.iconX, this.iconY, 18, 18);
       }
    }
 
    class BeaconPowerButton extends BeaconScreen.BeaconScreenButton {
       private final MobEffect effect;
@@ -220,50 +225,50 @@
             BeaconScreen.this.init();
             BeaconScreen.this.tick();
          }
       }
 
-      public void renderToolTip(int var1, int var2) {
-         String var3 = I18n.get(this.effect.getDescriptionId());
+      public void renderToolTip(PoseStack var1, int var2, int var3) {
+         TranslatableComponent var4 = new TranslatableComponent(this.effect.getDescriptionId());
          if (!this.isPrimary && this.effect != MobEffects.REGENERATION) {
-            var3 = var3 + " II";
+            var4.append(" II");
          }
 
-         BeaconScreen.this.renderTooltip(var3, var1, var2);
+         BeaconScreen.this.renderTooltip(var1, var4, var2, var3);
       }
 
-      protected void renderIcon() {
+      protected void renderIcon(PoseStack var1) {
          Minecraft.getInstance().getTextureManager().bind(this.sprite.atlas().location());
-         blit(this.x + 2, this.y + 2, this.getBlitOffset(), 18, 18, this.sprite);
+         blit(var1, this.x + 2, this.y + 2, this.getBlitOffset(), 18, 18, this.sprite);
       }
    }
 
    abstract static class BeaconScreenButton extends AbstractButton {
       private boolean selected;
 
       protected BeaconScreenButton(int var1, int var2) {
-         super(var1, var2, 22, 22, "");
+         super(var1, var2, 22, 22, TextComponent.EMPTY);
       }
 
-      public void renderButton(int var1, int var2, float var3) {
+      public void renderButton(PoseStack var1, int var2, int var3, float var4) {
          Minecraft.getInstance().getTextureManager().bind(BeaconScreen.BEACON_LOCATION);
          RenderSystem.color4f(1.0F, 1.0F, 1.0F, 1.0F);
-         boolean var4 = true;
-         int var5 = 0;
+         boolean var5 = true;
+         int var6 = 0;
          if (!this.active) {
-            var5 += this.width * 2;
+            var6 += this.width * 2;
          } else if (this.selected) {
-            var5 += this.width * 1;
+            var6 += this.width * 1;
          } else if (this.isHovered()) {
-            var5 += this.width * 3;
+            var6 += this.width * 3;
          }
 
-         this.blit(this.x, this.y, var5, 219, this.width, this.height);
-         this.renderIcon();
+         this.blit(var1, this.x, this.y, var6, 219, this.width, this.height);
+         this.renderIcon(var1);
       }
 
-      protected abstract void renderIcon();
+      protected abstract void renderIcon(PoseStack var1);
 
       public boolean isSelected() {
          return this.selected;
       }
 
