--- net/minecraft/client/gui/screens/inventory/LecternScreen.java
+++ net/minecraft/client/gui/screens/inventory/LecternScreen.java
@@ -1,12 +1,13 @@
 package net.minecraft.client.gui.screens.inventory;
 
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.screens.Screen;
-import net.minecraft.client.resources.language.I18n;
 import net.minecraft.core.NonNullList;
+import net.minecraft.network.chat.CommonComponents;
 import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TranslatableComponent;
 import net.minecraft.world.entity.player.Inventory;
 import net.minecraft.world.inventory.AbstractContainerMenu;
 import net.minecraft.world.inventory.ContainerListener;
 import net.minecraft.world.inventory.LecternMenu;
 import net.minecraft.world.item.ItemStack;
@@ -53,14 +54,14 @@
       this.menu.removeSlotListener(this.listener);
    }
 
    protected void createMenuControls() {
       if (this.minecraft.player.mayBuild()) {
-         this.addButton(new Button(this.width / 2 - 100, 196, 98, 20, I18n.get("gui.done"), (var1) -> {
+         this.addButton(new Button(this.width / 2 - 100, 196, 98, 20, CommonComponents.GUI_DONE, (var1) -> {
             this.minecraft.setScreen((Screen)null);
          }));
-         this.addButton(new Button(this.width / 2 + 2, 196, 98, 20, I18n.get("lectern.take_book"), (var1) -> {
+         this.addButton(new Button(this.width / 2 + 2, 196, 98, 20, new TranslatableComponent("lectern.take_book"), (var1) -> {
             this.sendButtonClick(3);
          }));
       } else {
          super.createMenuControls();
       }
