--- net/minecraft/world/entity/ai/behavior/PositionTracker.java
+++ net/minecraft/world/entity/ai/behavior/PositionTracker.java
@@ -1,0 +1,13 @@
+package net.minecraft.world.entity.ai.behavior;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.phys.Vec3;
+
+public interface PositionTracker {
+   Vec3 currentPosition();
+
+   BlockPos currentBlockPosition();
+
+   boolean isVisibleBy(LivingEntity var1);
+}
