--- net/minecraft/world/entity/ai/behavior/ShowTradesToPlayer.java
+++ net/minecraft/world/entity/ai/behavior/ShowTradesToPlayer.java
@@ -110,11 +110,11 @@
    }
 
    private LivingEntity lookAtTarget(Villager var1) {
       Brain var2 = var1.getBrain();
       LivingEntity var3 = (LivingEntity)var2.getMemory(MemoryModuleType.INTERACTION_TARGET).get();
-      var2.setMemory(MemoryModuleType.LOOK_TARGET, (Object)(new EntityPosWrapper(var3)));
+      var2.setMemory(MemoryModuleType.LOOK_TARGET, (Object)(new EntityTracker(var3, true)));
       return var3;
    }
 
    private void displayCyclingItems(Villager var1) {
       if (this.displayItems.size() >= 2 && ++this.cycleCounter >= 40) {
