--- net/minecraft/data/tags/FluidTagsProvider.java
+++ net/minecraft/data/tags/FluidTagsProvider.java
@@ -3,11 +3,10 @@
 import java.nio.file.Path;
 import net.minecraft.core.Registry;
 import net.minecraft.data.DataGenerator;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.tags.FluidTags;
-import net.minecraft.tags.TagCollection;
 import net.minecraft.world.level.material.Fluid;
 import net.minecraft.world.level.material.Fluids;
 
 public class FluidTagsProvider extends TagsProvider<Fluid> {
    public FluidTagsProvider(DataGenerator var1) {
@@ -24,10 +23,6 @@
    }
 
    public String getName() {
       return "Fluid Tags";
    }
-
-   protected void useTags(TagCollection<Fluid> var1) {
-      FluidTags.reset(var1);
-   }
 }
