--- net/minecraft/server/commands/ClearInventoryCommands.java
+++ net/minecraft/server/commands/ClearInventoryCommands.java
@@ -48,18 +48,19 @@
       int var4 = 0;
       Iterator var5 = var1.iterator();
 
       while(var5.hasNext()) {
          ServerPlayer var6 = (ServerPlayer)var5.next();
-         var4 += var6.inventory.clearInventory(var2, var3);
+         var4 += var6.inventory.clearOrCountMatchingItems(var2, var3, var6.inventoryMenu.getCraftSlots());
          var6.containerMenu.broadcastChanges();
+         var6.inventoryMenu.slotsChanged(var6.inventory);
          var6.broadcastCarriedItem();
       }
 
       if (var4 == 0) {
          if (var1.size() == 1) {
-            throw ERROR_SINGLE.create(((ServerPlayer)var1.iterator().next()).getName().getColoredString());
+            throw ERROR_SINGLE.create(((ServerPlayer)var1.iterator().next()).getName());
          } else {
             throw ERROR_MULTIPLE.create(var1.size());
          }
       } else {
          if (var3 == 0) {
