--- net/minecraft/server/commands/FillCommand.java
+++ net/minecraft/server/commands/FillCommand.java
@@ -104,11 +104,11 @@
       }
    }
 
    static {
       HOLLOW_CORE = new BlockInput(Blocks.AIR.defaultBlockState(), Collections.emptySet(), (CompoundTag)null);
-      ERROR_FAILED = new SimpleCommandExceptionType(new TranslatableComponent("commands.fill.failed", new Object[0]));
+      ERROR_FAILED = new SimpleCommandExceptionType(new TranslatableComponent("commands.fill.failed"));
    }
 
    static enum Mode {
       REPLACE((var0, var1, var2, var3) -> {
          return var2;
