--- net/minecraft/server/commands/GameModeCommand.java
+++ net/minecraft/server/commands/GameModeCommand.java
@@ -4,10 +4,11 @@
 import com.mojang.brigadier.builder.LiteralArgumentBuilder;
 import com.mojang.brigadier.context.CommandContext;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.Iterator;
+import net.minecraft.Util;
 import net.minecraft.commands.CommandSourceStack;
 import net.minecraft.commands.Commands;
 import net.minecraft.commands.arguments.EntityArgument;
 import net.minecraft.network.chat.TranslatableComponent;
 import net.minecraft.server.level.ServerPlayer;
@@ -35,16 +36,16 @@
 
       var0.register(var1);
    }
 
    private static void logGamemodeChange(CommandSourceStack var0, ServerPlayer var1, GameType var2) {
-      TranslatableComponent var3 = new TranslatableComponent("gameMode." + var2.getName(), new Object[0]);
+      TranslatableComponent var3 = new TranslatableComponent("gameMode." + var2.getName());
       if (var0.getEntity() == var1) {
          var0.sendSuccess(new TranslatableComponent("commands.gamemode.success.self", new Object[]{var3}), true);
       } else {
          if (var0.getLevel().getGameRules().getBoolean(GameRules.RULE_SENDCOMMANDFEEDBACK)) {
-            var1.sendMessage(new TranslatableComponent("gameMode.changed", new Object[]{var3}));
+            var1.sendMessage(new TranslatableComponent("gameMode.changed", new Object[]{var3}), Util.NIL_UUID);
          }
 
          var0.sendSuccess(new TranslatableComponent("commands.gamemode.success.other", new Object[]{var1.getDisplayName(), var3}), true);
       }
 
