--- net/minecraft/server/commands/ListPlayersCommand.java
+++ net/minecraft/server/commands/ListPlayersCommand.java
@@ -6,10 +6,11 @@
 import java.util.function.Function;
 import net.minecraft.commands.CommandSourceStack;
 import net.minecraft.commands.Commands;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.ComponentUtils;
+import net.minecraft.network.chat.MutableComponent;
 import net.minecraft.network.chat.TranslatableComponent;
 import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.server.players.PlayerList;
 import net.minecraft.world.entity.player.Player;
 
@@ -31,10 +32,10 @@
    }
 
    private static int format(CommandSourceStack var0, Function<ServerPlayer, Component> var1) {
       PlayerList var2 = var0.getServer().getPlayerList();
       List var3 = var2.getPlayers();
-      Component var4 = ComponentUtils.formatList(var3, var1);
+      MutableComponent var4 = ComponentUtils.formatList(var3, var1);
       var0.sendSuccess(new TranslatableComponent("commands.list.players", new Object[]{var3.size(), var2.getMaxPlayers(), var4}), false);
       return var3.size();
    }
 }
