--- com/mojang/realmsclient/client/RealmsError.java
+++ com/mojang/realmsclient/client/RealmsError.java
@@ -6,24 +6,30 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class RealmsError {
    private static final Logger LOGGER = LogManager.getLogger();
-   private String errorMessage;
-   private int errorCode;
+   private final String errorMessage;
+   private final int errorCode;
 
-   public RealmsError(String var1) {
+   private RealmsError(String var1, int var2) {
+      this.errorMessage = var1;
+      this.errorCode = var2;
+   }
+
+   public static RealmsError create(String var0) {
       try {
-         JsonParser var2 = new JsonParser();
-         JsonObject var3 = var2.parse(var1).getAsJsonObject();
-         this.errorMessage = JsonUtils.getStringOr("errorMsg", var3, "");
-         this.errorCode = JsonUtils.getIntOr("errorCode", var3, -1);
-      } catch (Exception var4) {
-         LOGGER.error("Could not parse RealmsError: " + var4.getMessage());
-         LOGGER.error("The error was: " + var1);
+         JsonParser var1 = new JsonParser();
+         JsonObject var2 = var1.parse(var0).getAsJsonObject();
+         String var3 = JsonUtils.getStringOr("errorMsg", var2, "");
+         int var4 = JsonUtils.getIntOr("errorCode", var2, -1);
+         return new RealmsError(var3, var4);
+      } catch (Exception var5) {
+         LOGGER.error("Could not parse RealmsError: " + var5.getMessage());
+         LOGGER.error("The error was: " + var0);
+         return new RealmsError("Failed to parse response from server", -1);
       }
-
    }
 
    public String getErrorMessage() {
       return this.errorMessage;
    }
