--- net/minecraft/world/level/ChunkPos.java
+++ net/minecraft/world/level/ChunkPos.java
@@ -90,20 +90,16 @@
 
    public int getRegionLocalZ() {
       return this.z & 31;
    }
 
-   public BlockPos getBlockAt(int var1, int var2, int var3) {
-      return new BlockPos((this.x << 4) + var1, var2, (this.z << 4) + var3);
-   }
-
    public String toString() {
       return "[" + this.x + ", " + this.z + "]";
    }
 
    public BlockPos getWorldPosition() {
-      return new BlockPos(this.x << 4, 0, this.z << 4);
+      return new BlockPos(this.getMinBlockX(), 0, this.getMinBlockZ());
    }
 
    public int getChessboardDistance(ChunkPos var1) {
       return Math.max(Math.abs(this.x - var1.x), Math.abs(this.z - var1.z));
    }
