--- net/minecraft/world/level/levelgen/feature/configurations/DecoratorConfiguration.java
+++ net/minecraft/world/level/levelgen/feature/configurations/DecoratorConfiguration.java
@@ -1,10 +1,5 @@
 package net.minecraft.world.level.levelgen.feature.configurations;
 
-import com.mojang.datafixers.Dynamic;
-import com.mojang.datafixers.types.DynamicOps;
-
 public interface DecoratorConfiguration {
-   NoneDecoratorConfiguration NONE = new NoneDecoratorConfiguration();
-
-   <T> Dynamic<T> serialize(DynamicOps<T> var1);
+   NoneDecoratorConfiguration NONE = NoneDecoratorConfiguration.INSTANCE;
 }
