--- net/minecraft/world/level/levelgen/feature/configurations/MultiJigsawConfiguration.java
+++ net/minecraft/world/level/levelgen/feature/configurations/MultiJigsawConfiguration.java
@@ -1,0 +1,33 @@
+package net.minecraft.world.level.levelgen.feature.configurations;
+
+import com.mojang.serialization.Codec;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.stream.Collectors;
+import net.minecraft.resources.ResourceLocation;
+
+public class MultiJigsawConfiguration implements FeatureConfiguration {
+   public static final Codec<MultiJigsawConfiguration> CODEC;
+   private final List<JigsawConfiguration> configurations;
+
+   public MultiJigsawConfiguration(Map<String, Integer> var1) {
+      this((List)var1.entrySet().stream().map((var0) -> {
+         return new JigsawConfiguration(new ResourceLocation((String)var0.getKey()), (Integer)var0.getValue());
+      }).collect(Collectors.toList()));
+   }
+
+   private MultiJigsawConfiguration(List<JigsawConfiguration> var1) {
+      this.configurations = var1;
+   }
+
+   public JigsawConfiguration getRandomPool(Random var1) {
+      return (JigsawConfiguration)this.configurations.get(var1.nextInt(this.configurations.size()));
+   }
+
+   static {
+      CODEC = JigsawConfiguration.CODEC.listOf().xmap(MultiJigsawConfiguration::new, (var0) -> {
+         return var0.configurations;
+      });
+   }
+}
