--- net/minecraft/client/model/DrownedModel.java
+++ net/minecraft/client/model/DrownedModel.java
@@ -49,14 +49,14 @@
          this.rightArm.xRot = this.rightArm.xRot * 0.5F - 3.1415927F;
          this.rightArm.yRot = 0.0F;
       }
 
       if (this.swimAmount > 0.0F) {
-         this.rightArm.xRot = this.rotlerpRad(this.rightArm.xRot, -2.5132742F, this.swimAmount) + this.swimAmount * 0.35F * Mth.sin(0.1F * var4);
-         this.leftArm.xRot = this.rotlerpRad(this.leftArm.xRot, -2.5132742F, this.swimAmount) - this.swimAmount * 0.35F * Mth.sin(0.1F * var4);
-         this.rightArm.zRot = this.rotlerpRad(this.rightArm.zRot, -0.15F, this.swimAmount);
-         this.leftArm.zRot = this.rotlerpRad(this.leftArm.zRot, 0.15F, this.swimAmount);
+         this.rightArm.xRot = this.rotlerpRad(this.swimAmount, this.rightArm.xRot, -2.5132742F) + this.swimAmount * 0.35F * Mth.sin(0.1F * var4);
+         this.leftArm.xRot = this.rotlerpRad(this.swimAmount, this.leftArm.xRot, -2.5132742F) - this.swimAmount * 0.35F * Mth.sin(0.1F * var4);
+         this.rightArm.zRot = this.rotlerpRad(this.swimAmount, this.rightArm.zRot, -0.15F);
+         this.leftArm.zRot = this.rotlerpRad(this.swimAmount, this.leftArm.zRot, 0.15F);
          ModelPart var10000 = this.leftLeg;
          var10000.xRot -= this.swimAmount * 0.55F * Mth.sin(0.1F * var4);
          var10000 = this.rightLeg;
          var10000.xRot += this.swimAmount * 0.55F * Mth.sin(0.1F * var4);
          this.head.xRot = 0.0F;
