--- net/minecraft/client/model/ShulkerModel.java
+++ net/minecraft/client/model/ShulkerModel.java
@@ -1,18 +1,20 @@
 package net.minecraft.client.model;
 
 import com.google.common.collect.ImmutableList;
 import net.minecraft.client.model.geom.ModelPart;
+import net.minecraft.client.renderer.RenderType;
 import net.minecraft.util.Mth;
 import net.minecraft.world.entity.monster.Shulker;
 
 public class ShulkerModel<T extends Shulker> extends ListModel<T> {
    private final ModelPart base = new ModelPart(64, 64, 0, 28);
    private final ModelPart lid = new ModelPart(64, 64, 0, 0);
    private final ModelPart head = new ModelPart(64, 64, 0, 52);
 
    public ShulkerModel() {
+      super(RenderType::entityCutoutNoCullZOffset);
       this.lid.addBox(-8.0F, -16.0F, -8.0F, 16.0F, 12.0F, 16.0F);
       this.lid.setPos(0.0F, 24.0F, 0.0F);
       this.base.addBox(-8.0F, -8.0F, -8.0F, 16.0F, 8.0F, 16.0F);
       this.base.setPos(0.0F, 24.0F, 0.0F);
       this.head.addBox(-3.0F, 0.0F, -3.0F, 6.0F, 6.0F, 6.0F);
@@ -34,11 +36,11 @@
       } else {
          this.lid.yRot = 0.0F;
       }
 
       this.head.xRot = var6 * 0.017453292F;
-      this.head.yRot = var5 * 0.017453292F;
+      this.head.yRot = (var1.yHeadRot - 180.0F - var1.yBodyRot) * 0.017453292F;
    }
 
    public Iterable<ModelPart> parts() {
       return ImmutableList.of(this.base, this.lid);
    }
