--- com/mojang/datafixers/functions/Fold.java
+++ com/mojang/datafixers/functions/Fold.java
@@ -1,15 +1,15 @@
 package com.mojang.datafixers.functions;
 
 import com.google.common.collect.Maps;
 import com.mojang.datafixers.RewriteResult;
-import com.mojang.datafixers.types.DynamicOps;
 import com.mojang.datafixers.types.families.Algebra;
 import com.mojang.datafixers.types.families.RecursiveTypeFamily;
 import com.mojang.datafixers.types.families.TypeFamily;
 import com.mojang.datafixers.types.templates.RecursivePoint;
 import com.mojang.datafixers.util.Pair;
+import com.mojang.serialization.DynamicOps;
 import java.util.Map;
 import java.util.Objects;
 import java.util.function.Function;
 import java.util.function.IntFunction;
 
