--- com/mojang/datafixers/functions/ProfunctorTransformer.java
+++ com/mojang/datafixers/functions/ProfunctorTransformer.java
@@ -1,11 +1,11 @@
 package com.mojang.datafixers.functions;
 
 import com.mojang.datafixers.FunctionType;
 import com.mojang.datafixers.kinds.App2;
 import com.mojang.datafixers.optics.Optic;
-import com.mojang.datafixers.types.DynamicOps;
+import com.mojang.serialization.DynamicOps;
 import java.util.Objects;
 import java.util.function.Function;
 
 final class ProfunctorTransformer<S, T, A, B> extends PointFree<Function<Function<A, B>, Function<S, T>>> {
    protected final Optic<? super FunctionType.Instance.Mu, S, T, A, B> optic;
