--- net/minecraft/world/level/storage/loot/functions/LootItemFunctionType.java
+++ net/minecraft/world/level/storage/loot/functions/LootItemFunctionType.java
@@ -1,0 +1,10 @@
+package net.minecraft.world.level.storage.loot.functions;
+
+import net.minecraft.world.level.storage.loot.Serializer;
+import net.minecraft.world.level.storage.loot.SerializerType;
+
+public class LootItemFunctionType extends SerializerType<LootItemFunction> {
+   public LootItemFunctionType(Serializer<? extends LootItemFunction> var1) {
+      super(var1);
+   }
+}
