--- net/minecraft/world/level/storage/loot/functions/SetContainerLootTable.java
+++ net/minecraft/world/level/storage/loot/functions/SetContainerLootTable.java
@@ -20,10 +20,14 @@
       super(var1);
       this.name = var2;
       this.seed = var3;
    }
 
+   public LootItemFunctionType getType() {
+      return LootItemFunctions.SET_LOOT_TABLE;
+   }
+
    public ItemStack run(ItemStack var1, LootContext var2) {
       if (var1.isEmpty()) {
          return var1;
       } else {
          CompoundTag var3 = new CompoundTag();
@@ -51,12 +55,11 @@
 
       }
    }
 
    public static class Serializer extends LootItemConditionalFunction.Serializer<SetContainerLootTable> {
-      protected Serializer() {
-         super(new ResourceLocation("set_loot_table"), SetContainerLootTable.class);
+      public Serializer() {
       }
 
       public void serialize(JsonObject var1, SetContainerLootTable var2, JsonSerializationContext var3) {
          super.serialize(var1, (LootItemConditionalFunction)var2, var3);
          var1.addProperty("name", var2.name.toString());
