--- net/minecraft/world/level/storage/loot/functions/SetStewEffectFunction.java
+++ net/minecraft/world/level/storage/loot/functions/SetStewEffectFunction.java
@@ -32,10 +32,14 @@
    private SetStewEffectFunction(LootItemCondition[] var1, Map<MobEffect, RandomValueBounds> var2) {
       super(var1);
       this.effectDurationMap = ImmutableMap.copyOf(var2);
    }
 
+   public LootItemFunctionType getType() {
+      return LootItemFunctions.SET_STEW_EFFECT;
+   }
+
    public ItemStack run(ItemStack var1, LootContext var2) {
       if (var1.getItem() == Items.SUSPICIOUS_STEW && !this.effectDurationMap.isEmpty()) {
          Random var3 = var2.getRandom();
          int var4 = var3.nextInt(this.effectDurationMap.size());
          Entry var5 = (Entry)Iterables.get(this.effectDurationMap.entrySet(), var4);
@@ -56,11 +60,10 @@
       return new SetStewEffectFunction.Builder();
    }
 
    public static class Serializer extends LootItemConditionalFunction.Serializer<SetStewEffectFunction> {
       public Serializer() {
-         super(new ResourceLocation("set_stew_effect"), SetStewEffectFunction.class);
       }
 
       public void serialize(JsonObject var1, SetStewEffectFunction var2, JsonSerializationContext var3) {
          super.serialize(var1, (LootItemConditionalFunction)var2, var3);
          if (!var2.effectDurationMap.isEmpty()) {
